{"nbformat":4,"nbformat_minor":0,"metadata":{"colab":{"name":"CreateTextcorpus","provenance":[],"collapsed_sections":[],"mount_file_id":"1verz2_5rtItNa_cdpjBStcaKhlX8N5Ir","authorship_tag":"ABX9TyM+uUvYv1A+O+yuA6MxdNRT"},"kernelspec":{"name":"python3","display_name":"Python 3"},"language_info":{"name":"python"}},"cells":[{"cell_type":"markdown","source":["STEP 1:\n","Extract texts from PubMed"],"metadata":{"id":"MZjFhe7rXg37"}},{"cell_type":"code","source":["import re\n","import requests\n","from xml.dom import minidom\n","import xml.etree.ElementTree as ET\n","\n","# todo: substitute with correct values:\n","path_to_pubmed_references = \"/content/drive/MyDrive/TextcorpusCreation/summary-geneexpres-set.txt\"\n","path_to_extracted_texts = \"/content/drive/MyDrive/ExtractedPubMedArticles/\"\n","\n","def get_pubmed_ids(filename):\n","  \"\"\"\n","  extract PubMed ids of articles \n","  from the file\n","  \"\"\"\n","  pubmed_ids = []\n","  pmid_regex = r'PMID: [0-9]+;'\n","  with open(filename, \"r\") as f:\n","    for line in f:\n","      match = re.search(pmid_regex, line)\n","      if match:\n","        result = match.group().replace(\"PMID: \", \"\").replace(\";\", \"\")\n","        pubmed_ids.append(result)\n","  return pubmed_ids\n","\n","def build_pubmed_links(pubmed_ids):\n","  \"\"\"\n","  create links to the articles in xml-format\n","  \"\"\"\n","  start = \"https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_xml/\"\n","  end = \"/unicode\"\n","  pubmed_links = []\n","  for pm_id in pubmed_ids:\n","    link = start + pm_id + end\n","    pubmed_links.append(link)\n","  return pubmed_links\n","\n","def clean_data(content: str) -> str:\n","  # characters in texts to remove\n","  noise = [\"\\x80\", \"\\x89\", \"±\", \"Â\", \"â\", \"Î\",\"¤\", \n","         \"\\x93\", \"\\x97\", \"Ã\", \"\\x88\", \"\\x92\",\n","         \"\", \"\", \"\", \"\", \"\", \"\", \"\", \"\",\n","          \"\", \"¶\", \"³\", \"®\", \"º\",  \"¼\", \"Î\",\n","          \"±\", \"\", \"Â\", \"²\", \"¤\", \"¥\", \"\", \"\" ]\n","  # clean the text\n","  for char in noise:\n","    content = content.replace(char, \"\")\n","  return content\n","\n","def get_method_description_from_article(uri):\n","  \"\"\"\n","  Go to the article with the address \n","  given as uri. The content is saved as an xml.\n","  We need only text located in the paragraph named Methods.\n","  So we extract content of the paragraph without headers\n","  \"\"\"\n","  text = [] \n","  try:\n","    response = requests.get(uri)\n","     # whole article\n","    content = response.text  \n","    root = ET.fromstring(content)\n","\n","    # all children of the 'document'-element that belong to 'passage'-type\n","    passages = root.findall(\"./document/passage\")  \n","    for passage in passages:\n","      infons = passage.findall(\"infon\")    \n","      section_type = infons[0]    \n","      text_type = infons[1]\n","\n","      # get passages that describe methods and materials\n","      if section_type.text == \"METHODS\" and text_type.text == \"paragraph\":\n","        content = passage.find(\"text\")\n","        cleaned_text = clean_data(content.text)\n","        # add cleaned text         \n","        text.append(cleaned_text)\n","  except: \n","    # write a link in the text file, if the article is not available \n","    print(\"Article is unavailable:\")\n","    print(uri)\n","  return text\n","   \n","\n","def write_file(data, filename, path):\n","  \"\"\"\n","  Write the given text into the file with the given name and\n","  save it in the given directory\n","  \"\"\"\n","  file_location = path + filename + \".txt\"\n","  with open(file_location, 'w') as f:\n","      f.write('\\n'.join(data))\n","\n","def extract_corpus():\n","  \"\"\"\n","  Extract information that contains description of methods\n","  from PubMed articles. This info is a base for a textcorpus \n","  needed to train our Transformer model\n","  \"\"\"\n","  # a file with summary information about medical articles that found in PubMed\n","  results = \"/content/drive/MyDrive/TextcorpusCreation/summary-geneexpres-set.txt\"\n","\n","  # directory where extracted texts will be saved\n","  path_to_textcorpus = \"/content/drive/MyDrive/ExtractedPubMedArticles/\"\n","\n","  # collect pubmed article ids\n","  pubmed_ids = get_pubmed_ids(results)\n","\n","  # create uris to pubmed articles\n","  pubmed_uris = build_pubmed_links(pubmed_ids)\n","\n","  counter = 0\n","  extracted_texts_counter = 0\n","  for uri in pubmed_uris:\n","    # extract method descriptions, that will be used as text data\n","    content = get_method_description_from_article(uri)  \n","    if len(content) > 0:\n","      # determine a name of the file, where the extracted text will be saved\n","      file_name = pubmed_ids[counter]\n","      \n","      # write content in file\n","      write_file(content, file_name, path_to_textcorpus)\n","      extracted_texts_counter += 1\n","    counter += 1\n"," \n","  print(\"extracted texts:\" + str(extracted_texts_counter))\n","\n","\n","# run the code\n","print(extract_corpus())\n","\n","\n","\n"],"metadata":{"id":"yE4wsSrmxjKZ","colab":{"base_uri":"https://localhost:8080/"},"outputId":"e0e70766-0dec-4af4-c7f2-69e45e3b3a86","executionInfo":{"status":"ok","timestamp":1654424124090,"user_tz":-120,"elapsed":91118,"user":{"displayName":"Victoria Dimanova","userId":"07172063658264964881"}}},"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["Article is unavailable:\n","https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_xml/29509546/unicode\n","Article is unavailable:\n","https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_xml/25502501/unicode\n","Article is unavailable:\n","https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_xml/32780725/unicode\n","Article is unavailable:\n","https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_xml/31990678/unicode\n","Article is unavailable:\n","https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_xml/26294732/unicode\n","Article is unavailable:\n","https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_xml/32790648/unicode\n","Article is unavailable:\n","https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_xml/31961829/unicode\n","Article is unavailable:\n","https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_xml/30126994/unicode\n","Article is unavailable:\n","https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_xml/29449276/unicode\n","Article is unavailable:\n","https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_xml/29105855/unicode\n","Article is unavailable:\n","https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_xml/22910291/unicode\n","Article is unavailable:\n","https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_xml/26123900/unicode\n","Article is unavailable:\n","https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_xml/23825117/unicode\n","Article is unavailable:\n","https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_xml/25896417/unicode\n","Article is unavailable:\n","https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_xml/24266327/unicode\n","Article is unavailable:\n","https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_xml/30275090/unicode\n","Article is unavailable:\n","https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_xml/31608930/unicode\n","Article is unavailable:\n","https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_xml/28197932/unicode\n","Article is unavailable:\n","https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_xml/29467325/unicode\n","Article is unavailable:\n","https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_xml/25172950/unicode\n","Article is unavailable:\n","https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_xml/23878197/unicode\n","Article is unavailable:\n","https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_xml/34250400/unicode\n","Article is unavailable:\n","https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_xml/29788332/unicode\n","Article is unavailable:\n","https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_xml/34376577/unicode\n","Article is unavailable:\n","https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_xml/32557565/unicode\n","Article is unavailable:\n","https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_xml/27297792/unicode\n","Article is unavailable:\n","https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_xml/32750039/unicode\n","Article is unavailable:\n","https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_xml/25061176/unicode\n","Article is unavailable:\n","https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_xml/28911141/unicode\n","Article is unavailable:\n","https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_xml/26098215/unicode\n","Article is unavailable:\n","https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_xml/27609643/unicode\n","Article is unavailable:\n","https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_xml/27103745/unicode\n","Article is unavailable:\n","https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_xml/25401470/unicode\n","Article is unavailable:\n","https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_xml/24038027/unicode\n","Article is unavailable:\n","https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_xml/23808760/unicode\n","Article is unavailable:\n","https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_xml/27281795/unicode\n","Article is unavailable:\n","https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_xml/27235135/unicode\n","Article is unavailable:\n","https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_xml/23886272/unicode\n","Article is unavailable:\n","https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_xml/24111526/unicode\n","Article is unavailable:\n","https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_xml/23632636/unicode\n","Article is unavailable:\n","https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_xml/22949664/unicode\n","Article is unavailable:\n","https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_xml/24048576/unicode\n","Article is unavailable:\n","https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_xml/23935018/unicode\n","Article is unavailable:\n","https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_xml/27710240/unicode\n","Article is unavailable:\n","https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_xml/25480661/unicode\n","Article is unavailable:\n","https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_xml/23687298/unicode\n","Article is unavailable:\n","https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_xml/22686479/unicode\n","Article is unavailable:\n","https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_xml/23382191/unicode\n","Article is unavailable:\n","https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_xml/25850433/unicode\n","Article is unavailable:\n","https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_xml/22682335/unicode\n","Article is unavailable:\n","https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_xml/23672216/unicode\n","Article is unavailable:\n","https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_xml/23941420/unicode\n","Article is unavailable:\n","https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_xml/23197858/unicode\n","Article is unavailable:\n","https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_xml/22761427/unicode\n","Article is unavailable:\n","https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_xml/23612980/unicode\n","Article is unavailable:\n","https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_xml/25452615/unicode\n","Article is unavailable:\n","https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_xml/28389592/unicode\n","Article is unavailable:\n","https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_xml/26002965/unicode\n","Article is unavailable:\n","https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_xml/26659918/unicode\n","Article is unavailable:\n","https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_xml/26195509/unicode\n","Article is unavailable:\n","https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_xml/25420915/unicode\n","Article is unavailable:\n","https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_xml/26088866/unicode\n","extracted texts:\n","220\n","None\n"]}]},{"cell_type":"code","source":["#check python version (just in case...)\n","import sys\n","print(sys.version)"],"metadata":{"id":"ioZbstISnUiZ","executionInfo":{"status":"ok","timestamp":1654780691105,"user_tz":-120,"elapsed":257,"user":{"displayName":"Victoria Dimanova","userId":"07172063658264964881"}},"outputId":"d6724417-23be-43ce-8518-98c93e5520d3","colab":{"base_uri":"https://localhost:8080/"}},"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["3.7.13 (default, Apr 24 2022, 01:04:09) \n","[GCC 7.5.0]\n"]}]},{"cell_type":"markdown","source":["STEP 2\n","\n","*   We have 220 extracted PubMed texts and the task is to label named entities in these texts. NEs of our interest are IVD concepts. \n","*   We try to automate labelling steps as much as possible.\n","*   First of all, we extract already tagged NEs from available annotated data and save them in a file.\n","*   per line: {concept:[lemma1, lemma2...]}\n","\n","\n","\n","\n","\n","\n"],"metadata":{"id":"4XPfnnYpmLMz"}},{"cell_type":"code","source":["import json\n","import nltk\n","from nltk.stem import WordNetLemmatizer \n","nltk.download('wordnet')\n","nltk.download('punkt')\n","\n","def find_substring(text: str, start: int, end: int) -> str:\n","  \"\"\"\n","  Find a substring located in the text \n","  between start and end index\n","  \"\"\"\n","  substring = \"\"  \n","  for x in range(start, end):    \n","    substring += text[x]  \n","  return substring \n","\n","lemmatizer = WordNetLemmatizer()\n","def lemmatize(concept: str):\n","  \"\"\"\n","  tokenize and lemmatize the given NE concept\n","  and get the result as a tuple(concept, [lemmas])\n","  \"\"\"  \n","  word_list = nltk.word_tokenize(concept)\n","  lemmatized_output = [lemmatizer.lemmatize(w) for w in word_list]\n","  return (concept, lemmatized_output)\n","\n","def write_data(data_to_write):\n","  \"\"\"\n","  write extracted concepts from the dictionary\n","  in the file\n","  \"\"\"\n","  # output info\n","  path_to_extracted_nes = \"/content/drive/MyDrive/TextcorpusCreation/extracted_nes.txt\" \n","  with open(path_to_extracted_nes, 'w') as f:\n","    f.write(json.dumps(ivd_entries))\n","\n"," \n","# extracted ivd concepts\n","ivd_entries = dict()\n","# path to a jsonl file with annotated texts\n","path_to_annotated = \"/content/drive/MyDrive/TextcorpusCreation/textkorpus_doccano.jsonl\"\n","\n","#start point\n","with open(path_to_annotated) as f:\n","  lines = f.readlines()\n","  for line in lines:      \n","    line_dict = json.loads(line)\n","    #get labels\n","    labels = line_dict[\"label\"]\n","    text = line_dict[\"data\"]    \n","    #find IVD\n","    for label in labels:\n","      start = label[0]      \n","      end = label[1]     \n","      concept = find_substring(text, start, end)\n","      #normalize\n","      (ne, lemmas) = lemmatize(concept)\n","      #save them in dictionary\n","      if ne not in ivd_entries:\n","        ivd_entries[ne] = lemmas\n","# write extracted entities in the file        \n","write_data(ivd_entries)\n","\n","        "],"metadata":{"id":"YjOZ9D5mqrUo"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["STEP 3:\n","To automate extraction, a special dictionary of already known NEs is created.\n","\n","To identify IVD concepts in unlabelled texts, following steps will be done:\n","*   read the new text\n","*   tokenize it\n","*   check if some entries from the extracted_nes.txt are available in the text, if yes -> label them as NE\n","*   check if some entries from above described dictionareis are available in the text. If yes -> label the as well\n","*   bring the data in the form that is expected by Label Studio\n","*   save in a json-file\n","\n","\n","STEP 3 and 4 are repeated as follows:\n","\n","\n","*   Annotate N texts (e.g. N = 10) using STEP 3\n","*   correct annotations using Label Studio\n","*   use manual annotations to extend a dictionary with NEs (extracted_nes.txt) using STEP 4\n","*   Repeat the cycle\n","\n","\n","\n","\n","\n","\n","\n","\n","\n","\n","\n","\n","\n","\n","\n","\n"," \n","\n","\n","\n","\n","\n"],"metadata":{"id":"2DFg8WZ89c2X"}},{"cell_type":"code","source":["import glob\n","import json\n","import nltk\n","from nltk.stem import WordNetLemmatizer \n","nltk.download('wordnet')\n","nltk.download('punkt')\n","nltk.download('omw-1.4')\n","\n","extracted_nes = \"/content/drive/MyDrive/TextcorpusCreation/extracted_nes.txt\"\n","path_to_unlabelled_texts = \"/content/drive/MyDrive/ExtractedPubMedArticles/\"\n","\n","all_files = glob.glob(path_to_unlabelled_texts + \"*.txt\")\n","\n","def read_data(path_to_file: str):\n","  \"\"\"\n","  path_to_file: path to a file with extracted\n","  read extracted named entities and save them in ivd_dict\n","  dictionary for further processing\n","  \"\"\"\n","  with open(extracted_nes) as f:\n","    data = f.read().strip()\n","    ivd_dict = json.loads(data)    \n","  return ivd_dict\n","    \n","def lemmatize_text(text: str):\n","  \"\"\"\n","  tokenize and lemmatize the given text \n","  and get the result as a list of tokens\n","  \"\"\"  \n","  tokenized_text = []\n","  word_list = nltk.word_tokenize(text)\n","  for w in word_list:\n","    lemma = lemmatizer.lemmatize(w)\n","    tokenized_text.append(lemma)\n","  return tokenized_text\n","\n","def handle_simple_concept(concept, concept_list, counter):\n","  \"\"\"\n","  append an index of the given concept \n","  in the list of concepts\n","  \"\"\"\n","  new_item = [counter]\n","  if new_item not in concept_list:\n","    concept_list.append(new_item)\n","\n","def handle_complex_concept(text_tokens, counter, concept, concept_list):\n","  \"\"\"\n","  determine whetter found tokens are a NE, if yes -> determine indices of these\n","  tokens in the text, create a list of them and add this list to the global\n","  list of concept positions\n","  text_tokens: tokenized and lemmatized text\n","  counter: help variable to control a position of a token in the text\n","  concept: a known NE tokenized and lemmatized\n","  concept_list: a list of list. Each entry contains positions of a found concept\n","  when the first token of the concept is found in the text\n","  check if other tokens of the same concepts are available.\n","  \"\"\" \n","  #add the found token in the list of concept tokens\n","  possible_concept_parts = [text_tokens[counter]]   \n","  concept_size = len(concept) \n","  \n","  #collect tokens with relevant indices in the text    \n","  for i in range(1, concept_size): \n","    if len(text_tokens) > (counter + i):\n","       possible_concept_parts.append(text_tokens[counter + i])\n","    else:\n","      print(\"text corpus length: \" + str(len(text_tokens)))\n","      print(\"current concept counter: \" + str(counter + i))\n","      continue;\n","\n","  same_tokens = []\n","  #compare collected tokens with concept tokens\n","  for item in possible_concept_parts:\n","    if item in concept:\n","      same_tokens.append(item)\n","\n","  #check if all text tokens are parts of the concept\n","  if len(same_tokens) == concept_size:\n","    #add indices of the found items to the concept list\n","    new_items = []\n","    for i in range(counter, counter + concept_size):\n","      new_items.append(i)\n","    if new_items not in concept_list:\n","      concept_list.append(new_items)   \n","\n","\n","def find_concepts(text, ivd_dict):\n","  \"\"\"\n","  find NEs in the given text and save them\n","  in the list  \n","  \"\"\"\n","  text_tokens = lemmatize_text(text)   \n","  #a list of indices of words that are NEs\n","  concepts = []\n","  #NE in dictionary\n","  for item in ivd_dict:  \n","    concept_tokens = ivd_dict[item]    \n","    #helper to get a token index\n","    counter = 0\n","    for token in text_tokens:\n","      #first token of the concept found\n","      if concept_tokens[0].lower() == token.lower():        \n","        if(len(concept_tokens) == 1):\n","          #concept consists of 1 word\n","          handle_simple_concept(token, concepts, counter)   \n","        else:\n","           #concept consists of several words\n","          handle_complex_concept(text_tokens, counter, concept_tokens, concepts)          \n","      counter += 1  \n","  return concepts\n","\n","def find_concept_in_text(word_list, concept_tokens)-> str:\n","  \"\"\"\n","  word_list: tokenized text\n","  concept_tokens: positions of concept tokens in text\n","  determine the whole concept string in the text\n","  \"\"\"\n","  found = \"\"\n","  for index in concept_tokens:\n","    word = word_list[index]    \n","    found += word\n","    found += \" \" \n","  return found.strip()\n","\n","def get_string_start_end(text: str, substring: str): \n","  \"\"\"\n","  text: a text that contains a substring\n","  substring: substring in the text\n","  find all occurences of the substring in the text.\n","  return a list of tuples: the position of the first and the last character of \n","  the given substring in the text\n","  \"\"\" \n","  all_occurences = []\n","  first_pos = 0\n","  while True:\n","    first_pos = text.find(substring, first_pos)\n","    last_pos = first_pos + len(substring)-1\n","    if first_pos == -1:\n","       break; \n","    result = (first_pos, last_pos)\n","    all_occurences.append(result)\n","    first_pos += len(substring)\n","  return all_occurences\n","\n","def create_substring_info(start:int, end:int, concept:str):\n","  \"\"\"\n","  start: start position of the concept in the text\n","  end: end position of the concept in the text\n","  concept: current concept\n","  create information for\n","  each found in the text concept.\n","  This information is saved in the list of\n","  dictionaries, a dictionary per concept  \n","  \"\"\"\n","  substring_info = {}\n","  value = {\"start\":start, \n","           \"end\":end,\n","           \"text\":substring,\n","           \"labels\":[\"MedTech\"]}\n","  substring_info[\"value\"] = value\n","  substring_info[\"id\"] = str(substring_info_id)\n","  substring_info[\"from_name\"]=\"label\"\n","  substring_info[\"to_name\"]=\"text\"\n","  substring_info[\"type\"]=\"labels\"\n","  substring_info[\"origin\"]=\"manual\"\n","  return substring_info\n","\n","def create_annotation_info(result):\n","  \"\"\"\n","  create a part of concept information \n","  for the key \"annotations\"\n","  \"\"\"\n","  annotations = []\n","  annot_info = {\n","        \"id\":annotation_id,\n","        \"completed_by\": 1,\n","        \"result\":result,\n","        \"was_cancelled\": False,\n","        \"ground_truth\": False,\n","        \"created_at\":timestamp,\n","        \"updated_at\":timestamp,\n","        \"lead_time\": 4.362,\n","        \"prediction\": {},\n","        \"result_count\": 0,\n","        \"task\": task_id,\n","        \"parent_prediction\":None,\n","        \"parent_annotation\":None\n","    } \n","  annotations.append(annot_info)\n","  return annotations \n","\n","def create_concept_info(annotations, line, file_name):\n","  \"\"\"\n","  create a dictionary with all needed information about\n","  the concept\n","  \"\"\"  \n","  concept_info={\n","      \"id\":entry_id,\n","      \"annotations\":annotations,\n","      \"file_upload\":file_name,\n","      \"drafts\":[],\n","      \"predictions\":[],\n","      \"data\":{\n","          \"text\": line.strip()\n","          },\n","      \"meta\": {},\n","      \"created_at\":timestamp,\n","      \"updated_at\":timestamp,\n","      \"project\":project\n","    }\n","  return concept_info\n","\n","def write_json(json_info, filename, path):\n","  \"\"\"\n","  Write the given text into the file with the given name and\n","  save it in the given directory\n","  \"\"\"\n","  json_name = filename.replace(\"txt\", \"json\")\n","  file_location = path + json_name\n","  with open(file_location, 'w') as f:\n","    json.dump(json_info, f)  \n","\n","#file to label\n","unlabelled_text = \"/content/drive/MyDrive/ExtractedPubMedArticles/29311296.txt\"\n","file_with_Nes = \"/content/drive/MyDrive/TextcorpusCreation/extracted_nes.txt\"\n","path_to_ls_json = \"/content/drive/MyDrive/TextcorpusCreation/ProcessedTexts/\"\n","ivd_dict = read_data(file_with_Nes)\n","lemmatizer = WordNetLemmatizer()\n","\n","#json creation helper\n","substring_info_id = 1000000000\n","entry_id = 1\n","annotation_id = 1\n","task_id = 1\n","project = 1\n","timestamp=\"2022-05-20T15:14:20.100993Z\"\n","ls_json_list=[]\n","\n","#startpoint\n","with open(unlabelled_text) as f:\n","  all_concepts = []  \n","  lines = f.readlines()\n","  whole_text = ' '.join(lines)\n","  for line in lines:    \n","    result = find_concepts(line, ivd_dict)     \n","    all_concepts.append(result) \n","  \n","  line_counter = 0\n","  for line in lines:\n","    #help variable to control dublicated occurences\n","    concept_occurences = []\n","    print(\"----------\")    \n","    current_concepts = all_concepts[line_counter]\n","    print(\"concepts in line:\")\n","    print(current_concepts)   \n","    \n","    word_list = nltk.word_tokenize(line) \n","    result = []   \n","    for concept in current_concepts:\n","      substring = find_concept_in_text(word_list, concept)\n","      print(\"concept substring\")\n","      print(substring)\n","      #get start and end position of each concept occurence in text\n","      all_occurences = get_string_start_end(line, substring)\n","      for item in all_occurences:        \n","        if item not in concept_occurences:\n","          concept_occurences.append(item)\n","          print(\"added concept occurence item\")\n","          print(item)\n","\n","          #create Label Studio json           \n","          #1. key \"result\" in json\n","          if len(substring) > 0:    \n","            start = item[0] \n","            #add 1 because of LS functionality\n","            end = item[1] + 1  \n","            concept_info = create_substring_info(start, end, substring)  \n","            result.append(concept_info)\n","            substring_info_id +=1\n","    # 2.key \"annotations\"in json   \n","    annotations = create_annotation_info(result)\n","    annotation_id += 1\n","    task_id +=1\n","\n","    #3.create info about whole text    \n","    filename = unlabelled_text.split(\"/\")[-1]\n","    concept_info = create_concept_info(annotations, line, filename)\n","    entry_id += 1\n","    project += 1\n","    line_counter += 1       \n","    ls_json_list.append(concept_info) \n","\n","  #convert result into json  \n","  endresult = json.dumps(ls_json_list)\n","  print(endresult)\n","  #write result in a file\n","  write_json(ls_json_list, filename, path_to_ls_json)\n","    "],"metadata":{"id":"8KQDpnlzBf8a","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1657702636239,"user_tz":-120,"elapsed":1921,"user":{"displayName":"Victoria Dimanova","userId":"07172063658264964881"}},"outputId":"032c4c56-7cce-4e44-f0cb-b4ff97873d3e"},"execution_count":12,"outputs":[{"output_type":"stream","name":"stderr","text":["[nltk_data] Downloading package wordnet to /root/nltk_data...\n","[nltk_data]   Package wordnet is already up-to-date!\n","[nltk_data] Downloading package punkt to /root/nltk_data...\n","[nltk_data]   Package punkt is already up-to-date!\n","[nltk_data] Downloading package omw-1.4 to /root/nltk_data...\n","[nltk_data]   Package omw-1.4 is already up-to-date!\n"]},{"output_type":"stream","name":"stdout","text":["text corpus length: 187\n","current concept counter: 187\n","----------\n","concepts in line:\n","[]\n","----------\n","concepts in line:\n","[]\n","----------\n","concepts in line:\n","[[109]]\n","concept substring\n","HPLC\n","added concept occurence item\n","(593, 596)\n","----------\n","concepts in line:\n","[]\n","----------\n","concepts in line:\n","[[19], [23]]\n","concept substring\n","assay\n","added concept occurence item\n","(110, 114)\n","concept substring\n","Assay\n","added concept occurence item\n","(132, 136)\n","----------\n","concepts in line:\n","[[67], [66, 67], [77], [98], [137, 138], [70], [118], [138], [57], [168], [63], [101]]\n","concept substring\n","plate\n","added concept occurence item\n","(384, 388)\n","concept substring\n","12-well plate\n","added concept occurence item\n","(376, 388)\n","concept substring\n","DMSO\n","added concept occurence item\n","(442, 445)\n","added concept occurence item\n","(540, 543)\n","concept substring\n","DMSO\n","concept substring\n","culture medium\n","added concept occurence item\n","(715, 728)\n","concept substring\n","medium\n","added concept occurence item\n","(395, 400)\n","added concept occurence item\n","(633, 638)\n","added concept occurence item\n","(723, 728)\n","concept substring\n","medium\n","concept substring\n","medium\n","concept substring\n","DMEM\n","added concept occurence item\n","(340, 343)\n","concept substring\n","buffer\n","added concept occurence item\n","(889, 894)\n","concept substring\n","well\n","added concept occurence item\n","(366, 369)\n","added concept occurence item\n","(379, 382)\n","added concept occurence item\n","(551, 554)\n","concept substring\n","wells\n","added concept occurence item\n","(551, 555)\n","----------\n","concepts in line:\n","[[166], [165], [255], [8, 9], [9], [94]]\n","concept substring\n","kit\n","added concept occurence item\n","(918, 920)\n","concept substring\n","ELISA\n","added concept occurence item\n","(912, 916)\n","added concept occurence item\n","(1419, 1423)\n","concept substring\n","ELISA\n","concept substring\n","culture medium\n","added concept occurence item\n","(48, 61)\n","concept substring\n","medium\n","added concept occurence item\n","(56, 61)\n","concept substring\n","buffer\n","added concept occurence item\n","(533, 538)\n","----------\n","concepts in line:\n","[[10], [40], [167], [141]]\n","concept substring\n","kit\n","added concept occurence item\n","(57, 59)\n","concept substring\n","Kit\n","added concept occurence item\n","(236, 238)\n","concept substring\n","CT\n","added concept occurence item\n","(943, 944)\n","concept substring\n","primer\n","added concept occurence item\n","(792, 797)\n","----------\n","concepts in line:\n","[[18]]\n","concept substring\n","reagents\n","added concept occurence item\n","(107, 114)\n","----------\n","concepts in line:\n","[[36], [49], [7], [48]]\n","concept substring\n","slide\n","added concept occurence item\n","(221, 225)\n","concept substring\n","software\n","added concept occurence item\n","(298, 305)\n","concept substring\n","antibody\n","added concept occurence item\n","(43, 50)\n","concept substring\n","ImageJ\n","added concept occurence item\n","(291, 296)\n","----------\n","concepts in line:\n","[[98], [60]]\n","concept substring\n","reagents\n","added concept occurence item\n","(647, 654)\n","concept substring\n","assay\n","added concept occurence item\n","(371, 375)\n","----------\n","concepts in line:\n","[[31], [30], [12], [11, 12]]\n","concept substring\n","Kit\n","added concept occurence item\n","(224, 226)\n","concept substring\n","ELISA\n","added concept occurence item\n","(218, 222)\n","concept substring\n","plates\n","added concept occurence item\n","(81, 86)\n","concept substring\n","12-well plates\n","added concept occurence item\n","(73, 86)\n","----------\n","concepts in line:\n","[[16], [29], [27]]\n","concept substring\n","kit\n","added concept occurence item\n","(97, 99)\n","concept substring\n","spectrophotometer\n","added concept occurence item\n","(160, 176)\n","concept substring\n","NanoDrop\n","added concept occurence item\n","(143, 150)\n","----------\n","concepts in line:\n","[]\n","[{\"id\": 1, \"annotations\": [{\"id\": 1, \"completed_by\": 1, \"result\": [], \"was_cancelled\": false, \"ground_truth\": false, \"created_at\": \"2022-05-20T15:14:20.100993Z\", \"updated_at\": \"2022-05-20T15:14:20.100993Z\", \"lead_time\": 4.362, \"prediction\": {}, \"result_count\": 0, \"task\": 1, \"parent_prediction\": null, \"parent_annotation\": null}], \"file_upload\": \"29311296.txt\", \"drafts\": [], \"predictions\": [], \"data\": {\"text\": \"The in vivo study was designed as a double-blinded, placebo-controlled, randomized intervention trial in which ibuprofen or placebo was administered to subjects for 2 wk before and 30 d after a single exercise session. Staff not involved in the project prepared and distributed the medication in boxes weekly. Study personnel and participants were blinded to treatment, and all later analyses were performed blinded to the treatment type, participant, and time point. The study was part of a broader investigation also focusing on muscle biopsies, collected on days 0, 2, 7, and 30 postexercise, a subset of which is described elsewhere.\"}, \"meta\": {}, \"created_at\": \"2022-05-20T15:14:20.100993Z\", \"updated_at\": \"2022-05-20T15:14:20.100993Z\", \"project\": 1}, {\"id\": 2, \"annotations\": [{\"id\": 2, \"completed_by\": 1, \"result\": [], \"was_cancelled\": false, \"ground_truth\": false, \"created_at\": \"2022-05-20T15:14:20.100993Z\", \"updated_at\": \"2022-05-20T15:14:20.100993Z\", \"lead_time\": 4.362, \"prediction\": {}, \"result_count\": 0, \"task\": 2, \"parent_prediction\": null, \"parent_annotation\": null}], \"file_upload\": \"29311296.txt\", \"drafts\": [], \"predictions\": [], \"data\": {\"text\": \"The study protocol was in compliance with the Helsinki Declaration, was approved by the Regional Scientific Ethical Committees of Copenhagen in Denmark (Ref: HD-2008-074), and was registered at ClinicalTrials.gov (no. NCT00832663). The study recruited 31 healthy white men, age 1835 y. Subjects were included after an interview, a questionnaire assessing physical activity status, and the results of a screening blood sample. Exclusion criteria included body mass index above 30, knee injuries, peptic ulcers, signs of liver or kidney dysfunction, and participation in regular physical activity (especially strength training) apart from cycling as a means of transport. All individuals provided written informed consent to participate in the study. Subsequently, the subjects were assigned to either a placebo (17 subjects) or ibuprofen (14 subjects) group; the groups were matched for age, height, and weight.\"}, \"meta\": {}, \"created_at\": \"2022-05-20T15:14:20.100993Z\", \"updated_at\": \"2022-05-20T15:14:20.100993Z\", \"project\": 2}, {\"id\": 3, \"annotations\": [{\"id\": 3, \"completed_by\": 1, \"result\": [{\"value\": {\"start\": 593, \"end\": 597, \"text\": \"HPLC\", \"labels\": [\"MedTech\"]}, \"id\": \"1000000000\", \"from_name\": \"label\", \"to_name\": \"text\", \"type\": \"labels\", \"origin\": \"manual\"}], \"was_cancelled\": false, \"ground_truth\": false, \"created_at\": \"2022-05-20T15:14:20.100993Z\", \"updated_at\": \"2022-05-20T15:14:20.100993Z\", \"lead_time\": 4.362, \"prediction\": {}, \"result_count\": 0, \"task\": 3, \"parent_prediction\": null, \"parent_annotation\": null}], \"file_upload\": \"29311296.txt\", \"drafts\": [], \"predictions\": [], \"data\": {\"text\": \"One group of subjects received ibuprofen, 2  600 mg/d, (Ibumetin; Nycomed Denmark Aps) for a period of 6 wk, from 14 d before to 4 wk after the electrical stimulation exercise. Ibuprofen was detected only in participants to whom ibuprofen was distributed and only after administration began. The second group received placebo pills (which were visually indistinguishable from the ibuprofen pills) over the same period. Subjects received the medication in Medidos No. 1 boxes (KiBodan A/S), which were refilled every week. To verify compliance, ibuprofen levels in the blood were determined by HPLC at every blood-sampling time point (see below). Additionally, to monitor liver and kidney function, blood samples were analyzed for creatine, C-reactive protein, alkaline phosphates, and total cholesterol during the study. No subjects reported any adverse signs of taking the medication, nor did any blood parameters indicate or suggest adverse effects.\"}, \"meta\": {}, \"created_at\": \"2022-05-20T15:14:20.100993Z\", \"updated_at\": \"2022-05-20T15:14:20.100993Z\", \"project\": 3}, {\"id\": 4, \"annotations\": [{\"id\": 4, \"completed_by\": 1, \"result\": [], \"was_cancelled\": false, \"ground_truth\": false, \"created_at\": \"2022-05-20T15:14:20.100993Z\", \"updated_at\": \"2022-05-20T15:14:20.100993Z\", \"lead_time\": 4.362, \"prediction\": {}, \"result_count\": 0, \"task\": 4, \"parent_prediction\": null, \"parent_annotation\": null}], \"file_upload\": \"29311296.txt\", \"drafts\": [], \"predictions\": [], \"data\": {\"text\": \"A 40-mL blood sample was collected from the antecubital vein of the nondominant arm when subjects arrived at the laboratory for screening. Samples were also taken at 3 and 2 wk before exercise (days 21 and 14), on the day of exercise before the exercise (day 0), after exercise (+2 h; day 0.1), and subsequently at 2, 4, 7, and 30 d after exercise. Hence, ibuprofen and placebo administration was ongoing for samples drawn on day 0, 0+2 h, 2, 4, 7, and 30. In the present study, we focused on samples drawn on day 0 after 14 d of supplementation. Plasma samples were stored at 80 \\u00b0C until being assayed. The samples were analyzed for LH, FSH, testosterone, 17-estradiol, AMH, inhibin B, and SHBG, as previously described in Aksglaede et al., and ibuprofen as previously described in Farrar et al.. Of note, the LH measurements of one subject in the placebo group were aberrant and were discarded, thus decreasing the n value from 17 to 16.\"}, \"meta\": {}, \"created_at\": \"2022-05-20T15:14:20.100993Z\", \"updated_at\": \"2022-05-20T15:14:20.100993Z\", \"project\": 4}, {\"id\": 5, \"annotations\": [{\"id\": 5, \"completed_by\": 1, \"result\": [{\"value\": {\"start\": 110, \"end\": 115, \"text\": \"assay\", \"labels\": [\"MedTech\"]}, \"id\": \"1000000001\", \"from_name\": \"label\", \"to_name\": \"text\", \"type\": \"labels\", \"origin\": \"manual\"}, {\"value\": {\"start\": 132, \"end\": 137, \"text\": \"Assay\", \"labels\": [\"MedTech\"]}, \"id\": \"1000000002\", \"from_name\": \"label\", \"to_name\": \"text\", \"type\": \"labels\", \"origin\": \"manual\"}], \"was_cancelled\": false, \"ground_truth\": false, \"created_at\": \"2022-05-20T15:14:20.100993Z\", \"updated_at\": \"2022-05-20T15:14:20.100993Z\", \"lead_time\": 4.362, \"prediction\": {}, \"result_count\": 0, \"task\": 5, \"parent_prediction\": null, \"parent_annotation\": null}], \"file_upload\": \"29311296.txt\", \"drafts\": [], \"predictions\": [], \"data\": {\"text\": \"To determine the direct effect of ibuprofen on adult human testis physiology, we used a validated organ model assay (Testis Explant Assay). Testes were obtained from prostate cancer patients who had not undergone any antihormonal treatment or from multiorgan donors (average age 46.9  4.4 y). The protocol was approved by the local ethics committee (Agence de la Biom\\u00a9dine; authorization no. PFS09-015), and informed consent was obtained from all donors or their next of kin.\"}, \"meta\": {}, \"created_at\": \"2022-05-20T15:14:20.100993Z\", \"updated_at\": \"2022-05-20T15:14:20.100993Z\", \"project\": 5}, {\"id\": 6, \"annotations\": [{\"id\": 6, \"completed_by\": 1, \"result\": [{\"value\": {\"start\": 384, \"end\": 389, \"text\": \"plate\", \"labels\": [\"MedTech\"]}, \"id\": \"1000000003\", \"from_name\": \"label\", \"to_name\": \"text\", \"type\": \"labels\", \"origin\": \"manual\"}, {\"value\": {\"start\": 376, \"end\": 389, \"text\": \"12-well plate\", \"labels\": [\"MedTech\"]}, \"id\": \"1000000004\", \"from_name\": \"label\", \"to_name\": \"text\", \"type\": \"labels\", \"origin\": \"manual\"}, {\"value\": {\"start\": 442, \"end\": 446, \"text\": \"DMSO\", \"labels\": [\"MedTech\"]}, \"id\": \"1000000005\", \"from_name\": \"label\", \"to_name\": \"text\", \"type\": \"labels\", \"origin\": \"manual\"}, {\"value\": {\"start\": 540, \"end\": 544, \"text\": \"DMSO\", \"labels\": [\"MedTech\"]}, \"id\": \"1000000006\", \"from_name\": \"label\", \"to_name\": \"text\", \"type\": \"labels\", \"origin\": \"manual\"}, {\"value\": {\"start\": 715, \"end\": 729, \"text\": \"culture medium\", \"labels\": [\"MedTech\"]}, \"id\": \"1000000007\", \"from_name\": \"label\", \"to_name\": \"text\", \"type\": \"labels\", \"origin\": \"manual\"}, {\"value\": {\"start\": 395, \"end\": 401, \"text\": \"medium\", \"labels\": [\"MedTech\"]}, \"id\": \"1000000008\", \"from_name\": \"label\", \"to_name\": \"text\", \"type\": \"labels\", \"origin\": \"manual\"}, {\"value\": {\"start\": 633, \"end\": 639, \"text\": \"medium\", \"labels\": [\"MedTech\"]}, \"id\": \"1000000009\", \"from_name\": \"label\", \"to_name\": \"text\", \"type\": \"labels\", \"origin\": \"manual\"}, {\"value\": {\"start\": 723, \"end\": 729, \"text\": \"medium\", \"labels\": [\"MedTech\"]}, \"id\": \"1000000010\", \"from_name\": \"label\", \"to_name\": \"text\", \"type\": \"labels\", \"origin\": \"manual\"}, {\"value\": {\"start\": 340, \"end\": 344, \"text\": \"DMEM\", \"labels\": [\"MedTech\"]}, \"id\": \"1000000011\", \"from_name\": \"label\", \"to_name\": \"text\", \"type\": \"labels\", \"origin\": \"manual\"}, {\"value\": {\"start\": 889, \"end\": 895, \"text\": \"buffer\", \"labels\": [\"MedTech\"]}, \"id\": \"1000000012\", \"from_name\": \"label\", \"to_name\": \"text\", \"type\": \"labels\", \"origin\": \"manual\"}, {\"value\": {\"start\": 366, \"end\": 370, \"text\": \"well\", \"labels\": [\"MedTech\"]}, \"id\": \"1000000013\", \"from_name\": \"label\", \"to_name\": \"text\", \"type\": \"labels\", \"origin\": \"manual\"}, {\"value\": {\"start\": 379, \"end\": 383, \"text\": \"well\", \"labels\": [\"MedTech\"]}, \"id\": \"1000000014\", \"from_name\": \"label\", \"to_name\": \"text\", \"type\": \"labels\", \"origin\": \"manual\"}, {\"value\": {\"start\": 551, \"end\": 555, \"text\": \"well\", \"labels\": [\"MedTech\"]}, \"id\": \"1000000015\", \"from_name\": \"label\", \"to_name\": \"text\", \"type\": \"labels\", \"origin\": \"manual\"}, {\"value\": {\"start\": 551, \"end\": 556, \"text\": \"wells\", \"labels\": [\"MedTech\"]}, \"id\": \"1000000016\", \"from_name\": \"label\", \"to_name\": \"text\", \"type\": \"labels\", \"origin\": \"manual\"}], \"was_cancelled\": false, \"ground_truth\": false, \"created_at\": \"2022-05-20T15:14:20.100993Z\", \"updated_at\": \"2022-05-20T15:14:20.100993Z\", \"lead_time\": 4.362, \"prediction\": {}, \"result_count\": 0, \"task\": 6, \"parent_prediction\": null, \"parent_annotation\": null}], \"file_upload\": \"29311296.txt\", \"drafts\": [], \"predictions\": [], \"data\": {\"text\": \"After collection of each testis, the organ was placed at 4 \\u00b0C and rapidly processed for experimentation. Observation by transillumination allowed us to discard the rare testes not displaying spermatogenesis. Four 3-mm3 testis explants were placed onto a polyethylene terephthalate insert (Falcon Labware) at the interface of air in 1 mL of DMEM (Sigma-Aldrich) in a well of a 12-well plate. The medium for exposure experiments contained 0.1% DMSO as a control and ibuprofen at different concentrations (purity >99%) (Sigma-Aldrich) in 0.1% DMSO. Four wells were analyzed for each condition. Exposures lasted 24 or 48 h, with a total medium change at 24 h, in a humidified atmosphere containing 5% CO2 at 34 \\u00b0C. The culture medium was then collected and stored at 80 \\u00b0C. On the day of collection, three testis explants for each culture condition were randomly selected and fixed in neutral buffer, 4% formaldehyde, or Bouins fixative for 2 h at 4 \\u00b0C, embedded in paraffin, sliced into 5.0-\\u00b5m-thick sections, and stored at 4 \\u00b0C until immunostaining.\"}, \"meta\": {}, \"created_at\": \"2022-05-20T15:14:20.100993Z\", \"updated_at\": \"2022-05-20T15:14:20.100993Z\", \"project\": 6}, {\"id\": 7, \"annotations\": [{\"id\": 7, \"completed_by\": 1, \"result\": [{\"value\": {\"start\": 918, \"end\": 921, \"text\": \"kit\", \"labels\": [\"MedTech\"]}, \"id\": \"1000000017\", \"from_name\": \"label\", \"to_name\": \"text\", \"type\": \"labels\", \"origin\": \"manual\"}, {\"value\": {\"start\": 912, \"end\": 917, \"text\": \"ELISA\", \"labels\": [\"MedTech\"]}, \"id\": \"1000000018\", \"from_name\": \"label\", \"to_name\": \"text\", \"type\": \"labels\", \"origin\": \"manual\"}, {\"value\": {\"start\": 1419, \"end\": 1424, \"text\": \"ELISA\", \"labels\": [\"MedTech\"]}, \"id\": \"1000000019\", \"from_name\": \"label\", \"to_name\": \"text\", \"type\": \"labels\", \"origin\": \"manual\"}, {\"value\": {\"start\": 48, \"end\": 62, \"text\": \"culture medium\", \"labels\": [\"MedTech\"]}, \"id\": \"1000000020\", \"from_name\": \"label\", \"to_name\": \"text\", \"type\": \"labels\", \"origin\": \"manual\"}, {\"value\": {\"start\": 56, \"end\": 62, \"text\": \"medium\", \"labels\": [\"MedTech\"]}, \"id\": \"1000000021\", \"from_name\": \"label\", \"to_name\": \"text\", \"type\": \"labels\", \"origin\": \"manual\"}, {\"value\": {\"start\": 533, \"end\": 539, \"text\": \"buffer\", \"labels\": [\"MedTech\"]}, \"id\": \"1000000022\", \"from_name\": \"label\", \"to_name\": \"text\", \"type\": \"labels\", \"origin\": \"manual\"}], \"was_cancelled\": false, \"ground_truth\": false, \"created_at\": \"2022-05-20T15:14:20.100993Z\", \"updated_at\": \"2022-05-20T15:14:20.100993Z\", \"lead_time\": 4.362, \"prediction\": {}, \"result_count\": 0, \"task\": 7, \"parent_prediction\": null, \"parent_annotation\": null}], \"file_upload\": \"29311296.txt\", \"drafts\": [], \"predictions\": [], \"data\": {\"text\": \"Hormone levels were assayed in duplicate in the culture medium. Testosterone levels were assayed with a specific RIA (Beckman Coulter). The intra- and interassay coefficients of variation were 8.6 and 11.9%, respectively. Control testis explants produced an average of 10.26  2.50 ng testosterone per milliliter of explant after 24 h of culture and 8.47  2.16 ng testosterone per milliliter of explant after 48 h of culture. INSL3 production was assayed with a commercial RIA (Phoenix France). Each sample was diluted twofold in RIA buffer before RIA. The intra- and interassay coefficients of variation were 15 and 7%, respectively, and the lower limit of detection was 20.17 pg/mL. Control testis explants produced an average of 0.29  0.25 ng INSL3 per milliliter of explant after 24 h of culture and 0.20  0.21 ng INSL3 per milliliter of explant after 48 h of culture. Inhibin B was assayed with a commercial ELISA kit (Beckman Coulter) according to the manufacturers instructions. Each sample was diluted twofold in sample diluent solution before reactions. The intra- and interassay coefficients of variation for serum samples were 5.6 and 7.6%, respectively. Control testis explants produced an average of 570.43  78.50 ng inhibin B per milliliter of explant after 24 h of culture and 529.44  79.17 ng inhibin B per milliliter of explant after 48 h of culture. PGD2 and prostaglandin E2 (PGE2) were assayed by an ELISA method (Cayman Chemical Company), as was AMH (Immunotech).\"}, \"meta\": {}, \"created_at\": \"2022-05-20T15:14:20.100993Z\", \"updated_at\": \"2022-05-20T15:14:20.100993Z\", \"project\": 7}, {\"id\": 8, \"annotations\": [{\"id\": 8, \"completed_by\": 1, \"result\": [{\"value\": {\"start\": 57, \"end\": 60, \"text\": \"kit\", \"labels\": [\"MedTech\"]}, \"id\": \"1000000023\", \"from_name\": \"label\", \"to_name\": \"text\", \"type\": \"labels\", \"origin\": \"manual\"}, {\"value\": {\"start\": 236, \"end\": 239, \"text\": \"Kit\", \"labels\": [\"MedTech\"]}, \"id\": \"1000000024\", \"from_name\": \"label\", \"to_name\": \"text\", \"type\": \"labels\", \"origin\": \"manual\"}, {\"value\": {\"start\": 943, \"end\": 945, \"text\": \"CT\", \"labels\": [\"MedTech\"]}, \"id\": \"1000000025\", \"from_name\": \"label\", \"to_name\": \"text\", \"type\": \"labels\", \"origin\": \"manual\"}, {\"value\": {\"start\": 792, \"end\": 798, \"text\": \"primer\", \"labels\": [\"MedTech\"]}, \"id\": \"1000000026\", \"from_name\": \"label\", \"to_name\": \"text\", \"type\": \"labels\", \"origin\": \"manual\"}], \"was_cancelled\": false, \"ground_truth\": false, \"created_at\": \"2022-05-20T15:14:20.100993Z\", \"updated_at\": \"2022-05-20T15:14:20.100993Z\", \"lead_time\": 4.362, \"prediction\": {}, \"result_count\": 0, \"task\": 8, \"parent_prediction\": null, \"parent_annotation\": null}], \"file_upload\": \"29311296.txt\", \"drafts\": [], \"predictions\": [], \"data\": {\"text\": \"RNA was extracted from testes with the NucleoSpin RNA II kit (Macherey-Nagel) according to the manufacturers instructions and then was precipitated. Each total RNA sample (250 ng) was reverse transcribed with the Iscript cDNA Synthesis Kit (Bio-Rad). Quantitative PCR was performed according to the manufacturers instructions with iTaq Universal SYBR Green Supermix (Bio-Rad) and a 2.5-\\u00b5L cDNA template in a CFX384 Touch Real-Time PCR Detection System (Bio-Rad). The amplification program was as follows: an initial denaturation of 3 min at 95 \\u00b0C; 40 cycles of 10-s denaturation at 95 \\u00b0C; and 30 s at 62 \\u00b0C for annealing and extension. Dissociation curves were produced with the thermal melting profile performed after the last PCR cycle. To avoid amplification of contaminating genomic DNA, primer pairs were selected on two different exons. Bzw1 and GusB mRNA were used as internal controls for normalization. Results were calculated by the CT method as n-fold differences in target gene expression with respect to the reference gene and the calibration sample.\"}, \"meta\": {}, \"created_at\": \"2022-05-20T15:14:20.100993Z\", \"updated_at\": \"2022-05-20T15:14:20.100993Z\", \"project\": 8}, {\"id\": 9, \"annotations\": [{\"id\": 9, \"completed_by\": 1, \"result\": [{\"value\": {\"start\": 107, \"end\": 115, \"text\": \"reagents\", \"labels\": [\"MedTech\"]}, \"id\": \"1000000027\", \"from_name\": \"label\", \"to_name\": \"text\", \"type\": \"labels\", \"origin\": \"manual\"}], \"was_cancelled\": false, \"ground_truth\": false, \"created_at\": \"2022-05-20T15:14:20.100993Z\", \"updated_at\": \"2022-05-20T15:14:20.100993Z\", \"lead_time\": 4.362, \"prediction\": {}, \"result_count\": 0, \"task\": 9, \"parent_prediction\": null, \"parent_annotation\": null}], \"file_upload\": \"29311296.txt\", \"drafts\": [], \"predictions\": [], \"data\": {\"text\": \"To analyze steroidogenesis in the explants, we performed solid-phase extraction (SPE) with C18 cartridges, reagents, and solvents from Solvent Documentation Synthesis. Standard reference steroids were from Sigma. Deuterated internal standards were from Steraloids. Quantification was performed by isotopic dilution. Samples were spiked with 400 pg of internal standards (etiocholanolone-d5, 17a-testosterone-d3, dihydrotestosterone-d3, 19-androstenedione-d3, progesterone-d9, 17a-methyltestosterone-d3, and 17b-estradiol-d3). Samples were applied to a C18 SPE column (2 g stationary phase) previously conditioned with 10 mL methanol and 10 mL water. The column was washed with water (5 mL) and then with cyclohexane (5 mL), and the steroids were eluted with methanol (10 mL). The extracts were dried (N2, 458C), and 400 pg of external standard (norgestrel) was added. Derivatization procedures and measurements in GC-MS/MS in electronization mode were performed as previously described.\"}, \"meta\": {}, \"created_at\": \"2022-05-20T15:14:20.100993Z\", \"updated_at\": \"2022-05-20T15:14:20.100993Z\", \"project\": 9}, {\"id\": 10, \"annotations\": [{\"id\": 10, \"completed_by\": 1, \"result\": [{\"value\": {\"start\": 221, \"end\": 226, \"text\": \"slide\", \"labels\": [\"MedTech\"]}, \"id\": \"1000000028\", \"from_name\": \"label\", \"to_name\": \"text\", \"type\": \"labels\", \"origin\": \"manual\"}, {\"value\": {\"start\": 298, \"end\": 306, \"text\": \"software\", \"labels\": [\"MedTech\"]}, \"id\": \"1000000029\", \"from_name\": \"label\", \"to_name\": \"text\", \"type\": \"labels\", \"origin\": \"manual\"}, {\"value\": {\"start\": 43, \"end\": 51, \"text\": \"antibody\", \"labels\": [\"MedTech\"]}, \"id\": \"1000000030\", \"from_name\": \"label\", \"to_name\": \"text\", \"type\": \"labels\", \"origin\": \"manual\"}, {\"value\": {\"start\": 291, \"end\": 297, \"text\": \"ImageJ\", \"labels\": [\"MedTech\"]}, \"id\": \"1000000031\", \"from_name\": \"label\", \"to_name\": \"text\", \"type\": \"labels\", \"origin\": \"manual\"}], \"was_cancelled\": false, \"ground_truth\": false, \"created_at\": \"2022-05-20T15:14:20.100993Z\", \"updated_at\": \"2022-05-20T15:14:20.100993Z\", \"lead_time\": 4.362, \"prediction\": {}, \"result_count\": 0, \"task\": 10, \"parent_prediction\": null, \"parent_annotation\": null}], \"file_upload\": \"29311296.txt\", \"drafts\": [], \"predictions\": [], \"data\": {\"text\": \"Cells were labeled with the primary rabbit antibody directed against cleaved caspase-3 (1/100; Cell Signaling) to enable detection of cells undergoing apoptosis in the explants. Slides were then scanned with a NanoZoomer slide scanner (Hamamatsu Photonics). Caspase-3+ cells were counted by ImageJ software, and the results were expressed as percentages of the control values.\"}, \"meta\": {}, \"created_at\": \"2022-05-20T15:14:20.100993Z\", \"updated_at\": \"2022-05-20T15:14:20.100993Z\", \"project\": 10}, {\"id\": 11, \"annotations\": [{\"id\": 11, \"completed_by\": 1, \"result\": [{\"value\": {\"start\": 647, \"end\": 655, \"text\": \"reagents\", \"labels\": [\"MedTech\"]}, \"id\": \"1000000032\", \"from_name\": \"label\", \"to_name\": \"text\", \"type\": \"labels\", \"origin\": \"manual\"}, {\"value\": {\"start\": 371, \"end\": 376, \"text\": \"assay\", \"labels\": [\"MedTech\"]}, \"id\": \"1000000033\", \"from_name\": \"label\", \"to_name\": \"text\", \"type\": \"labels\", \"origin\": \"manual\"}], \"was_cancelled\": false, \"ground_truth\": false, \"created_at\": \"2022-05-20T15:14:20.100993Z\", \"updated_at\": \"2022-05-20T15:14:20.100993Z\", \"lead_time\": 4.362, \"prediction\": {}, \"result_count\": 0, \"task\": 11, \"parent_prediction\": null, \"parent_annotation\": null}], \"file_upload\": \"29311296.txt\", \"drafts\": [], \"predictions\": [], \"data\": {\"text\": \"The NCI-H295R human adrenocortical carcinoma cell line was obtained from ATCC (CRL-2128), and experiments aiming at completing the ex vivo experiments using human adult testis explants were conducted in accordance with OECD guidelines. Steroids were analyzed after exposure to ibuprofen, abiraterone, and PGD2 (Sigma-Aldrich). Toxicity was evaluated with the Alamar Blue assay (Sigma-Aldrich). Deuterated steroid analogs were obtained for analysis from CDN Isotopes and Toronto Research Chemicals, and derivatization quality-control standards (7b-17-estradiol-17-acetate), instrument control standards (estrone-3-methyl ether), and derivatization reagents (N-trimethylsilylimidazole, N-methyl-N-trimethylsilyl-trifluoroacetamide, and 1,4-dithioerythritol) were obtained from Sigma-Aldrich. Steroid extraction and the subsequent quantification of steroid hormones were performed according to Holm et al..\"}, \"meta\": {}, \"created_at\": \"2022-05-20T15:14:20.100993Z\", \"updated_at\": \"2022-05-20T15:14:20.100993Z\", \"project\": 11}, {\"id\": 12, \"annotations\": [{\"id\": 12, \"completed_by\": 1, \"result\": [{\"value\": {\"start\": 224, \"end\": 227, \"text\": \"Kit\", \"labels\": [\"MedTech\"]}, \"id\": \"1000000034\", \"from_name\": \"label\", \"to_name\": \"text\", \"type\": \"labels\", \"origin\": \"manual\"}, {\"value\": {\"start\": 218, \"end\": 223, \"text\": \"ELISA\", \"labels\": [\"MedTech\"]}, \"id\": \"1000000035\", \"from_name\": \"label\", \"to_name\": \"text\", \"type\": \"labels\", \"origin\": \"manual\"}, {\"value\": {\"start\": 81, \"end\": 87, \"text\": \"plates\", \"labels\": [\"MedTech\"]}, \"id\": \"1000000036\", \"from_name\": \"label\", \"to_name\": \"text\", \"type\": \"labels\", \"origin\": \"manual\"}, {\"value\": {\"start\": 73, \"end\": 87, \"text\": \"12-well plates\", \"labels\": [\"MedTech\"]}, \"id\": \"1000000037\", \"from_name\": \"label\", \"to_name\": \"text\", \"type\": \"labels\", \"origin\": \"manual\"}], \"was_cancelled\": false, \"ground_truth\": false, \"created_at\": \"2022-05-20T15:14:20.100993Z\", \"updated_at\": \"2022-05-20T15:14:20.100993Z\", \"lead_time\": 4.362, \"prediction\": {}, \"result_count\": 0, \"task\": 12, \"parent_prediction\": null, \"parent_annotation\": null}], \"file_upload\": \"29311296.txt\", \"drafts\": [], \"predictions\": [], \"data\": {\"text\": \"For prostaglandin secretion analysis, the NIC-H295R cells were plated in 12-well plates and were exposed overnight; the global prostaglandin synthesis was evaluated immediately afterward with a Prostaglandin Screening ELISA Kit (Cayman Chemical Company).\"}, \"meta\": {}, \"created_at\": \"2022-05-20T15:14:20.100993Z\", \"updated_at\": \"2022-05-20T15:14:20.100993Z\", \"project\": 12}, {\"id\": 13, \"annotations\": [{\"id\": 13, \"completed_by\": 1, \"result\": [{\"value\": {\"start\": 97, \"end\": 100, \"text\": \"kit\", \"labels\": [\"MedTech\"]}, \"id\": \"1000000038\", \"from_name\": \"label\", \"to_name\": \"text\", \"type\": \"labels\", \"origin\": \"manual\"}, {\"value\": {\"start\": 160, \"end\": 177, \"text\": \"spectrophotometer\", \"labels\": [\"MedTech\"]}, \"id\": \"1000000039\", \"from_name\": \"label\", \"to_name\": \"text\", \"type\": \"labels\", \"origin\": \"manual\"}, {\"value\": {\"start\": 143, \"end\": 151, \"text\": \"NanoDrop\", \"labels\": [\"MedTech\"]}, \"id\": \"1000000040\", \"from_name\": \"label\", \"to_name\": \"text\", \"type\": \"labels\", \"origin\": \"manual\"}], \"was_cancelled\": false, \"ground_truth\": false, \"created_at\": \"2022-05-20T15:14:20.100993Z\", \"updated_at\": \"2022-05-20T15:14:20.100993Z\", \"lead_time\": 4.362, \"prediction\": {}, \"result_count\": 0, \"task\": 13, \"parent_prediction\": null, \"parent_annotation\": null}], \"file_upload\": \"29311296.txt\", \"drafts\": [], \"predictions\": [], \"data\": {\"text\": \"For transcriptional analysis with RT-qPCR, RNA was isolated from cells with the All Prep DNA/RNA kit (Qiagen). RNA integrity was measured on a NanoDrop ND-1000 spectrophotometer (Thermo Scientific), and only 260/280 and 260/230 ratios >2.0 were accepted for further processing. Quantitative PCR was performed as previously described.\"}, \"meta\": {}, \"created_at\": \"2022-05-20T15:14:20.100993Z\", \"updated_at\": \"2022-05-20T15:14:20.100993Z\", \"project\": 13}, {\"id\": 14, \"annotations\": [{\"id\": 14, \"completed_by\": 1, \"result\": [], \"was_cancelled\": false, \"ground_truth\": false, \"created_at\": \"2022-05-20T15:14:20.100993Z\", \"updated_at\": \"2022-05-20T15:14:20.100993Z\", \"lead_time\": 4.362, \"prediction\": {}, \"result_count\": 0, \"task\": 14, \"parent_prediction\": null, \"parent_annotation\": null}], \"file_upload\": \"29311296.txt\", \"drafts\": [], \"predictions\": [], \"data\": {\"text\": \"For the trial, each individuals samples were normalized by division by the mean of the baseline samples drawn before the intervention. Hence, samples from each volunteer were normalized with the individuals own baseline values before the administration. Unpaired Student t tests were used to compare the placebo and ibuprofen groups after 14 and 44 d of administration. For the ex vivo experiments, data were compared using the MannWhitney U test and slopes with P values and Spearman correlation when indicated. For in vitro cell experiments, analysis was performed with one-way ANOVA followed by a post hoc Dunnetts multiple comparison test. All data are expressed as mean  SEM, and differences were considered statistically significant when P  0.05.\"}, \"meta\": {}, \"created_at\": \"2022-05-20T15:14:20.100993Z\", \"updated_at\": \"2022-05-20T15:14:20.100993Z\", \"project\": 14}]\n"]}]},{"cell_type":"markdown","source":["STEP 4:\n","*   To extend the collected extracted NEs, automatically + manually annotated files will be searched for new concepts.\n","*   The extended collection of extracted NEs will be used for further NE identification in the untagged texts\n","\n","\n","\n","\n","\n"],"metadata":{"id":"P6t3Nr1rQEzX"}},{"cell_type":"code","source":["import glob\n","import json\n","import nltk\n","from nltk.stem import WordNetLemmatizer \n","nltk.download('wordnet')\n","nltk.download('punkt')\n","nltk.download('omw-1.4')\n","\n","path_to_labelled_texts = \"/content/drive/MyDrive/TextcorpusCreation/AnnotatedTexts/131-140/\"\n","extracted_nes = \"/content/drive/MyDrive/TextcorpusCreation/extracted_nes.txt\"\n","\n","\n","def read_data(path_to_file: str):\n","  \"\"\"\n","  path_to_file: path to a file with extracted\n","  read extracted named entities and save them in ivd_dict\n","  dictionary for further processing\n","  \"\"\"\n","  with open(path_to_file) as f:\n","    data = f.read().strip()\n","    ivd_dict = json.loads(data)    \n","  return ivd_dict\n","\n","def write_data(data_to_write):\n","  \"\"\"\n","  write extracted concepts from the dictionary\n","  in the file\n","  \"\"\"\n","  with open(extracted_nes, 'w') as f:\n","    f.write(json.dumps(data_to_write))\n","\n","def get_concepts(ls_json):\n","  \"\"\"\n","  get a label studio json-file\n","  and get NEs that are saved in the nested json\n","  \"\"\"\n","  found_concepts = []\n","  current_line = lines[0]\n","  info = json.loads(current_line)\n","  for item in info:\n","    annotations = item[\"annotations\"]\n","    current_annotation = annotations[0]\n","    result = current_annotation[\"result\"]\n","    for item in result:\n","      val = item[\"value\"]\n","      concept = val[\"text\"].strip()\n","      found_concepts.append(concept)\n","  return found_concepts\n","    \n","def lemmatize(concept: str):\n","  \"\"\"\n","  tokenize and lemmatize the given NE concept\n","  and get the result as a tuple(concept, [lemmas])\n","  \"\"\"  \n","  word_list = nltk.word_tokenize(concept)\n","  lemmatized_output = [lemmatizer.lemmatize(w) for w in word_list]\n","  return (concept, lemmatized_output)\n","\n","#manually annotated texts\n","all_files = glob.glob(path_to_labelled_texts + \"*.json\")\n","\n","#look-up\n","extracted_concepts = read_data(extracted_nes)\n","\n","#nltk WordNetLemmatizer \n","lemmatizer = WordNetLemmatizer()\n","\n","#startpoint\n","for file in all_files:  \n","  with open(file) as f:    \n","    lines = f.readlines()\n","    concepts = get_concepts(lines)   \n","    for each in concepts:      \n","      (ne, lemmas) = lemmatize(each.lower())\n","      #save them in dictionary\n","      if ne not in extracted_concepts:\n","        #avoid dublicated values for keys like plate/plates\n","        if lemmas not in extracted_concepts.values():\n","          extracted_concepts[ne] = lemmas\n","          print(extracted_concepts[ne])   \n","#extend look-up with new concepts          \n","write_data(extracted_concepts)\n","\n","print(\"done\")\n"],"metadata":{"id":"ZUz7Gkk2Qjkc","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1657702288696,"user_tz":-120,"elapsed":5313,"user":{"displayName":"Victoria Dimanova","userId":"07172063658264964881"}},"outputId":"b6640770-4eca-4de5-fb33-042a8ec65265"},"execution_count":1,"outputs":[{"output_type":"stream","name":"stderr","text":["[nltk_data] Downloading package wordnet to /root/nltk_data...\n","[nltk_data] Downloading package punkt to /root/nltk_data...\n","[nltk_data]   Unzipping tokenizers/punkt.zip.\n","[nltk_data] Downloading package omw-1.4 to /root/nltk_data...\n"]},{"output_type":"stream","name":"stdout","text":["['annexin', 'v/pi', 'assay']\n","['cytokine', 'cocktail']\n","['abt-199']\n","['flow', 'cytometry']\n","['rppa']\n","['reverse', 'phase', 'protein', 'array']\n","['immunoblots']\n","['li-cor', 'odyssey', 'clx', 'infrared', 'imaging', 'system']\n","['real-time', 'rt-pcr', 'assay']\n","['taqman', 'human', 'apoptosis', 'array']\n","['microfluidic', 'card']\n","['direct-zol', 'miniprep', 'kit']\n","['nanostrings', 'ncounter', 'gex', 'technology']\n","['anti-human', 'cd3-percp']\n","['anti-cd56-apc']\n","['anti-cd16-pecy7']\n","['anti-human', 'cd4-bv421']\n","['anti-human', 'cd107a-bv500']\n","['anti-human', 'cd38-fitc']\n","['anti-human', 'hla-dr-pe']\n","['anti-human', 'nkg2d-bv500']\n","['ghost', 'red', '780']\n","['mouse', 'igg1-bv500']\n","['igg1-apc']\n","['igg2a-pe']\n","['bd', 'facsverse', 'flow', 'cytometer']\n","['ibm', 'spss', 'statistic', '22']\n","['mononuclear', 'cell', 'preparation', 'tube']\n","['all', 'prep', 'dna/rna/mirna', 'universal', 'kit']\n","['truseq', 'rna', 'sample', 'preparation', 'kit']\n","['illumina', 'hiseq', '2000']\n","['of', 'medium']\n","['somalogic', 'somascan', 'assay']\n","['assay', 'panel']\n","['rneasy', 'total', 'rna', 'kit']\n","['illumina', 'hiseq', '2000', 'sequencer']\n","['applied', 'biosystems', '7900', 'real-time', 'pcr', 'system']\n","['dynabeads', 'cd14']\n","['fluorescence-activated', 'cell', 'sorting']\n","['bd', 'lsrfortessa']\n","['mrnaseq', 'gene', 'expression', 'file']\n","['fasta', 'file']\n","['96\\x90well', 'plate']\n","['luc\\x90wee1\\x903\\x90utr', 'reporter', 'vector']\n","['lipofectamine', '3000']\n","['lightswitch', 'luciferase', 'assay', 'reagent\\x84¢']\n","['mir\\x9015', 'precursor']\n","['mir', 'precursor', 'control', 'vector']\n","['nupage', '412', '%', 'bis\\x90tris', 'gel']\n","['odyssey', 'clx', 'imaging', 'equipment']\n","['mtt']\n","['96\\x90well', 'spectramax', 'plate', 'reader']\n","['graphpad', 'prism', '4', 'software']\n","['wee1', 'sirna']\n","['lipofectamine\\x84¢', 'rnaimax']\n","['pre\\x90mir\\x9015', 'precursor']\n","['ambion', 'pre\\x90mir\\x84¢', 'mir', 'precursor']\n","['mir\\x9015b', 'precursor']\n","['taqman', 'copy', 'number', 'assay']\n","['reference', 'assay', 'human', 'rnase', 'p']\n","['custom', 'assay', 'mir\\x9015b']\n","['steponeplus', 'real\\x90time', 'pcr', 'system']\n","['copycaller', '\\x84¢', 'software']\n","['taqman', 'mir', 'assay']\n","['mir', 'primer']\n","['taqman', 'mir', 'reverse', 'transcription', 'kit']\n","['taqman', 'mir', 'assay', 'primer']\n","['tigr', 'mev']\n","['multiple', 'experimental', 'viewer']\n","['mirnaneasy', 'mini', 'kit']\n","['mir', 'microarray', 'assay']\n","['ym\\x90100', 'microcon', 'centrifugal', 'filter']\n","['paraflo', 'microfluidic', 'chip']\n","['mirhuman_16.0']\n","['microcirculation', 'pump']\n","['microfluidic', 'chip']\n","['6', 'sspe', 'buffer']\n","['medium', 'cybernetics']\n","['array\\x90pro', 'image', 'analysis', 'software']\n","['statistical', 'analysis', 'system', '(', 'sa', ',', 'cary', ',', 'north', 'carolina', ')', 'software']\n","['phoenix', 'winnonlin', '6.3']\n","['water', 'xbridge', 'c18', '50', '2.1', 'mm', '(', 'particle', 'size', '5', 'm', ')', 'column']\n","['liquid', 'chromatography']\n","['mass', 'spectrometry']\n","['phenomenex', 'synergi', 'hydro\\x90rp', '75', '2.0', 'mm', '(', 'particle', 'size', '4', 'm', ')', 'column']\n","['sciex', 'api', '5500']\n","['sybr', 'green', 'validated', 'primer']\n","['applied', 'biosystem', '7900ht', 'fast', 'real\\x90time', 'pcr', 'system']\n","['cyp', 'mrna', 'expression']\n","['xenoblis']\n","['thermo', 'tsq', 'quantum', 'ultra', 'mass', 'spectrometer']\n","['thermo', 'scientific', 'hypersil', 'gold', 'c18', '50']\n","['ace', 'c18\\x90ar', '75']\n","['cyp', 'enzyme', 'marker', 'substrate']\n","['nadph']\n","['high', 'capacity', 'cdna', 'revers', 'transferase', 'kit']\n","['cfx96\\x84¢', 'real-time', 'system']\n","['human', 'csf1', 'elisa', 'kit']\n","['dmc00b']\n","['bc', 'assay']\n","['protein', 'assay', 'kit']\n","['signalp4']\n","['phobius']\n","['targetp']\n","['wolf', 'psort']\n","['agilent', 'rna', '6000', 'nano', 'chip']\n","['illumina', 'hiseq', '2000', 'system']\n","['r', 'package', 'oligo']\n","['spss', '19.0', 'computer', 'software']\n","['annexin', 'v-pe/7aad', 'apoptosis', 'detection', 'kit']\n","['bio-rad', 'protein', 'assay', 'reagent']\n","['mouse', 'anti-human', 'ku80']\n","['goat', 'anti-mouse', 'secondary', 'antibody']\n","['las-4000', 'mini', 'system']\n","['lentiviral', 'vector']\n","['1640', 'medium']\n","['m-mlv', 'reverse', 'transcriptase']\n","['sybr-green']\n","['lightcycler', '480', 'real-time', 'pcr', 'system']\n","['ku80']\n","['mrna', 'expression']\n","['rabbit', 'anti-human', 'ku80', 'monoclonal', 'antibody']\n","['dab', 'quanto']\n","['enhanced', 'amv-rt']\n","['dnasei', 'treatment']\n","['gene', 'expression', 'master', 'mix']\n","['24well', 'plate']\n","['pgl', 'cmv', 'renilla']\n","['luciferase', 'assay']\n","['varioskan', 'flash', 'luminometer']\n","['phenol-free', 'medium']\n","['quikchange', 'primer', 'design']\n","['sequalprep\\x84¢', 'long', 'pcr', 'kit']\n","['hot', 'star', 'high', 'fidelity', 'dna', 'polymerase']\n","['geneart']\n","['hi-capacity', 'cdna', 'reverse', 'transcription', 'kit']\n","['power', 'sybr']\n","['abi', '7900ht']\n","['ingenuity', 'pathway', 'analysis', 'software']\n","['agilent', 'human', 'promoter', 'whole-genome', 'chip-on-chip', 'array']\n","['bayesian', 'tool', 'for', 'methylation', 'analysis']\n","['batman']\n","['(', 'mtt', ')', 'assay']\n","['mir-mimic', 'scramble', 'control']\n","['mirna', 'expression', 'assay']\n","['ncounter', 'prep', 'station', 'digital', 'analyzer']\n","['nsolver', 'analysis', 'software']\n","['multiexperiment', 'viewer', '(', 'mev', ')', 'software']\n","['taqman', 'rt-pcr', 'microrna', 'assay']\n","['abi', '7900ht', 'fast', 'real-time', 'pcr', 'system']\n","['sequence', 'detection', 'system']\n","['mirna', 'taqman', 'probe']\n","['rnu6b']\n","['mmlv', 'reverse', 'transcriptase']\n","['oligo', '(', 'dt', ')', '16', 'primer']\n","['abi', 'quantstudio', '6', 'flex', 'real-time', 'pcr', 'system']\n","['quantstudio', 'real-time', 'pcr', 'software']\n","['human', 'ht-12', 'v4', 'expression', 'bead', 'chip']\n","['multiexperiment', 'viewer']\n","['mev4', ',', 'tm4', 'software', 'suite']\n","['david']\n","['mirpath']\n","['diana', ':', 'mirpath']\n","['mirvana', 'isolation', 'kit']\n","['ion', 'pi', 'template', 'ot2', '200', 'kit']\n","['ion', 'proton', 'system']\n","['fastq', 'file']\n","['ion', 'suite', 'software']\n","['mirbase']\n","['hg19', 'genome', 'build']\n","['hiseq2500', 'illumina', 'sequencer']\n","['casava']\n","['fastx_toolkit']\n","['illumina', 'adapter', 'sequence']\n","['ncpro-seq']\n","['bioconductor', 'deseq2', 'package']\n","['kapa', 'library', 'quantification', 'kit']\n","['ampure', 'xp', 'bead']\n","['ampure', 'bead']\n","['agilent', '2100', 'bioanalyzer', 'high', 'sensitivity', 'dna', 'chip']\n","['illumina', 'truseq', 'small', 'rna', 'protocol']\n","['paxgene', 'tube']\n","['paxgene', 'blood', 'mirna', 'kit']\n","['rneasy', 'minelute', 'cleanup', 'kit']\n","['rnase-free', 'dnase', 'kit']\n","['nanodrop', '1000']\n","done\n"]}]}]}