{"cells":[{"cell_type":"markdown","metadata":{"id":"VGSzuJgUYAXd"},"source":["In this notebook various transformer models will be fine-tuned with PubMed Data.\n","\n","---\n","\n","It will be done in several steps:\n","\n","\n","*   Set up environment\n","*   Convert Label Studio data into format accepted by SpaCy\n","*   Fine-tune models\n","*   Compare results\n","\n"]},{"cell_type":"markdown","metadata":{"id":"Ca_QwYtSYrkz"},"source":["STEP 1:Install SpaCy"]},{"cell_type":"code","execution_count":null,"metadata":{"colab":{"base_uri":"https://localhost:8080/","height":1000},"executionInfo":{"elapsed":19031,"status":"ok","timestamp":1669387034920,"user":{"displayName":"Victoria Dimanova","userId":"07172063658264964881"},"user_tz":-60},"id":"wsGT6oURT9mn","outputId":"e3ce9c75-1415-4bb8-dc82-34831ce80b2c"},"outputs":[{"name":"stdout","output_type":"stream","text":["Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/\n","Collecting spacy==3.2.0\n","  Downloading spacy-3.2.0-cp37-cp37m-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (6.0 MB)\n","\u001b[K     |████████████████████████████████| 6.0 MB 11.5 MB/s \n","\u001b[?25hRequirement already satisfied: blis<0.8.0,>=0.4.0 in /usr/local/lib/python3.7/dist-packages (from spacy==3.2.0) (0.7.9)\n","Requirement already satisfied: spacy-loggers<2.0.0,>=1.0.0 in /usr/local/lib/python3.7/dist-packages (from spacy==3.2.0) (1.0.3)\n","Requirement already satisfied: cymem<2.1.0,>=2.0.2 in /usr/local/lib/python3.7/dist-packages (from spacy==3.2.0) (2.0.7)\n","Requirement already satisfied: tqdm<5.0.0,>=4.38.0 in /usr/local/lib/python3.7/dist-packages (from spacy==3.2.0) (4.64.1)\n","Requirement already satisfied: spacy-legacy<3.1.0,>=3.0.8 in /usr/local/lib/python3.7/dist-packages (from spacy==3.2.0) (3.0.10)\n","Collecting thinc<8.1.0,>=8.0.12\n","  Downloading thinc-8.0.17-cp37-cp37m-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (660 kB)\n","\u001b[K     |████████████████████████████████| 660 kB 62.7 MB/s \n","\u001b[?25hCollecting pydantic!=1.8,!=1.8.1,<1.9.0,>=1.7.4\n","  Downloading pydantic-1.8.2-cp37-cp37m-manylinux2014_x86_64.whl (10.1 MB)\n","\u001b[K     |████████████████████████████████| 10.1 MB 28.0 MB/s \n","\u001b[?25hRequirement already satisfied: srsly<3.0.0,>=2.4.1 in /usr/local/lib/python3.7/dist-packages (from spacy==3.2.0) (2.4.5)\n","Requirement already satisfied: catalogue<2.1.0,>=2.0.6 in /usr/local/lib/python3.7/dist-packages (from spacy==3.2.0) (2.0.8)\n","Requirement already satisfied: pathy>=0.3.5 in /usr/local/lib/python3.7/dist-packages (from spacy==3.2.0) (0.8.1)\n","Requirement already satisfied: packaging>=20.0 in /usr/local/lib/python3.7/dist-packages (from spacy==3.2.0) (21.3)\n","Requirement already satisfied: numpy>=1.15.0 in /usr/local/lib/python3.7/dist-packages (from spacy==3.2.0) (1.21.6)\n","Collecting typing-extensions<4.0.0.0,>=3.7.4\n","  Downloading typing_extensions-3.10.0.2-py3-none-any.whl (26 kB)\n","Requirement already satisfied: wasabi<1.1.0,>=0.8.1 in /usr/local/lib/python3.7/dist-packages (from spacy==3.2.0) (0.10.1)\n","Collecting typer<0.5.0,>=0.3.0\n","  Downloading typer-0.4.2-py3-none-any.whl (27 kB)\n","Requirement already satisfied: setuptools in /usr/local/lib/python3.7/dist-packages (from spacy==3.2.0) (57.4.0)\n","Requirement already satisfied: preshed<3.1.0,>=3.0.2 in /usr/local/lib/python3.7/dist-packages (from spacy==3.2.0) (3.0.8)\n","Requirement already satisfied: requests<3.0.0,>=2.13.0 in /usr/local/lib/python3.7/dist-packages (from spacy==3.2.0) (2.23.0)\n","Requirement already satisfied: langcodes<4.0.0,>=3.2.0 in /usr/local/lib/python3.7/dist-packages (from spacy==3.2.0) (3.3.0)\n","Requirement already satisfied: murmurhash<1.1.0,>=0.28.0 in /usr/local/lib/python3.7/dist-packages (from spacy==3.2.0) (1.0.9)\n","Requirement already satisfied: jinja2 in /usr/local/lib/python3.7/dist-packages (from spacy==3.2.0) (2.11.3)\n","Requirement already satisfied: zipp>=0.5 in /usr/local/lib/python3.7/dist-packages (from catalogue<2.1.0,>=2.0.6->spacy==3.2.0) (3.10.0)\n","Requirement already satisfied: pyparsing!=3.0.5,>=2.0.2 in /usr/local/lib/python3.7/dist-packages (from packaging>=20.0->spacy==3.2.0) (3.0.9)\n","Requirement already satisfied: smart-open<6.0.0,>=5.2.1 in /usr/local/lib/python3.7/dist-packages (from pathy>=0.3.5->spacy==3.2.0) (5.2.1)\n","Requirement already satisfied: idna<3,>=2.5 in /usr/local/lib/python3.7/dist-packages (from requests<3.0.0,>=2.13.0->spacy==3.2.0) (2.10)\n","Requirement already satisfied: urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 in /usr/local/lib/python3.7/dist-packages (from requests<3.0.0,>=2.13.0->spacy==3.2.0) (1.24.3)\n","Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.7/dist-packages (from requests<3.0.0,>=2.13.0->spacy==3.2.0) (2022.9.24)\n","Requirement already satisfied: chardet<4,>=3.0.2 in /usr/local/lib/python3.7/dist-packages (from requests<3.0.0,>=2.13.0->spacy==3.2.0) (3.0.4)\n","Requirement already satisfied: click<9.0.0,>=7.1.1 in /usr/local/lib/python3.7/dist-packages (from typer<0.5.0,>=0.3.0->spacy==3.2.0) (7.1.2)\n","Requirement already satisfied: MarkupSafe>=0.23 in /usr/local/lib/python3.7/dist-packages (from jinja2->spacy==3.2.0) (2.0.1)\n","Installing collected packages: typing-extensions, typer, pydantic, thinc, spacy\n","  Attempting uninstall: typing-extensions\n","    Found existing installation: typing-extensions 4.1.1\n","    Uninstalling typing-extensions-4.1.1:\n","      Successfully uninstalled typing-extensions-4.1.1\n","  Attempting uninstall: typer\n","    Found existing installation: typer 0.7.0\n","    Uninstalling typer-0.7.0:\n","      Successfully uninstalled typer-0.7.0\n","  Attempting uninstall: pydantic\n","    Found existing installation: pydantic 1.10.2\n","    Uninstalling pydantic-1.10.2:\n","      Successfully uninstalled pydantic-1.10.2\n","  Attempting uninstall: thinc\n","    Found existing installation: thinc 8.1.5\n","    Uninstalling thinc-8.1.5:\n","      Successfully uninstalled thinc-8.1.5\n","  Attempting uninstall: spacy\n","    Found existing installation: spacy 3.4.3\n","    Uninstalling spacy-3.4.3:\n","      Successfully uninstalled spacy-3.4.3\n","\u001b[31mERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\n","en-core-web-sm 3.4.1 requires spacy<3.5.0,>=3.4.0, but you have spacy 3.2.0 which is incompatible.\u001b[0m\n","Successfully installed pydantic-1.8.2 spacy-3.2.0 thinc-8.0.17 typer-0.4.2 typing-extensions-3.10.0.2\n"]},{"data":{"application/vnd.colab-display-data+json":{"pip_warning":{"packages":["typing_extensions"]}}},"metadata":{},"output_type":"display_data"}],"source":["#install SpaCy3\n","!pip install spacy==3.2.0"]},{"cell_type":"markdown","metadata":{"id":"rMQJe8mt9nEM"},"source":["STEP 2: Convert Label Studio data into accepted by SpaCy JSON format\n","*   [[\"some text\", {'entities':[(start_index, end_index, \"label\"), start_index, end_index, \"label\")]} ]]\n","\n","\n","\n","\n"]},{"cell_type":"code","execution_count":null,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":64910,"status":"ok","timestamp":1669735188875,"user":{"displayName":"Victoria Dimanova","userId":"07172063658264964881"},"user_tz":-60},"id":"woxO4wfLZNSv","outputId":"ff328d95-016c-448b-ca32-bd2217e775c8"},"outputs":[{"output_type":"stream","name":"stdout","text":["['/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/1-20', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/20-40', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/40-60', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/60-80', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/80-100', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/120-140']\n","['/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/100-120/cleaned_r26518356.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/100-120/cleaned_r26614966.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/100-120/cleaned_r26732679.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/100-120/cleaned_r26844283.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/100-120/cleaned_r26673617.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/100-120/cleaned_r26852132.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/100-120/cleaned_r26915813.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/100-120/cleaned_r26938303.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/100-120/cleaned_r26984517.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/100-120/cleaned_r27317303.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/100-120/cleaned_r27028892.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/100-120/cleaned_r27391467.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/100-120/cleaned_r27297409.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/100-120/cleaned_r27405968.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/100-120/cleaned_r27581518.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/100-120/cleaned_r27557561.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/100-120/cleaned_r27411014.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/100-120/cleaned_r27611858.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/100-120/cleaned_r27870898.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/100-120/cleaned_r27708362.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/140-151/cleaned_r34090504.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/140-151/cleaned_r29311296.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/140-151/cleaned_r34193219.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/140-151/cleaned_r34199925.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/140-151/cleaned_r34535716.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/140-151/cleaned_r34638611.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/140-151/cleaned_r34802442.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/140-151/cleaned_r34845204.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/140-151/cleaned_r35051209.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/140-151/cleaned_r35387780.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/140-151/cleaned_r34304248.json']\n","['/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/1-20/cleaned_r22694757.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/1-20/cleaned_r22730036.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/1-20/cleaned_r22973876.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/1-20/cleaned_r23193181.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/1-20/cleaned_r23275061.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/1-20/cleaned_r23484124.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/1-20/cleaned_r23497334.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/1-20/cleaned_r23520479.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/1-20/cleaned_r23533265.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/1-20/cleaned_r23554857.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/1-20/cleaned_r23566200.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/1-20/cleaned_r23603844.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/1-20/cleaned_r23730838.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/1-20/cleaned_r23636469.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/1-20/cleaned_r23762029.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/1-20/cleaned_r23758755.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/1-20/cleaned_r23818845.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/1-20/cleaned_r23840361.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/1-20/cleaned_r23826101.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/1-20/cleaned_r23939394.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/20-40/cleaned_r24015269.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/20-40/cleaned_r23958896.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/20-40/cleaned_r24088534.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/20-40/cleaned_r24088176.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/20-40/cleaned_r24089524.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/20-40/cleaned_r24130482.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/20-40/cleaned_r24167613.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/20-40/cleaned_r24228851.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/20-40/cleaned_r24340030.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/20-40/cleaned_r24244429.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/20-40/cleaned_r24386482.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/20-40/cleaned_r24286116.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/20-40/cleaned_r24244422.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/20-40/cleaned_r24244418.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/20-40/cleaned_r24312495.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/20-40/cleaned_r24386490.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/20-40/cleaned_r24489795.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/20-40/cleaned_r24401205.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/20-40/cleaned_r24407160.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/20-40/cleaned_r24411942.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/40-60/cleaned_r24579088.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/40-60/cleaned_r24498169.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/40-60/cleaned_r24505242.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/40-60/cleaned_r24505328.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/40-60/cleaned_r24505359.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/40-60/cleaned_r24559095.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/40-60/cleaned_r24798046.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/40-60/cleaned_r24877141.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/40-60/cleaned_r24530099.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/40-60/cleaned_r24724096.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/40-60/cleaned_r24788791.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/40-60/cleaned_r24641624.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/40-60/cleaned_r24722454.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/40-60/cleaned_r24799129.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/40-60/cleaned_r24595810.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/40-60/cleaned_r24797520.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/40-60/cleaned_r24564888.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/40-60/cleaned_r24983375.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/40-60/cleaned_r24647473.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/40-60/cleaned_r24565113.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/60-80/cleaned_r25288473.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/60-80/cleaned_r25105144.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/60-80/cleaned_r25142032.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/60-80/cleaned_r25144233.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/60-80/cleaned_r25296654.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/60-80/cleaned_r25159818.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/60-80/cleaned_r25136578.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/60-80/cleaned_r24984017.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/60-80/cleaned_r25049337.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/60-80/cleaned_r25198178.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/60-80/cleaned_r25057003.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/60-80/cleaned_r25393287.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/60-80/cleaned_r25407966.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/60-80/cleaned_r25393025.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/60-80/cleaned_r25303370.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/60-80/cleaned_r25658088.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/60-80/cleaned_r25549355.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/60-80/cleaned_r25664323.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/60-80/cleaned_r25673979.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/60-80/cleaned_r25432738.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/80-100/cleaned_r25889654.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/80-100/cleaned_r25880873.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/80-100/cleaned_r26272639.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/80-100/cleaned_r25776251.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/80-100/cleaned_r26137475.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/80-100/cleaned_r25714011.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/80-100/cleaned_r26340264.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/80-100/cleaned_r25896637.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/80-100/cleaned_r26425691.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/80-100/cleaned_r26075243.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/80-100/cleaned_r26385385.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/80-100/cleaned_r26324406.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/80-100/cleaned_r26147651.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/80-100/cleaned_r25952687.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/80-100/cleaned_r26039063.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/80-100/cleaned_r26170027.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/80-100/cleaned_r25921341.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/80-100/cleaned_r25743486.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/80-100/cleaned_r26306400.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/80-100/cleaned_r25960826.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/120-140/cleaned_r27923814.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/120-140/cleaned_r28017967.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/120-140/cleaned_r28076376.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/120-140/cleaned_r28125622.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/120-140/cleaned_r28228279.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/120-140/cleaned_r28145098.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/120-140/cleaned_r28301096.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/120-140/cleaned_r28377874.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/120-140/cleaned_r28473239.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/120-140/cleaned_r28399858.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/120-140/cleaned_r28530238.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/120-140/cleaned_r28550414.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/120-140/cleaned_r28764810.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/120-140/cleaned_r28865459.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/120-140/cleaned_r28860604.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/120-140/cleaned_r28928160.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/120-140/cleaned_r29025767.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/120-140/cleaned_r29075020.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/120-140/cleaned_r29160417.json', '/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/120-140/cleaned_r29183350.json']\n","Skipping entity [2006, 2012, MedTech] in the following text because the character span 'CD45RO' does not align with token boundaries:\n","'We compared the change in C-peptide AUC over 1 year in the drug-treated responders and non-responders vs control or placebo-treated subjects in each trial. Due to a baseline imbalance in HbA1c between placebo and drug-treated subjects in the Delay trial, C-peptide analysis was corrected for HbA1c levels at study entry in a mixed linear model. Changes in the proportion of T cell subsets from flow studies were compared by repeated measures ANOVA and corrected for the baseline levels. Absolute cell counts (Figure 2B) were calculated from the proportion of cells multiplied by the lymphocyte counts at the same blood draw. The presented flow cytometry data were corrected for the baseline unless otherwise indicated. Unless otherwise indicated, the data shown in the figures represent the LSMEANS values from the statistical model performed with SAS. Grouped data were compared either by t-tests, Wilcoxon test, Chi-squared test, ANOVA, or Kruskal-Wallis test (for data without normal distribution) as indicated. Calculations were done with GraphPad Prism (version 6.0) and SAS (version 9.3). The Nanostring data were normalized by (Nalini transformation) followed by quantile normalization across all samples as implemented in the R BioConductor Project library preprocessCore (Bioconductor: Open software development for computational biology and bioinformatics R. (Gentleman, V. J. Carey, et al Genome Biology, Vol. 5, R80, 2004). For technical reasons, the results from one of the non-responders could not be normalized to the other 17 samples and the results from this sample was not used in the analyses. Principle Component Analysis was performed as described. To identify differences in gene expression among CD8CM T cells, PBMC from the 2 month time point from drug treated responders (n=6), non-responders (n=4), and placebo-treated control subjects (n=8) from the 2 mo time point in the Delay trial were labeled with mAbs to CD2, CD8, CD45RO, and CCR7 for cell sorting. CD8CM T cells (CD2+CD8+CD45RO+CCR7+) were sorted and RNA was prepared. The nanostring nCounter GX Human Immunology gene expression panel was used for gene expression analysis of 594 immunology-related genes. To analyze the effects of teplizumab on T cell proliferation, PBMCs from 3 different healthy controls were intracellularly labeled with CellTrace Violet dye (Life Technologies, Grand Island, NY). Thawed PBMCs were cultured with 0.5 g/ml teplizumab and 1.0 g/ml of anti-CD28 mAb for 3 days. Proliferation of different T cell subsets was determined based on dilution of CellTrace Violet dye, staining with the above mAbs and flow cytometry. For some experiments, CD8CM or na¯ve cells were sorted from PBMC and labeled with CellTrace Violet dye. The remaining cells were labeled with CFSE. The two populations were combined and cultured with teplizumab and anti-CD28 mAb as above. The frequency of CellViolet+ CD8CM T cells was determined after 1, 2, and 3 days by flow cytometry. Fresh whole blood samples from AbATE were analyzed at the Immune Tolerance Network Flow Cytometry Core and fresh whole blood samples from the Delay trial were analyzed at the Yale Flow Cytometry Core. Fresh samples from study participants were stained with mAbs to CD8, CD4, CD45RA, CD45RO, and CD62L (BD Pharmingen, CA) to identify the following subsets of CD4+ and CD8+ T cells: CM: CD45RACD45RO+CD62L+; EM: CD45RACD45RO+CD62L; EMRA: CD45RA+CD45ROCD62L; na¯ve: CD45RA+CD45ROCD62L+. Gating strategies were kept consistent throughout each study and percentages of cell types or absolute cell counts (using the number of circulating lymphocytes from the CBC) were compared to identify changes with treatment. For gene expression analysis and cell culture studies, frozen cells were thawed prior to study. Because CD62L expression is affected by freezing and thawing, CCR7 was used to identify CD4+ or CD8+ T cell subsets: CM: CD45RO+CCR7+; EM: CD45RO+CCR7; EMRA: CD45ROCCR7; na¯ve: CD45RO+CCR7+. Cells were acquired with an LSRII or a FACSAria cytometer and analyzed with FlowJo software. Samples were analyzed without knowledge of the treatment assignment. The subjects in both trials underwent a 4-hr mixed meal tolerance test (MMTT) at the time of study entry and after 6 and 12 months. C-peptide levels were measured at Northwest Lipid Research Laboratory (Seattle, WA), and the area under the curve (AUC) was calculated using the trapezoidal rule. We defined responders as subjects who lost < 7.5% of their baseline C-peptide response at month 12. This criterion was more stringent than criteria used to designate responders in the AbATE trial based on year-2 C-peptide results but had been used in previous analyses of treatment responses. We studied subjects with T1D between the ages of 835 enrolled in the AbATE and Delay studies (NCT00129259 and NCT00378508). The AbATE study was a randomized open-label study of patients with new-onset (< 3 months) diabetes, and the Delay study was a randomized placebo-controlled study that enrolled subjects with 412 months duration of disease. The drug-treated subjects in both trials received the same dose of teplizumab for 14 days (total dose = 9.033 mg/m2). The descriptions of the study groups and results of the primary endpoint analysis for both trials have previously been published. The data presented herein are limited to the first year of each study to permit direct comparison of studies. The study protocols and use of samples for the mechanistic studies were approved at each study site. Written informed consent was obtained from the participants.'\n","Skipping entity [2013, 2017, MedTech] in the following text because the character span 'CCR7' does not align with token boundaries:\n","'We compared the change in C-peptide AUC over 1 year in the drug-treated responders and non-responders vs control or placebo-treated subjects in each trial. Due to a baseline imbalance in HbA1c between placebo and drug-treated subjects in the Delay trial, C-peptide analysis was corrected for HbA1c levels at study entry in a mixed linear model. Changes in the proportion of T cell subsets from flow studies were compared by repeated measures ANOVA and corrected for the baseline levels. Absolute cell counts (Figure 2B) were calculated from the proportion of cells multiplied by the lymphocyte counts at the same blood draw. The presented flow cytometry data were corrected for the baseline unless otherwise indicated. Unless otherwise indicated, the data shown in the figures represent the LSMEANS values from the statistical model performed with SAS. Grouped data were compared either by t-tests, Wilcoxon test, Chi-squared test, ANOVA, or Kruskal-Wallis test (for data without normal distribution) as indicated. Calculations were done with GraphPad Prism (version 6.0) and SAS (version 9.3). The Nanostring data were normalized by (Nalini transformation) followed by quantile normalization across all samples as implemented in the R BioConductor Project library preprocessCore (Bioconductor: Open software development for computational biology and bioinformatics R. (Gentleman, V. J. Carey, et al Genome Biology, Vol. 5, R80, 2004). For technical reasons, the results from one of the non-responders could not be normalized to the other 17 samples and the results from this sample was not used in the analyses. Principle Component Analysis was performed as described. To identify differences in gene expression among CD8CM T cells, PBMC from the 2 month time point from drug treated responders (n=6), non-responders (n=4), and placebo-treated control subjects (n=8) from the 2 mo time point in the Delay trial were labeled with mAbs to CD2, CD8, CD45RO, and CCR7 for cell sorting. CD8CM T cells (CD2+CD8+CD45RO+CCR7+) were sorted and RNA was prepared. The nanostring nCounter GX Human Immunology gene expression panel was used for gene expression analysis of 594 immunology-related genes. To analyze the effects of teplizumab on T cell proliferation, PBMCs from 3 different healthy controls were intracellularly labeled with CellTrace Violet dye (Life Technologies, Grand Island, NY). Thawed PBMCs were cultured with 0.5 g/ml teplizumab and 1.0 g/ml of anti-CD28 mAb for 3 days. Proliferation of different T cell subsets was determined based on dilution of CellTrace Violet dye, staining with the above mAbs and flow cytometry. For some experiments, CD8CM or na¯ve cells were sorted from PBMC and labeled with CellTrace Violet dye. The remaining cells were labeled with CFSE. The two populations were combined and cultured with teplizumab and anti-CD28 mAb as above. The frequency of CellViolet+ CD8CM T cells was determined after 1, 2, and 3 days by flow cytometry. Fresh whole blood samples from AbATE were analyzed at the Immune Tolerance Network Flow Cytometry Core and fresh whole blood samples from the Delay trial were analyzed at the Yale Flow Cytometry Core. Fresh samples from study participants were stained with mAbs to CD8, CD4, CD45RA, CD45RO, and CD62L (BD Pharmingen, CA) to identify the following subsets of CD4+ and CD8+ T cells: CM: CD45RACD45RO+CD62L+; EM: CD45RACD45RO+CD62L; EMRA: CD45RA+CD45ROCD62L; na¯ve: CD45RA+CD45ROCD62L+. Gating strategies were kept consistent throughout each study and percentages of cell types or absolute cell counts (using the number of circulating lymphocytes from the CBC) were compared to identify changes with treatment. For gene expression analysis and cell culture studies, frozen cells were thawed prior to study. Because CD62L expression is affected by freezing and thawing, CCR7 was used to identify CD4+ or CD8+ T cell subsets: CM: CD45RO+CCR7+; EM: CD45RO+CCR7; EMRA: CD45ROCCR7; na¯ve: CD45RO+CCR7+. Cells were acquired with an LSRII or a FACSAria cytometer and analyzed with FlowJo software. Samples were analyzed without knowledge of the treatment assignment. The subjects in both trials underwent a 4-hr mixed meal tolerance test (MMTT) at the time of study entry and after 6 and 12 months. C-peptide levels were measured at Northwest Lipid Research Laboratory (Seattle, WA), and the area under the curve (AUC) was calculated using the trapezoidal rule. We defined responders as subjects who lost < 7.5% of their baseline C-peptide response at month 12. This criterion was more stringent than criteria used to designate responders in the AbATE trial based on year-2 C-peptide results but had been used in previous analyses of treatment responses. We studied subjects with T1D between the ages of 835 enrolled in the AbATE and Delay studies (NCT00129259 and NCT00378508). The AbATE study was a randomized open-label study of patients with new-onset (< 3 months) diabetes, and the Delay study was a randomized placebo-controlled study that enrolled subjects with 412 months duration of disease. The drug-treated subjects in both trials received the same dose of teplizumab for 14 days (total dose = 9.033 mg/m2). The descriptions of the study groups and results of the primary endpoint analysis for both trials have previously been published. The data presented herein are limited to the first year of each study to permit direct comparison of studies. The study protocols and use of samples for the mechanistic studies were approved at each study site. Written informed consent was obtained from the participants.'\n","Skipping entity [1998, 2001, MedTech] in the following text because the character span 'CD2' does not align with token boundaries:\n","'We compared the change in C-peptide AUC over 1 year in the drug-treated responders and non-responders vs control or placebo-treated subjects in each trial. Due to a baseline imbalance in HbA1c between placebo and drug-treated subjects in the Delay trial, C-peptide analysis was corrected for HbA1c levels at study entry in a mixed linear model. Changes in the proportion of T cell subsets from flow studies were compared by repeated measures ANOVA and corrected for the baseline levels. Absolute cell counts (Figure 2B) were calculated from the proportion of cells multiplied by the lymphocyte counts at the same blood draw. The presented flow cytometry data were corrected for the baseline unless otherwise indicated. Unless otherwise indicated, the data shown in the figures represent the LSMEANS values from the statistical model performed with SAS. Grouped data were compared either by t-tests, Wilcoxon test, Chi-squared test, ANOVA, or Kruskal-Wallis test (for data without normal distribution) as indicated. Calculations were done with GraphPad Prism (version 6.0) and SAS (version 9.3). The Nanostring data were normalized by (Nalini transformation) followed by quantile normalization across all samples as implemented in the R BioConductor Project library preprocessCore (Bioconductor: Open software development for computational biology and bioinformatics R. (Gentleman, V. J. Carey, et al Genome Biology, Vol. 5, R80, 2004). For technical reasons, the results from one of the non-responders could not be normalized to the other 17 samples and the results from this sample was not used in the analyses. Principle Component Analysis was performed as described. To identify differences in gene expression among CD8CM T cells, PBMC from the 2 month time point from drug treated responders (n=6), non-responders (n=4), and placebo-treated control subjects (n=8) from the 2 mo time point in the Delay trial were labeled with mAbs to CD2, CD8, CD45RO, and CCR7 for cell sorting. CD8CM T cells (CD2+CD8+CD45RO+CCR7+) were sorted and RNA was prepared. The nanostring nCounter GX Human Immunology gene expression panel was used for gene expression analysis of 594 immunology-related genes. To analyze the effects of teplizumab on T cell proliferation, PBMCs from 3 different healthy controls were intracellularly labeled with CellTrace Violet dye (Life Technologies, Grand Island, NY). Thawed PBMCs were cultured with 0.5 g/ml teplizumab and 1.0 g/ml of anti-CD28 mAb for 3 days. Proliferation of different T cell subsets was determined based on dilution of CellTrace Violet dye, staining with the above mAbs and flow cytometry. For some experiments, CD8CM or na¯ve cells were sorted from PBMC and labeled with CellTrace Violet dye. The remaining cells were labeled with CFSE. The two populations were combined and cultured with teplizumab and anti-CD28 mAb as above. The frequency of CellViolet+ CD8CM T cells was determined after 1, 2, and 3 days by flow cytometry. Fresh whole blood samples from AbATE were analyzed at the Immune Tolerance Network Flow Cytometry Core and fresh whole blood samples from the Delay trial were analyzed at the Yale Flow Cytometry Core. Fresh samples from study participants were stained with mAbs to CD8, CD4, CD45RA, CD45RO, and CD62L (BD Pharmingen, CA) to identify the following subsets of CD4+ and CD8+ T cells: CM: CD45RACD45RO+CD62L+; EM: CD45RACD45RO+CD62L; EMRA: CD45RA+CD45ROCD62L; na¯ve: CD45RA+CD45ROCD62L+. Gating strategies were kept consistent throughout each study and percentages of cell types or absolute cell counts (using the number of circulating lymphocytes from the CBC) were compared to identify changes with treatment. For gene expression analysis and cell culture studies, frozen cells were thawed prior to study. Because CD62L expression is affected by freezing and thawing, CCR7 was used to identify CD4+ or CD8+ T cell subsets: CM: CD45RO+CCR7+; EM: CD45RO+CCR7; EMRA: CD45ROCCR7; na¯ve: CD45RO+CCR7+. Cells were acquired with an LSRII or a FACSAria cytometer and analyzed with FlowJo software. Samples were analyzed without knowledge of the treatment assignment. The subjects in both trials underwent a 4-hr mixed meal tolerance test (MMTT) at the time of study entry and after 6 and 12 months. C-peptide levels were measured at Northwest Lipid Research Laboratory (Seattle, WA), and the area under the curve (AUC) was calculated using the trapezoidal rule. We defined responders as subjects who lost < 7.5% of their baseline C-peptide response at month 12. This criterion was more stringent than criteria used to designate responders in the AbATE trial based on year-2 C-peptide results but had been used in previous analyses of treatment responses. We studied subjects with T1D between the ages of 835 enrolled in the AbATE and Delay studies (NCT00129259 and NCT00378508). The AbATE study was a randomized open-label study of patients with new-onset (< 3 months) diabetes, and the Delay study was a randomized placebo-controlled study that enrolled subjects with 412 months duration of disease. The drug-treated subjects in both trials received the same dose of teplizumab for 14 days (total dose = 9.033 mg/m2). The descriptions of the study groups and results of the primary endpoint analysis for both trials have previously been published. The data presented herein are limited to the first year of each study to permit direct comparison of studies. The study protocols and use of samples for the mechanistic studies were approved at each study site. Written informed consent was obtained from the participants.'\n","Skipping entity [2002, 2005, MedTech] in the following text because the character span 'CD8' does not align with token boundaries:\n","'We compared the change in C-peptide AUC over 1 year in the drug-treated responders and non-responders vs control or placebo-treated subjects in each trial. Due to a baseline imbalance in HbA1c between placebo and drug-treated subjects in the Delay trial, C-peptide analysis was corrected for HbA1c levels at study entry in a mixed linear model. Changes in the proportion of T cell subsets from flow studies were compared by repeated measures ANOVA and corrected for the baseline levels. Absolute cell counts (Figure 2B) were calculated from the proportion of cells multiplied by the lymphocyte counts at the same blood draw. The presented flow cytometry data were corrected for the baseline unless otherwise indicated. Unless otherwise indicated, the data shown in the figures represent the LSMEANS values from the statistical model performed with SAS. Grouped data were compared either by t-tests, Wilcoxon test, Chi-squared test, ANOVA, or Kruskal-Wallis test (for data without normal distribution) as indicated. Calculations were done with GraphPad Prism (version 6.0) and SAS (version 9.3). The Nanostring data were normalized by (Nalini transformation) followed by quantile normalization across all samples as implemented in the R BioConductor Project library preprocessCore (Bioconductor: Open software development for computational biology and bioinformatics R. (Gentleman, V. J. Carey, et al Genome Biology, Vol. 5, R80, 2004). For technical reasons, the results from one of the non-responders could not be normalized to the other 17 samples and the results from this sample was not used in the analyses. Principle Component Analysis was performed as described. To identify differences in gene expression among CD8CM T cells, PBMC from the 2 month time point from drug treated responders (n=6), non-responders (n=4), and placebo-treated control subjects (n=8) from the 2 mo time point in the Delay trial were labeled with mAbs to CD2, CD8, CD45RO, and CCR7 for cell sorting. CD8CM T cells (CD2+CD8+CD45RO+CCR7+) were sorted and RNA was prepared. The nanostring nCounter GX Human Immunology gene expression panel was used for gene expression analysis of 594 immunology-related genes. To analyze the effects of teplizumab on T cell proliferation, PBMCs from 3 different healthy controls were intracellularly labeled with CellTrace Violet dye (Life Technologies, Grand Island, NY). Thawed PBMCs were cultured with 0.5 g/ml teplizumab and 1.0 g/ml of anti-CD28 mAb for 3 days. Proliferation of different T cell subsets was determined based on dilution of CellTrace Violet dye, staining with the above mAbs and flow cytometry. For some experiments, CD8CM or na¯ve cells were sorted from PBMC and labeled with CellTrace Violet dye. The remaining cells were labeled with CFSE. The two populations were combined and cultured with teplizumab and anti-CD28 mAb as above. The frequency of CellViolet+ CD8CM T cells was determined after 1, 2, and 3 days by flow cytometry. Fresh whole blood samples from AbATE were analyzed at the Immune Tolerance Network Flow Cytometry Core and fresh whole blood samples from the Delay trial were analyzed at the Yale Flow Cytometry Core. Fresh samples from study participants were stained with mAbs to CD8, CD4, CD45RA, CD45RO, and CD62L (BD Pharmingen, CA) to identify the following subsets of CD4+ and CD8+ T cells: CM: CD45RACD45RO+CD62L+; EM: CD45RACD45RO+CD62L; EMRA: CD45RA+CD45ROCD62L; na¯ve: CD45RA+CD45ROCD62L+. Gating strategies were kept consistent throughout each study and percentages of cell types or absolute cell counts (using the number of circulating lymphocytes from the CBC) were compared to identify changes with treatment. For gene expression analysis and cell culture studies, frozen cells were thawed prior to study. Because CD62L expression is affected by freezing and thawing, CCR7 was used to identify CD4+ or CD8+ T cell subsets: CM: CD45RO+CCR7+; EM: CD45RO+CCR7; EMRA: CD45ROCCR7; na¯ve: CD45RO+CCR7+. Cells were acquired with an LSRII or a FACSAria cytometer and analyzed with FlowJo software. Samples were analyzed without knowledge of the treatment assignment. The subjects in both trials underwent a 4-hr mixed meal tolerance test (MMTT) at the time of study entry and after 6 and 12 months. C-peptide levels were measured at Northwest Lipid Research Laboratory (Seattle, WA), and the area under the curve (AUC) was calculated using the trapezoidal rule. We defined responders as subjects who lost < 7.5% of their baseline C-peptide response at month 12. This criterion was more stringent than criteria used to designate responders in the AbATE trial based on year-2 C-peptide results but had been used in previous analyses of treatment responses. We studied subjects with T1D between the ages of 835 enrolled in the AbATE and Delay studies (NCT00129259 and NCT00378508). The AbATE study was a randomized open-label study of patients with new-onset (< 3 months) diabetes, and the Delay study was a randomized placebo-controlled study that enrolled subjects with 412 months duration of disease. The drug-treated subjects in both trials received the same dose of teplizumab for 14 days (total dose = 9.033 mg/m2). The descriptions of the study groups and results of the primary endpoint analysis for both trials have previously been published. The data presented herein are limited to the first year of each study to permit direct comparison of studies. The study protocols and use of samples for the mechanistic studies were approved at each study site. Written informed consent was obtained from the participants.'\n","Skipping entity [3901, 3905, MedTech] in the following text because the character span 'CCR7' does not align with token boundaries:\n","'We compared the change in C-peptide AUC over 1 year in the drug-treated responders and non-responders vs control or placebo-treated subjects in each trial. Due to a baseline imbalance in HbA1c between placebo and drug-treated subjects in the Delay trial, C-peptide analysis was corrected for HbA1c levels at study entry in a mixed linear model. Changes in the proportion of T cell subsets from flow studies were compared by repeated measures ANOVA and corrected for the baseline levels. Absolute cell counts (Figure 2B) were calculated from the proportion of cells multiplied by the lymphocyte counts at the same blood draw. The presented flow cytometry data were corrected for the baseline unless otherwise indicated. Unless otherwise indicated, the data shown in the figures represent the LSMEANS values from the statistical model performed with SAS. Grouped data were compared either by t-tests, Wilcoxon test, Chi-squared test, ANOVA, or Kruskal-Wallis test (for data without normal distribution) as indicated. Calculations were done with GraphPad Prism (version 6.0) and SAS (version 9.3). The Nanostring data were normalized by (Nalini transformation) followed by quantile normalization across all samples as implemented in the R BioConductor Project library preprocessCore (Bioconductor: Open software development for computational biology and bioinformatics R. (Gentleman, V. J. Carey, et al Genome Biology, Vol. 5, R80, 2004). For technical reasons, the results from one of the non-responders could not be normalized to the other 17 samples and the results from this sample was not used in the analyses. Principle Component Analysis was performed as described. To identify differences in gene expression among CD8CM T cells, PBMC from the 2 month time point from drug treated responders (n=6), non-responders (n=4), and placebo-treated control subjects (n=8) from the 2 mo time point in the Delay trial were labeled with mAbs to CD2, CD8, CD45RO, and CCR7 for cell sorting. CD8CM T cells (CD2+CD8+CD45RO+CCR7+) were sorted and RNA was prepared. The nanostring nCounter GX Human Immunology gene expression panel was used for gene expression analysis of 594 immunology-related genes. To analyze the effects of teplizumab on T cell proliferation, PBMCs from 3 different healthy controls were intracellularly labeled with CellTrace Violet dye (Life Technologies, Grand Island, NY). Thawed PBMCs were cultured with 0.5 g/ml teplizumab and 1.0 g/ml of anti-CD28 mAb for 3 days. Proliferation of different T cell subsets was determined based on dilution of CellTrace Violet dye, staining with the above mAbs and flow cytometry. For some experiments, CD8CM or na¯ve cells were sorted from PBMC and labeled with CellTrace Violet dye. The remaining cells were labeled with CFSE. The two populations were combined and cultured with teplizumab and anti-CD28 mAb as above. The frequency of CellViolet+ CD8CM T cells was determined after 1, 2, and 3 days by flow cytometry. Fresh whole blood samples from AbATE were analyzed at the Immune Tolerance Network Flow Cytometry Core and fresh whole blood samples from the Delay trial were analyzed at the Yale Flow Cytometry Core. Fresh samples from study participants were stained with mAbs to CD8, CD4, CD45RA, CD45RO, and CD62L (BD Pharmingen, CA) to identify the following subsets of CD4+ and CD8+ T cells: CM: CD45RACD45RO+CD62L+; EM: CD45RACD45RO+CD62L; EMRA: CD45RA+CD45ROCD62L; na¯ve: CD45RA+CD45ROCD62L+. Gating strategies were kept consistent throughout each study and percentages of cell types or absolute cell counts (using the number of circulating lymphocytes from the CBC) were compared to identify changes with treatment. For gene expression analysis and cell culture studies, frozen cells were thawed prior to study. Because CD62L expression is affected by freezing and thawing, CCR7 was used to identify CD4+ or CD8+ T cell subsets: CM: CD45RO+CCR7+; EM: CD45RO+CCR7; EMRA: CD45ROCCR7; na¯ve: CD45RO+CCR7+. Cells were acquired with an LSRII or a FACSAria cytometer and analyzed with FlowJo software. Samples were analyzed without knowledge of the treatment assignment. The subjects in both trials underwent a 4-hr mixed meal tolerance test (MMTT) at the time of study entry and after 6 and 12 months. C-peptide levels were measured at Northwest Lipid Research Laboratory (Seattle, WA), and the area under the curve (AUC) was calculated using the trapezoidal rule. We defined responders as subjects who lost < 7.5% of their baseline C-peptide response at month 12. This criterion was more stringent than criteria used to designate responders in the AbATE trial based on year-2 C-peptide results but had been used in previous analyses of treatment responses. We studied subjects with T1D between the ages of 835 enrolled in the AbATE and Delay studies (NCT00129259 and NCT00378508). The AbATE study was a randomized open-label study of patients with new-onset (< 3 months) diabetes, and the Delay study was a randomized placebo-controlled study that enrolled subjects with 412 months duration of disease. The drug-treated subjects in both trials received the same dose of teplizumab for 14 days (total dose = 9.033 mg/m2). The descriptions of the study groups and results of the primary endpoint analysis for both trials have previously been published. The data presented herein are limited to the first year of each study to permit direct comparison of studies. The study protocols and use of samples for the mechanistic studies were approved at each study site. Written informed consent was obtained from the participants.'\n","Skipping entity [3957, 3961, MedTech] in the following text because the character span 'CCR7' does not align with token boundaries:\n","'We compared the change in C-peptide AUC over 1 year in the drug-treated responders and non-responders vs control or placebo-treated subjects in each trial. Due to a baseline imbalance in HbA1c between placebo and drug-treated subjects in the Delay trial, C-peptide analysis was corrected for HbA1c levels at study entry in a mixed linear model. Changes in the proportion of T cell subsets from flow studies were compared by repeated measures ANOVA and corrected for the baseline levels. Absolute cell counts (Figure 2B) were calculated from the proportion of cells multiplied by the lymphocyte counts at the same blood draw. The presented flow cytometry data were corrected for the baseline unless otherwise indicated. Unless otherwise indicated, the data shown in the figures represent the LSMEANS values from the statistical model performed with SAS. Grouped data were compared either by t-tests, Wilcoxon test, Chi-squared test, ANOVA, or Kruskal-Wallis test (for data without normal distribution) as indicated. Calculations were done with GraphPad Prism (version 6.0) and SAS (version 9.3). The Nanostring data were normalized by (Nalini transformation) followed by quantile normalization across all samples as implemented in the R BioConductor Project library preprocessCore (Bioconductor: Open software development for computational biology and bioinformatics R. (Gentleman, V. J. Carey, et al Genome Biology, Vol. 5, R80, 2004). For technical reasons, the results from one of the non-responders could not be normalized to the other 17 samples and the results from this sample was not used in the analyses. Principle Component Analysis was performed as described. To identify differences in gene expression among CD8CM T cells, PBMC from the 2 month time point from drug treated responders (n=6), non-responders (n=4), and placebo-treated control subjects (n=8) from the 2 mo time point in the Delay trial were labeled with mAbs to CD2, CD8, CD45RO, and CCR7 for cell sorting. CD8CM T cells (CD2+CD8+CD45RO+CCR7+) were sorted and RNA was prepared. The nanostring nCounter GX Human Immunology gene expression panel was used for gene expression analysis of 594 immunology-related genes. To analyze the effects of teplizumab on T cell proliferation, PBMCs from 3 different healthy controls were intracellularly labeled with CellTrace Violet dye (Life Technologies, Grand Island, NY). Thawed PBMCs were cultured with 0.5 g/ml teplizumab and 1.0 g/ml of anti-CD28 mAb for 3 days. Proliferation of different T cell subsets was determined based on dilution of CellTrace Violet dye, staining with the above mAbs and flow cytometry. For some experiments, CD8CM or na¯ve cells were sorted from PBMC and labeled with CellTrace Violet dye. The remaining cells were labeled with CFSE. The two populations were combined and cultured with teplizumab and anti-CD28 mAb as above. The frequency of CellViolet+ CD8CM T cells was determined after 1, 2, and 3 days by flow cytometry. Fresh whole blood samples from AbATE were analyzed at the Immune Tolerance Network Flow Cytometry Core and fresh whole blood samples from the Delay trial were analyzed at the Yale Flow Cytometry Core. Fresh samples from study participants were stained with mAbs to CD8, CD4, CD45RA, CD45RO, and CD62L (BD Pharmingen, CA) to identify the following subsets of CD4+ and CD8+ T cells: CM: CD45RACD45RO+CD62L+; EM: CD45RACD45RO+CD62L; EMRA: CD45RA+CD45ROCD62L; na¯ve: CD45RA+CD45ROCD62L+. Gating strategies were kept consistent throughout each study and percentages of cell types or absolute cell counts (using the number of circulating lymphocytes from the CBC) were compared to identify changes with treatment. For gene expression analysis and cell culture studies, frozen cells were thawed prior to study. Because CD62L expression is affected by freezing and thawing, CCR7 was used to identify CD4+ or CD8+ T cell subsets: CM: CD45RO+CCR7+; EM: CD45RO+CCR7; EMRA: CD45ROCCR7; na¯ve: CD45RO+CCR7+. Cells were acquired with an LSRII or a FACSAria cytometer and analyzed with FlowJo software. Samples were analyzed without knowledge of the treatment assignment. The subjects in both trials underwent a 4-hr mixed meal tolerance test (MMTT) at the time of study entry and after 6 and 12 months. C-peptide levels were measured at Northwest Lipid Research Laboratory (Seattle, WA), and the area under the curve (AUC) was calculated using the trapezoidal rule. We defined responders as subjects who lost < 7.5% of their baseline C-peptide response at month 12. This criterion was more stringent than criteria used to designate responders in the AbATE trial based on year-2 C-peptide results but had been used in previous analyses of treatment responses. We studied subjects with T1D between the ages of 835 enrolled in the AbATE and Delay studies (NCT00129259 and NCT00378508). The AbATE study was a randomized open-label study of patients with new-onset (< 3 months) diabetes, and the Delay study was a randomized placebo-controlled study that enrolled subjects with 412 months duration of disease. The drug-treated subjects in both trials received the same dose of teplizumab for 14 days (total dose = 9.033 mg/m2). The descriptions of the study groups and results of the primary endpoint analysis for both trials have previously been published. The data presented herein are limited to the first year of each study to permit direct comparison of studies. The study protocols and use of samples for the mechanistic studies were approved at each study site. Written informed consent was obtained from the participants.'\n","Skipping entity [1309, 1329, MedTech] in the following text because the character span 'galectin-9 ELISA kit' does not align with token boundaries:\n","\"Survival was measured as the number of months from resection to the last review. The Kaplan-Meier method was used to calculate survival curves, and the log rank test was used for the statistical analyses. The statistical analyses were conducted using Pearsons Ï2 tests, Fishers exact tests, t tests and Spearman's rank correlation. A Cox proportional hazards model was used for the univariate and multivariate analyses to evaluate the prognostic value of clinicopathological factors. All tests were two-sided, and P0.05 was considered to indicate statistical significance. All statistical analyses were performed using SPSS 17.0 software (IBM Corporation, Armonk, NY, USA). NK-92 cells treated with or without rh-galectin-9 for 4 h were added to anti-off slides, fixed in 4% paraformaldehyde and permeabilized using 0.2% Triton X-100 (X100, Sigma-Aldrich, USA). The cells were then incubated with rhodamine-labeled phalloidin (PHDR1, Cytoskeleton, USA) and DAPI (C1002, Beyotime Biotechnology, Shanghai, China) according to the manufacturers instructions. Image capturing and processing were performed using a laser scanning confocal microscope (LSM710, ZEISS, Germany). The supernatants of HT29 tumor cells were collected at 36h post-transfection. Each supernatant was centrifuged and stored at -80°C.Ahuman galectin-9 ELISA kit(Cusabio, WuHan, HuBei Province, China) was used according to the manufacturers instructions to measure galectin-9 production in each supernatant. Total RNA was extracted from cultured/transfected cells using TRIzol reagent (15596026, Invitrogen, USA) and then stored at -20°C. Using aReverTra Ace qPCR RT Kit (FSQ-101, Toyobo, Japan), we reverse transcribed 1 g of total RNA into DNA in a one-step reaction.This was followed by PCR to amplify the transcripts of the corresponding target molecule. The RT reactions and PCR steps were performed according to the manufacturers instructions. Table 2 lists the primer sequences that were obtained from Primer Bank. The target gene expression levels were normalized to GAPDH levels that were run during the same reaction. Cell lysates were resuspended in a small volume of lysis buffer as previously described and then subjected to western blot analysis. The details of the western blotprocedure are described in the Supplementary Materials and Methods (S1 File). We used transwell chambers with a 5-mm pore membrane (Costar, Corning, NY, USA). NK-92 cells or primary NK cells were plated in the top chambers. Colon tumor cell culture supernatant or varying concentrations of rh-galectin-9 (2045-GA-050, R&D, Minneapolis, MN, USA) were added to the bottom chamber. NK-92 cells or primary NK cells were permitted to migrate into the bottom chamber for 4 h at 37°C, They were then harvested and counted using cell-count boards. Because NK cells have a relatively high rate of spontaneous migration, we used the following the migration index (MI) to evaluate the results: migration index = the number of cells migrated/the number of randomly migrating cells (no chemokine). Y-27632 dihydrochloride (20 M) (ab120129, Abcam, Cambridge, Massachusetts,UK) and C3 transferase (2.0 g/mL) (CT04-A, Cytoskeleton, Denvor, CO, USA) were added to test the role of Rho/ROCK signaling in NK migration. IL-12 (7.5 ng/mL) was used as a positive control for NK cell chemoattraction (200-12P80, Peprotech, Chicago, USA). We transiently transfected of siRNA against human galectin-9 (siRNA #378 and siRNA #690, GenePharma, Shanghai, China) or scramble siRNA into HT-29 cells using Lipofectamine 2000 (11668019, Invitrogen, Carlsbad, CA, USA) according to the manufacturers instructions. The knockdown efficacy of the siRNAs targeting galectin-9 was examined using RT-PCR, western blot analysis and ELISA at 36 h post-transfection. The following siRNA sequences targeting human galectin-9 were used: Peripheral blood mononuclear cells (PBMCs) were obtained after centrifugation with Ficoll-Paque Plus (Amersham Biosciences, Sweden). NK cells were isolated using NK Cell Isolation Kits (MiltenyiBiotec, Germany) according the manufacturers instructions. The purity of CD3-CD56+ cells was consistently over 95%, measured through flow cytometric analysis. Cells were incubated according to the standard procedures described in the ATCC culture guide. The human colon tumor cell lines SW480, SW620 and HT29 were grown in Dulbecco's modified Eagles medium (DMEM) (Invitrogen, USA) supplemented with 5% fetal bovine serum (FCS) (Bio International, USA). The human NK92 cell line was a gift from The Second Military Medical University (Shanghai, China) and cultivated with Alpha Minimum Essential medium (Gibco, USA) containing 2 mM L-glutamine and 1.5 g/L sodium bicarbonate but without ribonucleosides and deoxyribonucleosides, and various components were added to obtain complete growth medium: 0.2 mM inositol (17508, Sigma, USA); 0.1 mM 2-mercaptoethanol (ES-007E, Merck Millipore, Germany); 0.02 mM folic acid (F8758, Sigma, USA); 100200 U/mL recombinant IL-2; and adjustment to a final concentration of 12.5% horse serum (16050, Gibco, USA) and 12.5% fetal bovine serum. All cells were incubated at 5% CO2 according to the standard procedures in the ATCC culture guide. Sections (4 m) of formalin-fixed paraffin-embedded colon tumor tissue samples were dewaxed, rehydrated, and incubated with the following primary antibodies: rabbit polyclonal anti-human galectin-9 (sc-366141, 1:200; Santa Cruz Biotechnology, Santa Cruz, CA, USA) and mouse monoclonal anti-human CD56 (3576, 1:50; Cell Signaling Technology, Beverly, MA,USA). The sections were subsequently stained with the corresponding secondary antibodies. Two independent pathologists who were blinded to the clinical data, evaluated the immunohistochemical results. The method used to evaluate the tissues is described in the Supplementary Materials and methods (S1 File). Our study included data that was obtained from 128 patients with histologically confirmed colon cancer who underwent surgery at the Qilu Hospital of Shandong University from January 2004 to December 2011 (Jinan, Shandong, China),This including one group of 38 patients in which we compared para-tumor with tumor tissue and another group of 90 patients were included in the survival analysis. The collection and use of tissue samples complied with the relevant guidelines and institutional practices of the Ethics Committee of Qilu Hospital of Shandong University, and written approval was obtained in each case before tissue sample collection. The ethics committee of Qilu Hospital of Shandong University approved this study (KYLL-2013-069). The key clinicopathological data are shown in Table 1 and the Supplementary Materials and methods (S1 File). siRNA378: sense, 5-GGAAGACACACAUGCCUUUTT-3; antisense, 5-AAAGGCAUGUGUGUCUUCCTT-3; and siRNA 690: sense, 5-CCAUUACCCAGACAGUCAUTT-3; antisense, 5-AUGACUGUCUGGGUAAUGGTT-3.\"\n","Skipping entity [4227, 4230, MedTech] in the following text because the character span 'PBS' does not align with token boundaries:\n","'The information of Total Therapy (TT) 2 [NCT00083551 and TT3 [NCT00081939 trials were published at the web site of National Institutes of Health (https://clinicaltrials.gov). The baseline bone marrow aspirates were harvested from the patients at diagnosis of MM according to established procedures at the University of Arkansas for Medical Sciences (UAMS). Bone marrow specimens were processed to obtain CD138+ plasma cells for gene expression profiling. The study was approved by the Institutional Review Board of UAMS, and informed consents were obtained from all study subjects and kept on record in accordance with the Declaration of Helsinki. Gene expression data for newly diagnosed myeloma patients without prior therapy in the total therapy 2 (TT2) and total therapy 3A (TT3A) trials were retrieved from NIH GEO omnibus under accession number GSE2658 and EBI ArrayExpress under the accession number E-TABM-1138. Gene expression data for the relapsed myeloma patients with one to three prior therapies in the APEX phase 3 trial 039 and 040 trials were retrieved from NIH GEO omnibus under accession number GSE9782. There are six probe sets designated to EPOR (19p13.2) on both U133Plus2.0 and U133A microarray platforms. Among them, probe set 209962_at was consensually selected due to compatibility of two types of Affymetrix microarrays and oligo-sequence alignment to EPOR locus. Because the patients enrolled to TT2 and TT3A trials were treated under different regimens, the receiver-operating characteristic (ROC) optimal cut point for EPOR in each clinical trial was identified with the running log-rank test, respectively, based on overall survival of the cohorts at the time of data collection. Hazard ratios were calculated by the Cox proportional model. All statistical computations were performed using R (version 2.13.2; available from http://www.r-project.org) and Bioconductor (available from http://www.bioconductor.org). Cell viability experiments of co-cultures were performed using a ScanR-automated fluorescence microscope (Olympus, Hamburg, Germany). Acquired images were analysed using the ScanR Image Analysis software. Image acquisition and analysis of myeloma cell viability was performed as described ]. INA-6 was washed 4 in Hanks balanced salt solution, incubated for 4\\xa0h in 1\\xa0% BSA in RPMI before treatment with 10\\xa0U/ml rhEPO for 5\\xa0min. Primary samples were seeded in 1\\xa0% BSA in RPMI and starved for 4\\xa0h before treatment with rhEPO, 10\\xa0U/ml, 5\\xa0min. Lysates of INA-6 and primary cells were made as previously described. Samples were subjected to electrophoreses and transferred to membranes by blotting. Membranes were then blocked with non-fat dried milk (5\\xa0%) in Tris-buffered saline supplemented with Tween-20 (TBS-T) (0.1\\xa0%) and incubated with antibody against p-ERK-1/2 (rabbit polyclonal IgG, Cell Signalling, 1:1000 and EPOR (sc101444, Santa Cruz, 1:200). For detecting p-JAK-2, membranes were blocked in 5\\xa0% BSA in TBS-T and incubated with antibody against p-JAK-2 (rabbit IgG anti p-JAK-2, Cell Signalling, 1:200). All antibodies were incubated up to 72\\xa0h at 4\\xa0°C. GAPDH (Abcam, Cambridge, UK, 1:50,000) was used as loading control. Antibody binding was visualized with Odyssey Licor detecting system using L680 anti-rabbit or L800 anti-mouse (1:20 000 in TBS-T). AnnexinV-FITC/PI staining (Abcam, ab14085) and analysis by flow cytometry was used to assess viability in the primary cells incubated with rhEPO in the presence or absence of anti-EPOR antibodies (Santa Cruz: sc101444, clone MM0031-6G7). The anti-EPOR was preincubated for 30\\xa0min together with the primary cells before rhEPO was added. Cell proliferation assay was performed after 48\\xa0h incubation with rhEPO using CellTiter-Glo Luminescent Cell Viability Assay (Promega, Madison, Wisconsin) according to the manufacturers instructions. EPOR on the surface of myeloma cells was stained with allophycocyanin (APC)-labelled anti-EPO receptor antibody [clone 12\\xa0K90] from Life Span Biosciences Inc (LS-C182845), 20\\xa0l/ml for 30\\xa0min on ice after 20\\xa0min preincubation with human Fc receptor binding inhibitor (eBioscience, 50\\xa0l/ml). Isotype control mouse IgG1 APC MG105 (Invitrogen) was used. All incubations were performed in PBS/1\\xa0% BSA. Flow cytometry was performed using LSR II (Becton Dickinson) with FACS Diva software (Becton Dickinson), and analysed with Cytobank. Gates were set on live cells with forward and side scatter and duplets were excluded. RNA was isolated and cDNA was made as previously described from primary patient samples and cell lines. TaqMan probes were used to detect the EPOR and EPO expression (Hs00959432_g1 and Hs01071096_g1, respectively, gene expression assays, Applied Biosystems, Foster City, CA). The comparative Ct-method was used for quantification with GAPDH (HS99999905_m1) as housekeeping gene on the StepOnePlus Real-Time PCR system (Applied Biosystems). For transient gene expression studies, INA-6 cells were transfected following the instructions for the Nucleofector device (Amaxa Biosystems, Cologne, Germany) using 2,5\\xa0g CD4 plasmid, Hs_ EpoR_5 Flexitube siRNA, Hs_EpoR_6 siRNA (SI02780351, SI02780400, Qiagen, MD, USA) or control siRNA (Qiagen). The INA-6 cell line was used in the knockdown studies since it is transfectable and because it had a relatively high level of EPOR expression. After 24\\xa0h, transfected cells were isolated using dynaBeads anti-CD4 separation (Dynal, Oslo, Norway). Cells were treated with rhEPO (10\\xa0U/ml) (sigma E5627, 100,000\\xa0U/mg protein) 48\\xa0h after transfection. The myeloma cell lines and CD138+ primary plasma cells from myeloma patients were grown and maintained as described. Isolation of CD138+ cells for experiment demanding fresh viable cells and co-culture of CD138+ cells and bone marrow stromal cells were done as previously reported. The project was approved by the Regional Ethics Committee (REK 2011/2029) and all patients has given informed consent.'\n","Skipping entity [10626, 10646, MedTech] in the following text because the character span 'Tris-buffered saline' does not align with token boundaries:\n","'The authors declare that they agree to submit the article for publication. All the patients were informed of sample collection and usage. The tissue samples were collected and used in accordance with approval by the ethics committee of The First Affiliated Hospital of Wenzhou Medical University. Approval for all the mouse experiments was obtained from the Animal Ethics and Welfare Committee of Wenzhou Medical University and all the animal experiments were conducted strictly according to the Guide for the Care and Use of Laboratory Animals and prior to animal experiments. The data are expressed as the meanstandard deviation (SD) for continuous variables and frequency (percentage) for categorical variables. Students t-test or Mann-Whitney U test were used to compare continuous variables, while the chi-square test or Fishers exact test was used to compare categorical variables between two groups. One-way analysis of variance or Kruskal-Wallis test were used for multiple group comparisons. Kaplan-Meier survival curve analysis was used for survival analysis, and a log-rank method was used to compare survival time between groups. All the statistical tests were two-sided, and a P-value <0.05 was considered statistically significant. All the analyses were conducted using SPSS V22.0 and GraphPad Prism 6 software. The cells in each group were evenly seeded into a low-viscosity 96-well plate (500 cells per well) and cultured in DMEM/F12 medium (1:1) containing B27 supplement, antibiotics, epidermal growth factor, basic fibroblast growth factor, hepatocyte growth factor, and 1% methylcellulose. The cells were incubated at 37°C, and the culture medium was changed every 3 days. After 14days of culture, spheres with a diameter>75m were counted. The cells were seeded in Transwell inserts containing Matrigel-coated porous membranes (1104 cells per insert) and cultured in serum-free DMEM. Complete medium containing 10% FBS was added to the lower chamber. After 12h of incubation, the cells remaining in the insert were removed with cotton swabs. The cells that were adherent to the lower side of the membrane were fixed with 10% acetic acid/90% methanol and subjected to haematoxylin and eosin (H&E) staining. For each Transwell membrane, five visual fields with the most migrated cells were selected, and the cells were counted under a microscope at 400 magnification. The experiment was repeated three times. Cells were seeded in six-well plates at a density of 5103 cells per well, and the medium was changed every 3 days. After 2 weeks of culture, the medium was removed, and the cells were washed with PBS three times to remove the unbound cells. The adherent cells were fixed with 4% paraformaldehyde at room temperature for 30min and then stained with 0.01% crystal violet for 30min. Cell colonies containing >50 cells were counted. The experiment was repeated three times. Cells, including LM3 cells, SNU-387 cells transfected with short hairpin RNA (shRNA) or scrambled (scrRNA) lentiviruses; Hep3B cells, PRF5 cells transfected with p-POSTN or p-cDNA3.1 overexpressed plasmid, were seeded in 96-well plates (5000 cells/well). After 6 h of incubation, the cells were washed with PBS three times. Fresh culture medium containing 0.5% FBS was added. Cell proliferation was assessed by the Cell Counting Kit-8 cell proliferation assay kit, and the absorbance was measured at a wavelength of 450nm. Each experiment was repeated three times. Dual-luciferase reporter plasmids, in which the luciferase reporter gene was inserted downstream of the CD133 promoter region, were synthesized by GeneChem Biotechnology Co., Ltd. The cells were seeded in 24-well plates; when the cells reached approximately 80% confluence, they were co-transfected with plasmids containing the Renilla luciferase gene and reporter gene with Lipofectamine\\x84¢ 3000 for 48h following the instructions. After the medium was discarded, the cells were washed twice with PBS and then lysed in 100l of Passive Lysis Buffer (PLB) on a shaker at room temperature for 30min. Next, 20l of the cell lysates was transferred to an EP tube and gently mixed with 50l of luciferase assay reagent II (LAR II) using a pipette to avoid generating bubbles. The fluorescence assay was performed using a GloMax 20/20 Luminometer (Promega). After detection, the EP tube was removed, and the reaction was quenched by adding 50l of Stop & Glo Reagent. The fluorescence intensity of the internal reference, Renilla luciferase, was measured again. Cells were washed with Dulbeccos PBS (D-PBS) and fixed with N,N-dipiperazine containing 2.5% formaldehyde at room temperature for 20min. After fixation, the cells were washed with D-PBS and then blocked with 5% goat serum at 37°C for 1h. Next, the cells were incubated with primary antibodies against POSTN, AP-2, CD133, or TGF1 at 37°C for 1 h. After washing three times with D-PBS, the cells were then incubated with Alexa Fluor 488-conjugated anti-mouse IgG and Alexa Fluor 594-conjugated anti-rabbit IgG in the dark for 1 h. After three washes, the cells were incubated with 4-6-diamidino-2-phenylindole (DAPI) for 1 min, sealed with Prolong Gold anti-fading reagent, and observed under a confocal microscope. In the negative control group, PBS was used to replace the primary antibodies. After crosslinking with 1% formaldehyde, the cells were lysed, and the DNA was broken into approximately 600-bp fragments using a Bioruptor 300 (Diagenode). Subsequently, immunoprecipitation was performed using magnetic beads and POSTN antibodies or normal rabbit IgG antibodies to bind the cleaved chromatin. After reversing the formaldehyde cross-linking of the protein and DNA complexes, the immunoprecipitated DNA was purified by using a ChIP column. The identity and quantity of the isolated DNA fragments were determined by PCR using primers designed to amplify the POSTN promoter region, which potentially contained CD133 binding sites. The primer sequences are listed in Supplementary Table S2. A C1000 Thermal Cycler (Bio-Rad) was used for PCR quantification, and each experiment was repeated three times. After washing three times using PBS with Triton X-100 (PBST), magnetic beads were incubated with anti-av3, anti-TGF1 and IgG antibodies on a shaker at room temperature for 1h and then washed three times with PBST. Next, the antibody-coupled beads were incubated with cellular protein lysates overnight at 4°C. After the supernatant was removed, the beads were washed with PBST three times and incubated with diluted elution buffer at 70°C for 10min. Finally, the pulled down protein complexes were analysed by SDS-PAGE/immunoblotting analysis. Cell or tissue samples were lysed in lysis solution, and the total protein concentration was determined by using the bicinchoninic acid (BCA) kit. The samples were loaded into 415% sodium dodecyl sulfate - polyacrylamide gel electrophoresis (SDS PAGE) gels (30g of protein per lane) for electrophoresis, and then, the proteins were transferred onto polyvinylidene fluoride (PVDF) membranes. Next, the membranes were incubated with 5% BSA for 1h to block nonspecific signals and then washed with TBST [10mM tris (pH8.0), 150mM NaCl, and 0.05% Tween 20] three times. Then, the membranes were incubated with primary antibodies against POSTN, FOXP3, c-Myc, AP-2, CD133, av3, TGF1, -actin, or GAPDH overnight at 4°C. The concentrations of the antibodies are listed in Supplementary Table S1. After being thoroughly washed three times with TBST, the membranes were incubated with secondary goat anti-rabbit or goat anti-rat antibodies coupled to horseradish peroxidase at room temperature for 1h, followed by three washes with TBST. Finally, the immune complexes were detected using a HyGLO HRP assay kit and analysed by a gel imaging analysis system. -actin used as the internal control. Total RNA was extracted from cells and HCC specimens using TRIzol reagent and was used for cDNA synthesis after its concentration was determined. Reverse transcription was conducted following the instructions provided in the RevertAid First-Strand cDNA Synthesis Kit. Subsequently, the synthesized cDNA was amplified using SYBR Premix Ex Taq (Perfect Real Time) in an ABI 7300 Real-Time System. The mRNA expression level of each candidate transcription factor was measured and normalized to the expression of 18S ribosomal RNA as an internal reference. The reaction conditions were as follows: 95°C for 10min, followed by 40cycles of 95°C for 15s and 60°C for 60s. The primer sequences are listed in Supplementary Table S2. To determine the proportion of CD133+ cells among hepatocellular carcinoma cells, cells were harvested, washed twice with PBS, resuspended in cell staining buffer at a concentration of 5105/100l and incubated on ice for 30min in the dark with FITC-conjugated antibodies against CD133 at a dilution of 1:200. After washing, the cells were sorted by using a BD FACSCalibur. Follow-up analysis was performed using FlowJo Software (Ashland, Kentucky, USA). Positive and negative gates were determined using immunoglobulin G (IgG)-stained and unstained controls. To semi-quantitatively assess protein expression, the IHC staining intensity was scored from 0 to 3, with 0, 1, 2 and 3 indicating no staining and weak, medium and strong staining, respectively. In addition, the percentage of positive cells was scored as 0 points for less than 5%, 1 point for 625%, 2 points for 2650%, 3 points for 5175%, and 4 points for above 75%. The final staining score was determined by multiplying the staining intensity score by the corresponding positive-cell percentage point. The microvascular density (MVD) was calculated by counting the CD31-positive cells, which were identified by a brown precipitate in the cytoplasm of endothelial cells; the vessels in each section were counted in 5 microscopic fields. Following the previously described IHC staining protocol, tumour tissues and the corresponding adjacent normal tissues were fixed in 4% paraformaldehyde immediately after harvest. After fixation, the tissues were subjected to a series of procedures, including paraffin embedding, sectioning (thickness of 4m), hydration in different concentrations of ethanol after dimethyl benzene dewaxing, antigen repair after washing (boiling in 1 citric acid-disodium hydrogen phosphate buffer), and blocking of the endogenous catalase activity. Then, the sections were stained with antibodies against CD90, CD133, CK19, TWIST, SNAIL, CXCR4, and SMA. Negative controls were established by substituting the primary antibody solutions with 1% bovine serum albumin (BSA)/Tris-buffered saline (TBS). Finally, the slides were cover slipped and observed under a microscope. Fresh tumour tissues extracted from patients in the operating room were cut into small blocks approximately 1mm3 in size under aseptic conditions. The nude mice were anaesthetized with isoflurane and maintained in the anaesthetic state. In a sterile environment, a skin incision approximately 0.5cm long was made on the lateral side of the right hind limb using a sterilized apparatus. The small human tumour tissue blocks were then subcutaneously implanted into the mice, and the skin incision was closed. Tumour growth was monitored regularly by measuring the size of the tumours twice a week with a Vernier caliper. When the tumours grew to approximately 200mm3, the mice were given intragastric administration of lenvatinib (10mg/kg) and intraperitoneal injection of cilengitide (20mg/kg) every 3 days. The body weight and tumour growth of the mice were monitored. All the animal experiments were approved by the Wenzhou Medical University Animal Care and Use Committee. A total of 48 BALB/c nude mice (Laboratory Animal Center, Wenzhou Medical University, Wenzhou, China) were randomly divided into eight groups with six mice in each group, including four groups that underwent orthotopic implantation of liver cancer cells (Hep3B p-cDNA3.1, Hep3B p-POSTN, LM3 Scr-POSTN, and LM3 Sh-POSTN) and four groups that underwent subcutaneous implantation of tumours. After the mice were anaesthetized, laparotomy was conducted to expose the left lobe of the liver for the orthotopic implantation of liver cancer cells carrying a luciferase reporter gene. After the operation, the growth status of the mice was evaluated weekly, and the fluorescence intensity was measured regularly. For imaging, the mice were intraperitoneally injected with the D-luciferin substrate (15mg/mL) at a dose of 10L/g; 1015min after injection, imaging was performed on the IVIS-SPECTRUM in vivo imaging system (PerkinElmer). The images were analysed, and the fluorescence intensity values after tumour cell inoculation were measured by using in vivo imaging software (Caliper Life Sciences). To establish a subcutaneously transplanted tumour model, the mice were subcutaneously implanted with liver cancer cells in the lateral side of the right hind limb. Tumour growth was monitored by measuring the tumour size twice a week with a Vernier caliper. On the 33rd day after xenograft establishment, all the mice were sacrificed to harvest subcutaneous tumour, lung, and brain tissue samples. The volume of the tumour was determined by using the formula: volume=widthlength0.52. All three tissue types were fixed with 4% paraformaldehyde and embedded in paraffin for subsequent IHC analysis. All the animal experiments were strictly conducted according to the Guide for the Care and Use of Laboratory Animals, and prior to the animal experiments, an agreement was drafted based on the 1975 Declaration of Helsinki and approved by the Animal Ethics and Welfare Committee of Wenzhou Medical University. Transcriptome sequencing of HCC cell lines with different levels of POSTN gene expression was conducted by GeneChem Biotechnology Co., Ltd., Shanghai (GSAE0154536). The change in the POSTN transcript copy number was calculated based on the ratio of the expression level in tumour tissues to that in normal tissues, and the median value in normal tissues was used as the baseline. The Venn diagram shows the genes associated with the top 50 upregulated genes in Hep3B cells after transfection with control and p-POSTN expression plasmids, the genes associated with the top 50 downregulated genes in HCCLM3 cells after transfection with lentivirus carrying shRNA targeting POSTN and matched scrambled RNA, and 19 genes encoding transcriptional regulators of CD133 as predicted by the TRANSFAC transcription factor prediction database. The GSE1898 dataset includes data of HCC tumour tissues and matched benign tumour-adjacent tissues obtained from 139 individuals (61 Chinese patients and 78 Caucasian patients) by surgical resection. The details of the microarray gene expression profile were reported by Lee JS et al.. The fold change (FC) in the POSTN transcript was calculated as the ratio of the gene expression value in tumour tissues to that in normal tissues, and the mean value in normal tissues was used as the baseline. A log2-transformed tumour/normal value ¥1.0 (i.e., a fold change of tumour vs. normal ¥2) or1.0 indicated an increase or decrease in gene expression, respectively. Spearman rank correlation analysis was used to analyse the correlation between POSTN and stemness-related factors. The mRNA expression data of 349 HCC specimens from The Cancer Genome Atlas (TCGA) database and 240 HCC specimens from the International Cancer Genome Consortium (ICGC) database were extracted for analysis. The Phatmap package and the corolot package in R language were used to generate the correlation matrix heat map between the POSTN gene and stemness-related factors. Specimens, including tumour tissues and matched adjacent normal tissues (¥2cm from the resection margin), were collected from 110 HCC patients who underwent surgical resection in the Department of Hepatobiliary Surgery, The First Affiliated Hospital of Wenzhou Medical University, between November 2008 and October 2010. The HCC patients were between 24 and 79years old (median age was 52years old), and the median follow-up time was 38.2months. The clinicopathological features analysed in this study included postoperative tumour-node-metastasis (TNM) staging and Edmondson-Steiner grading by experienced pathologists. In addition, another batch of liver cancer and adjacent tissue samples were obtained between November 2017 and March 2018. Each of these samples was divided into three parts: one was used to establish a PDX model, one was flash frozen and stored at 80°C for molecular analysis, and one was fixed in formaldehyde for histopathological examination. This study was approved by the ethics committee of The First Affiliated Hospital of Wenzhou Medical University, and written informed consent was obtained from each patient. A comprehensive list of all the reagents, kits, antibodies, and primers used in this study can be found in Supplementary Tables S1 and S2. All data and materials supporting the findings of this work are available from its supplementary information files and from the corresponding author upon reasonable request.'\n","Skipping entity [10621, 10624, MedTech] in the following text because the character span 'BSA' does not align with token boundaries:\n","'The authors declare that they agree to submit the article for publication. All the patients were informed of sample collection and usage. The tissue samples were collected and used in accordance with approval by the ethics committee of The First Affiliated Hospital of Wenzhou Medical University. Approval for all the mouse experiments was obtained from the Animal Ethics and Welfare Committee of Wenzhou Medical University and all the animal experiments were conducted strictly according to the Guide for the Care and Use of Laboratory Animals and prior to animal experiments. The data are expressed as the meanstandard deviation (SD) for continuous variables and frequency (percentage) for categorical variables. Students t-test or Mann-Whitney U test were used to compare continuous variables, while the chi-square test or Fishers exact test was used to compare categorical variables between two groups. One-way analysis of variance or Kruskal-Wallis test were used for multiple group comparisons. Kaplan-Meier survival curve analysis was used for survival analysis, and a log-rank method was used to compare survival time between groups. All the statistical tests were two-sided, and a P-value <0.05 was considered statistically significant. All the analyses were conducted using SPSS V22.0 and GraphPad Prism 6 software. The cells in each group were evenly seeded into a low-viscosity 96-well plate (500 cells per well) and cultured in DMEM/F12 medium (1:1) containing B27 supplement, antibiotics, epidermal growth factor, basic fibroblast growth factor, hepatocyte growth factor, and 1% methylcellulose. The cells were incubated at 37°C, and the culture medium was changed every 3 days. After 14days of culture, spheres with a diameter>75m were counted. The cells were seeded in Transwell inserts containing Matrigel-coated porous membranes (1104 cells per insert) and cultured in serum-free DMEM. Complete medium containing 10% FBS was added to the lower chamber. After 12h of incubation, the cells remaining in the insert were removed with cotton swabs. The cells that were adherent to the lower side of the membrane were fixed with 10% acetic acid/90% methanol and subjected to haematoxylin and eosin (H&E) staining. For each Transwell membrane, five visual fields with the most migrated cells were selected, and the cells were counted under a microscope at 400 magnification. The experiment was repeated three times. Cells were seeded in six-well plates at a density of 5103 cells per well, and the medium was changed every 3 days. After 2 weeks of culture, the medium was removed, and the cells were washed with PBS three times to remove the unbound cells. The adherent cells were fixed with 4% paraformaldehyde at room temperature for 30min and then stained with 0.01% crystal violet for 30min. Cell colonies containing >50 cells were counted. The experiment was repeated three times. Cells, including LM3 cells, SNU-387 cells transfected with short hairpin RNA (shRNA) or scrambled (scrRNA) lentiviruses; Hep3B cells, PRF5 cells transfected with p-POSTN or p-cDNA3.1 overexpressed plasmid, were seeded in 96-well plates (5000 cells/well). After 6 h of incubation, the cells were washed with PBS three times. Fresh culture medium containing 0.5% FBS was added. Cell proliferation was assessed by the Cell Counting Kit-8 cell proliferation assay kit, and the absorbance was measured at a wavelength of 450nm. Each experiment was repeated three times. Dual-luciferase reporter plasmids, in which the luciferase reporter gene was inserted downstream of the CD133 promoter region, were synthesized by GeneChem Biotechnology Co., Ltd. The cells were seeded in 24-well plates; when the cells reached approximately 80% confluence, they were co-transfected with plasmids containing the Renilla luciferase gene and reporter gene with Lipofectamine\\x84¢ 3000 for 48h following the instructions. After the medium was discarded, the cells were washed twice with PBS and then lysed in 100l of Passive Lysis Buffer (PLB) on a shaker at room temperature for 30min. Next, 20l of the cell lysates was transferred to an EP tube and gently mixed with 50l of luciferase assay reagent II (LAR II) using a pipette to avoid generating bubbles. The fluorescence assay was performed using a GloMax 20/20 Luminometer (Promega). After detection, the EP tube was removed, and the reaction was quenched by adding 50l of Stop & Glo Reagent. The fluorescence intensity of the internal reference, Renilla luciferase, was measured again. Cells were washed with Dulbeccos PBS (D-PBS) and fixed with N,N-dipiperazine containing 2.5% formaldehyde at room temperature for 20min. After fixation, the cells were washed with D-PBS and then blocked with 5% goat serum at 37°C for 1h. Next, the cells were incubated with primary antibodies against POSTN, AP-2, CD133, or TGF1 at 37°C for 1 h. After washing three times with D-PBS, the cells were then incubated with Alexa Fluor 488-conjugated anti-mouse IgG and Alexa Fluor 594-conjugated anti-rabbit IgG in the dark for 1 h. After three washes, the cells were incubated with 4-6-diamidino-2-phenylindole (DAPI) for 1 min, sealed with Prolong Gold anti-fading reagent, and observed under a confocal microscope. In the negative control group, PBS was used to replace the primary antibodies. After crosslinking with 1% formaldehyde, the cells were lysed, and the DNA was broken into approximately 600-bp fragments using a Bioruptor 300 (Diagenode). Subsequently, immunoprecipitation was performed using magnetic beads and POSTN antibodies or normal rabbit IgG antibodies to bind the cleaved chromatin. After reversing the formaldehyde cross-linking of the protein and DNA complexes, the immunoprecipitated DNA was purified by using a ChIP column. The identity and quantity of the isolated DNA fragments were determined by PCR using primers designed to amplify the POSTN promoter region, which potentially contained CD133 binding sites. The primer sequences are listed in Supplementary Table S2. A C1000 Thermal Cycler (Bio-Rad) was used for PCR quantification, and each experiment was repeated three times. After washing three times using PBS with Triton X-100 (PBST), magnetic beads were incubated with anti-av3, anti-TGF1 and IgG antibodies on a shaker at room temperature for 1h and then washed three times with PBST. Next, the antibody-coupled beads were incubated with cellular protein lysates overnight at 4°C. After the supernatant was removed, the beads were washed with PBST three times and incubated with diluted elution buffer at 70°C for 10min. Finally, the pulled down protein complexes were analysed by SDS-PAGE/immunoblotting analysis. Cell or tissue samples were lysed in lysis solution, and the total protein concentration was determined by using the bicinchoninic acid (BCA) kit. The samples were loaded into 415% sodium dodecyl sulfate - polyacrylamide gel electrophoresis (SDS PAGE) gels (30g of protein per lane) for electrophoresis, and then, the proteins were transferred onto polyvinylidene fluoride (PVDF) membranes. Next, the membranes were incubated with 5% BSA for 1h to block nonspecific signals and then washed with TBST [10mM tris (pH8.0), 150mM NaCl, and 0.05% Tween 20] three times. Then, the membranes were incubated with primary antibodies against POSTN, FOXP3, c-Myc, AP-2, CD133, av3, TGF1, -actin, or GAPDH overnight at 4°C. The concentrations of the antibodies are listed in Supplementary Table S1. After being thoroughly washed three times with TBST, the membranes were incubated with secondary goat anti-rabbit or goat anti-rat antibodies coupled to horseradish peroxidase at room temperature for 1h, followed by three washes with TBST. Finally, the immune complexes were detected using a HyGLO HRP assay kit and analysed by a gel imaging analysis system. -actin used as the internal control. Total RNA was extracted from cells and HCC specimens using TRIzol reagent and was used for cDNA synthesis after its concentration was determined. Reverse transcription was conducted following the instructions provided in the RevertAid First-Strand cDNA Synthesis Kit. Subsequently, the synthesized cDNA was amplified using SYBR Premix Ex Taq (Perfect Real Time) in an ABI 7300 Real-Time System. The mRNA expression level of each candidate transcription factor was measured and normalized to the expression of 18S ribosomal RNA as an internal reference. The reaction conditions were as follows: 95°C for 10min, followed by 40cycles of 95°C for 15s and 60°C for 60s. The primer sequences are listed in Supplementary Table S2. To determine the proportion of CD133+ cells among hepatocellular carcinoma cells, cells were harvested, washed twice with PBS, resuspended in cell staining buffer at a concentration of 5105/100l and incubated on ice for 30min in the dark with FITC-conjugated antibodies against CD133 at a dilution of 1:200. After washing, the cells were sorted by using a BD FACSCalibur. Follow-up analysis was performed using FlowJo Software (Ashland, Kentucky, USA). Positive and negative gates were determined using immunoglobulin G (IgG)-stained and unstained controls. To semi-quantitatively assess protein expression, the IHC staining intensity was scored from 0 to 3, with 0, 1, 2 and 3 indicating no staining and weak, medium and strong staining, respectively. In addition, the percentage of positive cells was scored as 0 points for less than 5%, 1 point for 625%, 2 points for 2650%, 3 points for 5175%, and 4 points for above 75%. The final staining score was determined by multiplying the staining intensity score by the corresponding positive-cell percentage point. The microvascular density (MVD) was calculated by counting the CD31-positive cells, which were identified by a brown precipitate in the cytoplasm of endothelial cells; the vessels in each section were counted in 5 microscopic fields. Following the previously described IHC staining protocol, tumour tissues and the corresponding adjacent normal tissues were fixed in 4% paraformaldehyde immediately after harvest. After fixation, the tissues were subjected to a series of procedures, including paraffin embedding, sectioning (thickness of 4m), hydration in different concentrations of ethanol after dimethyl benzene dewaxing, antigen repair after washing (boiling in 1 citric acid-disodium hydrogen phosphate buffer), and blocking of the endogenous catalase activity. Then, the sections were stained with antibodies against CD90, CD133, CK19, TWIST, SNAIL, CXCR4, and SMA. Negative controls were established by substituting the primary antibody solutions with 1% bovine serum albumin (BSA)/Tris-buffered saline (TBS). Finally, the slides were cover slipped and observed under a microscope. Fresh tumour tissues extracted from patients in the operating room were cut into small blocks approximately 1mm3 in size under aseptic conditions. The nude mice were anaesthetized with isoflurane and maintained in the anaesthetic state. In a sterile environment, a skin incision approximately 0.5cm long was made on the lateral side of the right hind limb using a sterilized apparatus. The small human tumour tissue blocks were then subcutaneously implanted into the mice, and the skin incision was closed. Tumour growth was monitored regularly by measuring the size of the tumours twice a week with a Vernier caliper. When the tumours grew to approximately 200mm3, the mice were given intragastric administration of lenvatinib (10mg/kg) and intraperitoneal injection of cilengitide (20mg/kg) every 3 days. The body weight and tumour growth of the mice were monitored. All the animal experiments were approved by the Wenzhou Medical University Animal Care and Use Committee. A total of 48 BALB/c nude mice (Laboratory Animal Center, Wenzhou Medical University, Wenzhou, China) were randomly divided into eight groups with six mice in each group, including four groups that underwent orthotopic implantation of liver cancer cells (Hep3B p-cDNA3.1, Hep3B p-POSTN, LM3 Scr-POSTN, and LM3 Sh-POSTN) and four groups that underwent subcutaneous implantation of tumours. After the mice were anaesthetized, laparotomy was conducted to expose the left lobe of the liver for the orthotopic implantation of liver cancer cells carrying a luciferase reporter gene. After the operation, the growth status of the mice was evaluated weekly, and the fluorescence intensity was measured regularly. For imaging, the mice were intraperitoneally injected with the D-luciferin substrate (15mg/mL) at a dose of 10L/g; 1015min after injection, imaging was performed on the IVIS-SPECTRUM in vivo imaging system (PerkinElmer). The images were analysed, and the fluorescence intensity values after tumour cell inoculation were measured by using in vivo imaging software (Caliper Life Sciences). To establish a subcutaneously transplanted tumour model, the mice were subcutaneously implanted with liver cancer cells in the lateral side of the right hind limb. Tumour growth was monitored by measuring the tumour size twice a week with a Vernier caliper. On the 33rd day after xenograft establishment, all the mice were sacrificed to harvest subcutaneous tumour, lung, and brain tissue samples. The volume of the tumour was determined by using the formula: volume=widthlength0.52. All three tissue types were fixed with 4% paraformaldehyde and embedded in paraffin for subsequent IHC analysis. All the animal experiments were strictly conducted according to the Guide for the Care and Use of Laboratory Animals, and prior to the animal experiments, an agreement was drafted based on the 1975 Declaration of Helsinki and approved by the Animal Ethics and Welfare Committee of Wenzhou Medical University. Transcriptome sequencing of HCC cell lines with different levels of POSTN gene expression was conducted by GeneChem Biotechnology Co., Ltd., Shanghai (GSAE0154536). The change in the POSTN transcript copy number was calculated based on the ratio of the expression level in tumour tissues to that in normal tissues, and the median value in normal tissues was used as the baseline. The Venn diagram shows the genes associated with the top 50 upregulated genes in Hep3B cells after transfection with control and p-POSTN expression plasmids, the genes associated with the top 50 downregulated genes in HCCLM3 cells after transfection with lentivirus carrying shRNA targeting POSTN and matched scrambled RNA, and 19 genes encoding transcriptional regulators of CD133 as predicted by the TRANSFAC transcription factor prediction database. The GSE1898 dataset includes data of HCC tumour tissues and matched benign tumour-adjacent tissues obtained from 139 individuals (61 Chinese patients and 78 Caucasian patients) by surgical resection. The details of the microarray gene expression profile were reported by Lee JS et al.. The fold change (FC) in the POSTN transcript was calculated as the ratio of the gene expression value in tumour tissues to that in normal tissues, and the mean value in normal tissues was used as the baseline. A log2-transformed tumour/normal value ¥1.0 (i.e., a fold change of tumour vs. normal ¥2) or1.0 indicated an increase or decrease in gene expression, respectively. Spearman rank correlation analysis was used to analyse the correlation between POSTN and stemness-related factors. The mRNA expression data of 349 HCC specimens from The Cancer Genome Atlas (TCGA) database and 240 HCC specimens from the International Cancer Genome Consortium (ICGC) database were extracted for analysis. The Phatmap package and the corolot package in R language were used to generate the correlation matrix heat map between the POSTN gene and stemness-related factors. Specimens, including tumour tissues and matched adjacent normal tissues (¥2cm from the resection margin), were collected from 110 HCC patients who underwent surgical resection in the Department of Hepatobiliary Surgery, The First Affiliated Hospital of Wenzhou Medical University, between November 2008 and October 2010. The HCC patients were between 24 and 79years old (median age was 52years old), and the median follow-up time was 38.2months. The clinicopathological features analysed in this study included postoperative tumour-node-metastasis (TNM) staging and Edmondson-Steiner grading by experienced pathologists. In addition, another batch of liver cancer and adjacent tissue samples were obtained between November 2017 and March 2018. Each of these samples was divided into three parts: one was used to establish a PDX model, one was flash frozen and stored at 80°C for molecular analysis, and one was fixed in formaldehyde for histopathological examination. This study was approved by the ethics committee of The First Affiliated Hospital of Wenzhou Medical University, and written informed consent was obtained from each patient. A comprehensive list of all the reagents, kits, antibodies, and primers used in this study can be found in Supplementary Tables S1 and S2. All data and materials supporting the findings of this work are available from its supplementary information files and from the corresponding author upon reasonable request.'\n","Skipping entity [3734, 3743, MedTech] in the following text because the character span 'RPMI 1640' does not align with token boundaries:\n","\"Statistical analysis was performed using the Shapiro-Wilk W test for normality, Student's t-test and nonparametric Wilcoxon signed-rank test for paired and unpaired samples. Parametric (Pearson's r) and nonparametric (Spearman's rho) statistics were used for measurements of correlation. Analyses were performed with the JMP program (SAS, Cary, NC). P values<0.05 were considered to be statistically significant. Macaques were inoculated intravenously or intrarectally with 100 ID50 of SIVmac251 and plasma IFN levels were measured by Luminex as previously described with monoclonals MMHA-11 and MMHA-2 (PBL) used as capture and detector antibodies, respectively. Viral load was determined by TaqMan RNA RT-PCR assay with a sensitivity of 200 SIV RNA copies/ml of blood (Applied Biosystems). Absolute CD4+ T cell numbers were determined by flow cytometry. A high-titer virus stock was produced following a 3 day cell-associated infection of PM1 cells with HIV-1Ba-L. The infectious cell-free virus present in the culture supernatant was treated with 250 µM aldrithiol-2 (AT-2; Sigma Aldrich, St. Louis, MO) for 1 hour at 37°C, concentrated and the AT-2 HIV-1 stock was tested for infection. CD4+ T cells were isolated from whole blood by negative selection using the RosetteSep system (STEMCELL Technologies Inc., Vancouver, BC). Purity was >96%, as determined by flow cytometry. Cells were cultured in complete RPMI at a density of 106 cells/ml/well and pre-treated for 1 hour with 5 µM of a TLR7/9-specific antagonist (13mer phosphorothioate deoxyribose inhibitor that we have previously characterized); Invitrogen Corporation, Carlsbad, CA), 10 µg/ml of an anti-IFN/ receptor blocking antibody (Millipore), or an isotype control (eBioscience); 7104105 TCID50/ml of infectious HIV-1Ba-L or an equivalent p24 amount of AT-2 HIV-1 was then added for 3 days before the cells were stimulated with anti-CD95/Fas-antibodies for 14 hours. Percentages of apoptotic cells were determined by flow cytometry as described above. IFN present in supernatants was measured at 24 hours following virus exposure using the Human Interferon Alpha Multi-Subtype ELISA Kit (PBL). PBMC from chronically HIV-1-infected patients were isolated by Ficoll-Hypaque gradient centrifugation. Knockdown of Bak expression was achieved through siRNA transfection by electroporation using a Gene Pulser XCell (BioRad). PBMC (7.58106 cells) were washed and resuspended in 300 µl of Opti-MEM (Life Technologies) in a 2-mm cuvette and pulsed (square wave, 500 V, 1 ms) with 1 nmol of siRNA (ON-TARGET Non-Targeting pool or Bak ON-TARGETplus SMARTpool, Dharmacon). The siRNA pools consist of a mixture of 4 siRNAs that are pre-designed to reduce off-target effects by up to 90%. Knockdown efficiency in HIV-1 patient PBMC following siRNA transfection was determined by real-time quantitative PCR with Taqman gene expression assays (Applied Biosystems, Foster City, CA) for Bak (assay Hs00832876_g1), Bax (assay Hs00180269_m1) which served as a negative control for knock-down specificity, and 18S rRNA (assay Hs03928985_g1) which served as a loading and normalization control. Transfection efficiency was assessed with siGlo fluorescent oligonucleotides (Dharmacon). Following a 72 hour incubation period, cells were left unstimulated or stimulated with plate-bound anti-CD95 monoclonal antibody for 14 hours and assayed for apoptosis as described above. CD95/Fas-induced specific apoptosis was calculated by subtracting the level of spontaneous cell death and apoptosis caused by electroporation alone. To determine the sensitivity of T cells to apoptosis stimuli, PBMC (106 cells/ml) from HIV-1-infected individuals or healthy donors were stimulated in complete RPMI (RPMI 1640/10% heat-inactivated fetal bovine serum/2 mM L-glutamine/100 U/ml penicillin/100 µg/ml streptomycin sulfate, Cellgro, Manassas, VA) at 37°C in 5% CO2, in the presence or absence of plate-bound anti-CD95 monoclonal antibody (plates coated with 5 µg/ml, CH11, Millipore), soluble SuperKiller (cross-linked) TRAIL (10 and 100 ng/ml, Alexis Biochemicals, Lausen, Switzerland) or TNF (10 and 100 ng/ml, R & D Systems, Inc., Minneapolis, MN) for 14 hours, harvested and stained. For IFN/ apoptosis sensitization studies, PBMC from healthy and HIV-1-infected individuals were first stimulated with human IFN or IFN (1000 U/ml; PBL, Piscataway, NJ) for 3 days in the presence or absence of plate-bound anti-CD3 antibody (plates coated with 0.1 µg/ml OKT3) at 37°C in 5% CO2, before apoptosis sensitivity was evaluated by re-stimulating the cells with anti-CD95 monoclonal antibody (plates coated with 5 µg/ml), anti-CD3 monoclonal antibody (plates coated with 5 µg/ml), TRAIL (10 ng/ml) or TNF (10 ng/ml). Plates were coated with antibodies as previously described. Cells were then stained and fixed for flow cytometry. Specific apoptosis was calculated using the following formula: [(percentage of induced apoptosis - percentage of spontaneous apoptosis)/(100 - percentage of spontaneous apoptosis)]100. PBMC were freshly isolated from heparinized venous blood by centrifugation over a Ficoll-Hypaque gradient (Amersham Pharmacia Biotech, Uppsala, Sweden). HIV-specific, CMV-specific and EBV-specific CD8+ T cells were detected using tetramers of HLA class I A*0201 loaded with either HIV-Gag p17 7785 (SLYNTVATL), HIV-Pol 476484 (ILKEPVHGV), CMV p65 495503 (NLVPMVATV) or EBV 280288 (GLCTLVAML) peptide. To analyze the apoptosis sensitivity of PBMC, cells were stained with Annexin V Cy5.5/anti-CD8 PE-Texas Red (Caltag, Burlingame, CA)/anti-CD4 FITC/anti-CD3 Pacific Blue/HIV- or CMV-specific tetramer APC. Annexin V and all other antibodies were purchased from BD Biosciences (San Diego, CA). Briefly, 106 cells were stained with tetramers and antibodies in FACS wash (HBSS (Cellgro, Herndon, VA), 3% horse serum (Life Technologies, Carlsbad, CA), 0.02% NaN3) for 30 minutes on ice; washed with FACS wash and fixed with 1% paraformaldehyde. When Annexin V staining was performed, 2.5 mM CaCl2 was included in all steps. Intracellular levels of pro- and anti-apoptotic proteins were measured directly ex vivo in PBMC or following a 3 day incubation in the presence or absence of 1000 U/ml IFN/  anti-CD3 antibody (plates coated with 0.1 µg/ml). Following Live/Dead (Invitrogen) and surface staining with anti-CD8 PE-Texas Red/anti-CD4 FITC/anti-CD3 Pacific Blue, cells were permeabilized with cytotofix/cytoperm buffer (BD Biosciences) and intracellular staining was performed for 1 hour with fluorochrome conjugated antibodies (anti-Bcl-2 and anti-Bcl-xL antibodies, BD Biosciences), or specific primary antibodies (anti-FADD, Biovision, Mountain View, CA; anti-Bid, Epitomics, Burlingame, CA; anti-Bim, Millipore, Billerica, MA; anti-Bax, Abgent, San Diego, CA; anti-Bak, Epitomics) and appropriate isotype controls, followed by a 1 hour incubation with a secondary antibody (PE-conjugated anti-rat IgG or anti-rabbit IgG, Southern Biotechnologies). Samples were collected on a FACSAria (BD Biosciences) and analyzed using FlowJo software (Treestar, San Carlos, CA). Mean fluorescence intensity (MFI) of intracellular protein expression as expressed as delta () MFI indicates: MFI of specific antibody  MFI of isotype control. All of the patients were HIV-1 positive for at least 1 year (range 130 years); median CD4 count was 432 cells/µl (range 101,551 cells/µl); median CD8 count was 980 cells/µl (range 2242,565 cells/µl); median viral load was 325 RNA copies/ml blood (range <20978,190 copies/ml blood); 66 patients were on antiretroviral therapy (ART). HIV-1 viral loads (HIV RNA copies/ml plasma) were determined using COBAS AmpliPrep/COBAS TaqMan HIV-1 Test, v2.0 (Roche Diagnostics, Indianapolis, IN) with an assay range from 2010,000,000 HIV RNA copies/ml. IFN-induced gene expression studies were conducted with ART-untreated HIV-1-infected patients (ART na¯ve). Control samples were obtained from HIV-1 seronegative age-matched healthy individuals. All assays were performed on freshly isolated PBMC from HIV-1-infected and healthy individuals. Peripheral blood was collected from individuals following Drexel University College of Medicine Institutional Review Board (IRB) approval and obtaining written informed consent. Rhesus macaques (Macaca mulatta) were housed at BIOQUAL, Inc. (Rockville, MD), in accordance with the standards of the American Association for Accreditation of Laboratory Animal Care. The protocol was approved by the BIOQUAL's Institutional Animal care and Use Committee under OLAW Assurance Number A-3086-01. Bioqual is IAAALAC accredited and procedures were carried out in accordance with the recommendations of the Weatherall report. Total RNA was extracted from the freshly isolated PBMC of HIV-1-infected individuals and healthy donors. The RNA was reverse transcribed into complementary DNA which was then subjected to real-time PCR using gene-specific primers for IFI6-16 (5-cctgctgctcttcacttgca-3 and 5-ccgacggccatgaaggt-3) and ISG56 (5-ctggactggcaatagcaagct-3 and 5-gagggtcaatggcgttctga-3) as described previously. IFN-stimulated gene expression levels were normalized to -actin controls and results were calculated for each HIV-1-infected patient as a fold change in gene expression, relative to healthy donors using the delta-delta Ct method of analysis.\"\n","Skipping entity [5575, 5588, MedTech] in the following text because the character span 'anti-CD4 FITC' does not align with token boundaries:\n","\"Statistical analysis was performed using the Shapiro-Wilk W test for normality, Student's t-test and nonparametric Wilcoxon signed-rank test for paired and unpaired samples. Parametric (Pearson's r) and nonparametric (Spearman's rho) statistics were used for measurements of correlation. Analyses were performed with the JMP program (SAS, Cary, NC). P values<0.05 were considered to be statistically significant. Macaques were inoculated intravenously or intrarectally with 100 ID50 of SIVmac251 and plasma IFN levels were measured by Luminex as previously described with monoclonals MMHA-11 and MMHA-2 (PBL) used as capture and detector antibodies, respectively. Viral load was determined by TaqMan RNA RT-PCR assay with a sensitivity of 200 SIV RNA copies/ml of blood (Applied Biosystems). Absolute CD4+ T cell numbers were determined by flow cytometry. A high-titer virus stock was produced following a 3 day cell-associated infection of PM1 cells with HIV-1Ba-L. The infectious cell-free virus present in the culture supernatant was treated with 250 µM aldrithiol-2 (AT-2; Sigma Aldrich, St. Louis, MO) for 1 hour at 37°C, concentrated and the AT-2 HIV-1 stock was tested for infection. CD4+ T cells were isolated from whole blood by negative selection using the RosetteSep system (STEMCELL Technologies Inc., Vancouver, BC). Purity was >96%, as determined by flow cytometry. Cells were cultured in complete RPMI at a density of 106 cells/ml/well and pre-treated for 1 hour with 5 µM of a TLR7/9-specific antagonist (13mer phosphorothioate deoxyribose inhibitor that we have previously characterized); Invitrogen Corporation, Carlsbad, CA), 10 µg/ml of an anti-IFN/ receptor blocking antibody (Millipore), or an isotype control (eBioscience); 7104105 TCID50/ml of infectious HIV-1Ba-L or an equivalent p24 amount of AT-2 HIV-1 was then added for 3 days before the cells were stimulated with anti-CD95/Fas-antibodies for 14 hours. Percentages of apoptotic cells were determined by flow cytometry as described above. IFN present in supernatants was measured at 24 hours following virus exposure using the Human Interferon Alpha Multi-Subtype ELISA Kit (PBL). PBMC from chronically HIV-1-infected patients were isolated by Ficoll-Hypaque gradient centrifugation. Knockdown of Bak expression was achieved through siRNA transfection by electroporation using a Gene Pulser XCell (BioRad). PBMC (7.58106 cells) were washed and resuspended in 300 µl of Opti-MEM (Life Technologies) in a 2-mm cuvette and pulsed (square wave, 500 V, 1 ms) with 1 nmol of siRNA (ON-TARGET Non-Targeting pool or Bak ON-TARGETplus SMARTpool, Dharmacon). The siRNA pools consist of a mixture of 4 siRNAs that are pre-designed to reduce off-target effects by up to 90%. Knockdown efficiency in HIV-1 patient PBMC following siRNA transfection was determined by real-time quantitative PCR with Taqman gene expression assays (Applied Biosystems, Foster City, CA) for Bak (assay Hs00832876_g1), Bax (assay Hs00180269_m1) which served as a negative control for knock-down specificity, and 18S rRNA (assay Hs03928985_g1) which served as a loading and normalization control. Transfection efficiency was assessed with siGlo fluorescent oligonucleotides (Dharmacon). Following a 72 hour incubation period, cells were left unstimulated or stimulated with plate-bound anti-CD95 monoclonal antibody for 14 hours and assayed for apoptosis as described above. CD95/Fas-induced specific apoptosis was calculated by subtracting the level of spontaneous cell death and apoptosis caused by electroporation alone. To determine the sensitivity of T cells to apoptosis stimuli, PBMC (106 cells/ml) from HIV-1-infected individuals or healthy donors were stimulated in complete RPMI (RPMI 1640/10% heat-inactivated fetal bovine serum/2 mM L-glutamine/100 U/ml penicillin/100 µg/ml streptomycin sulfate, Cellgro, Manassas, VA) at 37°C in 5% CO2, in the presence or absence of plate-bound anti-CD95 monoclonal antibody (plates coated with 5 µg/ml, CH11, Millipore), soluble SuperKiller (cross-linked) TRAIL (10 and 100 ng/ml, Alexis Biochemicals, Lausen, Switzerland) or TNF (10 and 100 ng/ml, R & D Systems, Inc., Minneapolis, MN) for 14 hours, harvested and stained. For IFN/ apoptosis sensitization studies, PBMC from healthy and HIV-1-infected individuals were first stimulated with human IFN or IFN (1000 U/ml; PBL, Piscataway, NJ) for 3 days in the presence or absence of plate-bound anti-CD3 antibody (plates coated with 0.1 µg/ml OKT3) at 37°C in 5% CO2, before apoptosis sensitivity was evaluated by re-stimulating the cells with anti-CD95 monoclonal antibody (plates coated with 5 µg/ml), anti-CD3 monoclonal antibody (plates coated with 5 µg/ml), TRAIL (10 ng/ml) or TNF (10 ng/ml). Plates were coated with antibodies as previously described. Cells were then stained and fixed for flow cytometry. Specific apoptosis was calculated using the following formula: [(percentage of induced apoptosis - percentage of spontaneous apoptosis)/(100 - percentage of spontaneous apoptosis)]100. PBMC were freshly isolated from heparinized venous blood by centrifugation over a Ficoll-Hypaque gradient (Amersham Pharmacia Biotech, Uppsala, Sweden). HIV-specific, CMV-specific and EBV-specific CD8+ T cells were detected using tetramers of HLA class I A*0201 loaded with either HIV-Gag p17 7785 (SLYNTVATL), HIV-Pol 476484 (ILKEPVHGV), CMV p65 495503 (NLVPMVATV) or EBV 280288 (GLCTLVAML) peptide. To analyze the apoptosis sensitivity of PBMC, cells were stained with Annexin V Cy5.5/anti-CD8 PE-Texas Red (Caltag, Burlingame, CA)/anti-CD4 FITC/anti-CD3 Pacific Blue/HIV- or CMV-specific tetramer APC. Annexin V and all other antibodies were purchased from BD Biosciences (San Diego, CA). Briefly, 106 cells were stained with tetramers and antibodies in FACS wash (HBSS (Cellgro, Herndon, VA), 3% horse serum (Life Technologies, Carlsbad, CA), 0.02% NaN3) for 30 minutes on ice; washed with FACS wash and fixed with 1% paraformaldehyde. When Annexin V staining was performed, 2.5 mM CaCl2 was included in all steps. Intracellular levels of pro- and anti-apoptotic proteins were measured directly ex vivo in PBMC or following a 3 day incubation in the presence or absence of 1000 U/ml IFN/  anti-CD3 antibody (plates coated with 0.1 µg/ml). Following Live/Dead (Invitrogen) and surface staining with anti-CD8 PE-Texas Red/anti-CD4 FITC/anti-CD3 Pacific Blue, cells were permeabilized with cytotofix/cytoperm buffer (BD Biosciences) and intracellular staining was performed for 1 hour with fluorochrome conjugated antibodies (anti-Bcl-2 and anti-Bcl-xL antibodies, BD Biosciences), or specific primary antibodies (anti-FADD, Biovision, Mountain View, CA; anti-Bid, Epitomics, Burlingame, CA; anti-Bim, Millipore, Billerica, MA; anti-Bax, Abgent, San Diego, CA; anti-Bak, Epitomics) and appropriate isotype controls, followed by a 1 hour incubation with a secondary antibody (PE-conjugated anti-rat IgG or anti-rabbit IgG, Southern Biotechnologies). Samples were collected on a FACSAria (BD Biosciences) and analyzed using FlowJo software (Treestar, San Carlos, CA). Mean fluorescence intensity (MFI) of intracellular protein expression as expressed as delta () MFI indicates: MFI of specific antibody  MFI of isotype control. All of the patients were HIV-1 positive for at least 1 year (range 130 years); median CD4 count was 432 cells/µl (range 101,551 cells/µl); median CD8 count was 980 cells/µl (range 2242,565 cells/µl); median viral load was 325 RNA copies/ml blood (range <20978,190 copies/ml blood); 66 patients were on antiretroviral therapy (ART). HIV-1 viral loads (HIV RNA copies/ml plasma) were determined using COBAS AmpliPrep/COBAS TaqMan HIV-1 Test, v2.0 (Roche Diagnostics, Indianapolis, IN) with an assay range from 2010,000,000 HIV RNA copies/ml. IFN-induced gene expression studies were conducted with ART-untreated HIV-1-infected patients (ART na¯ve). Control samples were obtained from HIV-1 seronegative age-matched healthy individuals. All assays were performed on freshly isolated PBMC from HIV-1-infected and healthy individuals. Peripheral blood was collected from individuals following Drexel University College of Medicine Institutional Review Board (IRB) approval and obtaining written informed consent. Rhesus macaques (Macaca mulatta) were housed at BIOQUAL, Inc. (Rockville, MD), in accordance with the standards of the American Association for Accreditation of Laboratory Animal Care. The protocol was approved by the BIOQUAL's Institutional Animal care and Use Committee under OLAW Assurance Number A-3086-01. Bioqual is IAAALAC accredited and procedures were carried out in accordance with the recommendations of the Weatherall report. Total RNA was extracted from the freshly isolated PBMC of HIV-1-infected individuals and healthy donors. The RNA was reverse transcribed into complementary DNA which was then subjected to real-time PCR using gene-specific primers for IFI6-16 (5-cctgctgctcttcacttgca-3 and 5-ccgacggccatgaaggt-3) and ISG56 (5-ctggactggcaatagcaagct-3 and 5-gagggtcaatggcgttctga-3) as described previously. IFN-stimulated gene expression levels were normalized to -actin controls and results were calculated for each HIV-1-infected patient as a fold change in gene expression, relative to healthy donors using the delta-delta Ct method of analysis.\"\n","Skipping entity [8202, 8211, MedTech] in the following text because the character span 'RPMI 1640' does not align with token boundaries:\n","'Differences in cell apoptosis and differences in transcript levels between cytarabine/DNR and panobinostat treated (individually or combined) and untreated cells were compared using the pair-wise two-sample t-test. The Kaplan-Meier method was used to estimate survival probability. Statistical analyses were performed with GraphPad Prism software. Error bars on all bar graphs represent standard errors of the means. The chromatin immunoprecipitation (ChIP) assay was performed as described previously. THP-1 cells were treated with or without 10 nM panobinostat for 48 h. Anti-E2F1 antibody (sc-193x) or normal rabbit IgG (sc-2027, Santa Cruz Biotechnology) were used for the immunoprecipitation. Real-time PCR for the E2F1 binding regions in the BRCA1, CHK1, or RAD51 promoter was performed using the primers listed in Table S2. Pharmacodynamic (PD) studies were performed to determine in vivo effects of panobinostat treatment on BRCA1, CHK1 and RAD51. When bioluminescence reached 1.5107 p/s/cm2/sr, day 17 after injection of U937 cells, mice received panobinostat (5 mg/kg once daily2). Four hours after the second dose mice were sacrificed, bone marrow was harvested, and snap frozen pellets were stored at 80°C. The percentage of leukemic cell infiltration in bone marrow was determined by hematoxylin and eosin staining. To evaluate antileukemic activity, groups of 810 tumor-bearing mice were randomly assigned to receive panobinostat [5 mg/kg once daily for 3 weeks, provided by Novartis Pharma AG (Basel, Switzerland)], cytarabine (6.25 mg/kg once daily for 4 weeks), or combination (panobinostat 5 mg/kg once daily for 3 weeks plus ara-C 6.25 mg/kg once daily for 4 weeks) starting on day 3 after injection of U937 cells. A group of vehicle treated mice were included as controls with each experiment. For intraperitoneal injection panobinostat was formulated in 5% dextrose and cytarabine was formulated in PBS. U937 cells were transfected with a murine stem cell virus-green fluorescent protein vector containing the firefly Luciferase gene (pMSCV-luc-IRES-GFP), sorted by flow cytometry, expanded in culture and 10,000 cells were injected intravenously into the tail vein of 8- to 12-week-old female NOD-SCID-IL2Rnull (NSG) mice (n\\x8a=\\x8a36), as described previously. THP-1 cells, infected with NTC-, BRCA1-, CHK1- or RAD51-shRNA, were treated with 50 µM cytarabine for 3 h. The cells were washed and cultured for up to 8 h. The harvested cells were mixed with 1.0% low melting point agarose in PBS (Mg and Ca free) at 37oC. Approximately 5000 cells were layered onto pre-coated microscope slides (pre-coated slides were coated with a thin layer of 1% low melting point agarose and allowed to dry for 90 min), a glass coverslip was placed on top of the cell/agarose suspension and the gels were allowed to gel for 10 min at 4°C. The slides were then placed in lysis solution (2.5 M NaCl, 10 mM Tris, 100 mM EDTA, 10% DMSO and 1% Triton X-100, pH 10.0) overnight at 4°C. The slides were placed in electrophoresis buffer (1 mM EDTA, 300 mM NaOH, pH 13.0) for 40 min at 4°C to allow unwinding of the DNA. Electrophoresis was conducted for 30 min at 30 V (1.25 V/cm). After electrophoresis, the slides were rinsed in 400 mL distilled water, dipped in 95% ethanol and dried. DNA was stained with 130,000 SYBR Gold (Life Techonologies) in 10 mM Tris, pH 7.5 plus 1 mM EDTA for 40 min at room temperature. The slides were rinsed in distilled water and imaged on an Olympus BX-40 microscope with an Olympus DP72 microscope camera and Olympus cellSens Dimension software (Olympus America Inc., Center Valley, PA). 50 comets per gel were scored using CometScore (TriTek Corp, Sumerduck, VA). The median percent DNA in the tail from at least three replicate gels were averaged and graphed. The error bars indicate standard errors of the means. The pMD-VSV-G and delta 8.2 plasmids were gifts from Dr. Dong at the Tulane University. The transfection was carried out by using Lipofectamine and Plus reagents (Life Technologies) according to the manufacturers instructions. Briefly, a lentivirus vector (BRCA1, CHK1, or RAD51 shRNA construct from Sigma-Aldrich or CHK1 cDNA construct from Thermo Fisher Scientific Biosciences, Lafayette, CO), pMD-VSV-G and delta 8.2 were cotransfected into TLA-HEK293T cells and the culture medium was harvested 48 h post-transfection. One million THP-1 Cells were transduced by adding 1 ml of virus supernatant and 4 µg of polybrene (Sigma-Aldrich). In vitro cytotoxicities of AML diagnostic blasts were measured by using MTT (3-[4,5-dimethyl-thiazol-2-yl]-2,5-diphenyltetrazoliumbromide, Sigma-Aldrich) assays, as previously described. Briefly, the cells were cultured as mentioned in \\x9cPanobinostat Treatment of Diagnostic Blast Samples\\x9d and treated with variable concentrations of panobinostat (0160 nM) for 48 hours. IC50 values were calculated as drug concentrations necessary to inhibit 50% proliferation compared to untreated control cells. IC50 values are means of duplicates from one experiment. Cells were lysed in Tris buffer (10 mM, pH 8.0) containing protease inhibitors (Roche Diagnostics). Whole cell lysates were subjected to SDS-polyacrylamide gel electrophoresis, electrophoretically transferred onto polyvinylidene difluoride (PVDF) membranes (Thermo Fisher Inc., Rockford, IL) and immunoblotted with anti-acetyl-histone 4 (ac-H4, 06598), -H4 (07108) (Upstate Biotechnology, Lake Placid, NY), -BRCA1 (9010), -H2AX (2577), -RPL13a (2765), -Caspase 3 (9661), -PARP (9542), -p-CDC25C (9528), -E2F1 (3742) (Cell Signaling Technology, Danvers, MA), -RAD51 (sc-8349), -CHK1 (sc-8408, Santa Cruz Biotechnology, Santa Cruz, CA) -acetyl-tubulin (T7451) or --actin (A2228, Sigma-Aldrich) antibody, as described previously. Immunoreactive proteins were visualized using the Odyssey Infrared Imaging System (Li-Cor, Lincoln, NE), as described by the manufacturer. Western blots were repeated at least 3 times and one representative blot is shown. Total RNA was extracted using TRIzol (Life Technologies) and cDNAs were prepared from 1 µg total RNA using random hexamer primers and a RT-PCR kit (Life Technologies), and purified with the QIAquick PCR Purification Kit (Qiagen, Valencia, CA) as previously described. BRCA1 (Hs00173237_m1), CHK1 (Hx00967506_m1), RAD51 (Hs00153418_m1) and E2F1 (Hs00153451_m1) transcripts were quantitated using Taqman probes (Life Technologies) and a LightCycler real-time PCR machine (Roche Diagnostics, Indianapolis, IN), based on the manufacturers instructions. Real-time PCR results were expressed as means from 3 independent experiments and were normalized to GAPDH (4333764) or RPL13a (Hs03043885_g1) transcripts. Fold changes were calculated using the comparative Ct method. Cell cycle analysis was done as previously described. Briefly, AML Cells were harvested and fixed with ice-cold 80% (v/v) ethanol for 24 h. After centrifugation at 200g for 5 min, the cell pellets were washed with phosphate-buffered saline (PBS, pH 7.4) and resuspended in PBS containing PI (50 mg/mL), triton X-100 (0.1%, v/v), and DNase-free RNase (1 mg/mL). The DNA contents were determined by flow cytometry using a FACScan flow cytometer (BD Biosciences, San Jose, CA). Data analysis was done with the Multicycle software (Phoenix Flow Systems, Inc., San Diego, CA). Experiments were performed 3 independent times in triplicate. Data presented are from one representative experiment. AML cell lines were treated with panobinostat and cytarabine or DNR for 48 h and subjected to flow cytometry analysis to determine drug-induced apoptosis using the Annexin V-fluorescein isothiocyanate (FITC)/propidium iodide (PI) Apoptosis Kit (Beckman Coulter; Brea, CA), as previously described. Results were expressed as percent of Annexin V+ cells. Synergy was quantified using the cooperativity index (cooperativity index\\x8a=\\x8asum of apoptosis of single-agent treatment/apoptosis on combined treatment). Cooperativity index <1, 1, or >1 is termed synergistic, additive, or antagonistic, respectively. Experiments were performed 3 independent times in triplicate. Data presented are from one representative experiment. Diagnostic blast samples were cultured in RPMI 1640/20% dialyzed fetal bovine serum (Life Technologies, Carlsbad, CA) supplemented with ITS solution (Sigma-Aldrich) and 20% supernatant of the 5637 bladder cancer cell line (as a source of granulocyte-macrophage colony-stimulating factor). The cells were plated at a density of 1106 cells/mL and cultured in the presence of 040 nM panobinostat for 48 h in a 37°C humidified atmosphere containing 5% CO2/95% air. THP-1 and U937 cell lines were purchased from the American Type Culture Collection (Manassas, VA). The OCI-AML3 cell line was purchased from the German Collection of Microorganisms and Cell Cultures (DSMZ, Braunschweig, Germany). The CTS cell line was a gift from Dr. A Fuse from the National Institute of Infectious Diseases, Tokyo, Japan. The cell lines were cultured in RPMI 1640 (or MEM for OCI-AML3 cells) media with 10% fetal bovine serum (Hyclone, Logan, UT) and 2 mM L-glutamine, plus 100 U/ml penicillin and 100 µg/ml streptomycin, in a 37°C humidified atmosphere containing 5% CO2/95% air. Cytarabine and DNR were purchased from Sigma-Aldrich (St Louis, MO). Panobinostat was purchased from Selleck Chemicals (Houston, TX). Clinically, panobinostat steady-state plasma concentrations range from 15 to 22 nM over 48 h (data from Novartis Investigators Brochure). Diagnostic bone marrow samples (n\\x8a=\\x8a9, Table S1) from children with de novo AML were obtained from the Childrens Hospital of Michigan leukemia cell bank. Mononuclear cells were purified by standard Ficoll-Hypaque density centrifugation. Written informed consent was provided by the parent or legal guardian according to the Declaration of Helsinki. Sample handling and data analysis protocols were approved by the Human Investigation Committee of the Wayne State University School of Medicine. Noninvasive imaging was performed twice weekly in all experiments to monitor tumor engraftment, as previously described. The Luciferase substrate D-luciferin firefly potassium salt (Caliper, Hopkinton, MA) at a dose of 150 mg/kg was administered by intraperitoneal injection. The mice were then anesthetized by 1.5%2.5% isoflurane inhalation, and bioluminescence was done 5 minutes later using a Xenogen in vivo imaging system (Hopkinton, MA) in the Animal Imaging Facility at St. Jude Childrens Research Hospital. Total bioluminescence was quantified for the body area that included each mouse in its entirety. Mice were monitored daily and were sacrificed by CO2 asphyxiation when they showed signs of terminal illness, including hind limb paralysis, inability to eat or drink, and/or moribund. All animal experiments were approved by the Institutional Animal Care and Use Committee of St. Jude Childrens Research Hospital.'\n","Skipping entity [1713, 1733, MedTech] in the following text because the character span 'SYBR Premix Ex Taq¢' does not align with token boundaries:\n","'All data were analyzed using SPSS 16.0 software (IBM, Chicago, IL, USA). Independent t tests were used to compare plasma IFNG levels and intracellular differences between patients and healthy controls. The data are summarized as meanSD. Methylation differences in the IFNG promoter between different groups were assessed using a Pearson Chi-square test. P values less than 0.05 were considered statistically significant. The most prominent PCR band was separated on 2% agarose gel electrophoresis, and purified by gel extraction (Qiagen), treated with TA cloning using DNA-Tailing (TaKaRa), and sequenced on an ABI 3730 (Applied Biosystems). All operations were performed by GeneScript Corporation (a sino-America joint venture, Nanjing, China). Sequences were analyzed by bioinformatics software Chromas Version 1.45 (Technelysium, South Brisbane, Australia). Methylation analysis was conducted using bisulfite sequencing. Genomic DNA of CD4+ T cells was isolated using QIAamp Mini Kit (Qiagen) as recommended by the manufacturer. The DNA was bisulfite converted using CpGenome\\x84¢ DNA modification Kit (Millipore Corporation, Billerica, MA, USA) according to the manufactures instructions. The primers used to amplify the IFNG promoter were: IFNG-forward, 5-TGT GAA TGA AGA GTT AAT ATT TTA TTA-3; IFNG-reverse, 5-TTG GTA GTA ATA GTT AAG AGA ATT TA-3. PCR products were generated using the bisulfite-treated DNA as template. RNA isolation was performed using miRNeasy Mini Kit (Qiagen, Germantown, MD, USA) and reverse transcribed using a PrimeScript RT Reagent Kit (TaKaRa, Tokyo, Japan) according to the manufacturers protocol. cDNA was analyzed for the expression of target genes and quantified by RT-PCR using SYBR Premix Ex Taq\\x84¢.Quantitative PCR was performed on an ABI 7500 real-time PCR system (Applied Biosystems, Foster City, CA, USA). The PCR primers used were: IFN--forward, 5-GCA GGT CAT TCA GAT GTA GCG G-3; IFN--reverse, 5-TGT CTT CCT TGA TGG TCT CCA CAC-3; DNMT1-forward, 5-GAT CGA ATT CAT GCC GGC GCG TAC CGC CCC AG-3; DNMT1-reverse, 5-ATG GTG GTT TGC CTG GTG C-3; DNMT3b-forward, 5-CCT GCT GAA TTA CTC ACG CCC C-3; DNMT3b-reverse, 5-GTC TGT GTA GTG CAC AGG AAA GCC-3; -actin-forward, 5-TGG CAC CCA GCA CAA TGA A-3; -actin-reverse, 5-CTA AGT CAT AGT CCG CCT AGA AGC A-3. Ct values were normalized to expression in CD4+ T cells using the 2Ct method and -actin as the housekeeping gene. Experiments were done in triplicate, and results from three independent experiments were averaged. PBMCs were stimulated with PMA (10 ng/mL). Brefeldin A (Sigma-Aldrich, St. Louis, MO, USA) was added at a final concentration of 10 µg/ml and incubated 4 h. PBMCs samples were taken from responding cultures and stained with a combination of fluorochrome-conjugated mAbs consisting of anti-CD8-PE-CY5 and anti-CD3-FITC (BD PharMingen, San Diego, CA, USA). After fixation and permeabilization (Fix Perm, Caltag, Buckingham, UK), cells were stained with anti-IFNG-PE and isotypic mAbs (BD PharMingen). Isotype-matched control antibodies were used to determine the level of background staining for the mAb mixture. All events were acquired using a FACSCalibur flow cytometer (BD Biosciences, San Jose, CA, USA). Data were analyzed with CELL Quest software (BD PharMingen). IFNG levels in culture supernatants and plasma were determined by ELISA (R&D Systems, Minneapolis, MN, USA), according to the manufacturers protocol. Transwell experiments were performed in 24 well plates with pore size 0.4 µm (Corning Costar, Corning, NY, USA). SPC-A1 cells (2105) were cultured in the outer wells of 24 well plates in 1640-medium supplemented with 10% human AB serum. CD4+ T cells (6105) separated from healthy adults were added into the inner wells in the same medium. As shown in Fig. 3, control groups were established with CD4+ T cells growing in the inner wells without SPC-A1 cells. After 5 days of culture, CD4+ T cells were washed, and 1106 cells were collected for DNA extraction, while the remaining was transferred to 96-well culture plates. 5104 cells/well were stimulated with 1 µg/ml soluble anti-CD3 (eBioscience, San Diego, CA, USA) and 1 µg/ml soluble anti-CD28 (eBioscience) (in a total volume of 200 µl) for 6 or 24 h. After stimulating, supernatants were collected for IFNG detection by ELISA, CD4+ T cells were collected for IFNG mRNA analysis. The human NSCLC cell line SPC-A1 was purchased from the Chinese Academy of Sciences in Shanghai, and cultured in 5% CO2 in RPMI 1640 with 2 mmol/L L-glutamine, 10% fetal bovine serum (Invitrogen, Carlsbad, CA, USA), penicillin G (50 U/mL), and streptomycin (50 µg/mL). Venous blood (10 ml) with EDTA-K2 was collected from healthy controls and lung cancer patients before operation. Plasma was separated and stored at 70°C prior to measuring IFNG. Peripheral blood mononuclear cells (PBMCs) were separated by Ficoll-Hypaque density gradient centrifugation (GE Health Care Life Sciences, Piscataway, NJ, USA). CD4+ T cells were isolated from PBMCs using a CD4-positive isolation kit (Dynal, Oslo, Norway). The purity of isolated CD4+ T cells was determined by flow cytometry using PE-conjugated anti-CD4 (Beckman, Marseille, France). CD4+ T cells were then collected for DNA extraction, bisulfite modification, and sequencing. Lung cancer patients (N\\x8a=\\x8a30) and healthy adults (N\\x8a=\\x8a30) were from the first affiliated hospital of Nanjing Medical University (Nanjing, China). Lung cancer patients were diagnosed as carcinoma by pathologic diagnosis. Patients who received preoperative chemotherapy, radiotherapy or operation before collecting blood sample had been excluded. The detailed clinical data (including age, gender, smoking history) were obtained from each objects medical records and listed in  Table 3 . Among lung cancer patients, there were 24 cases of adenocarcinoma, 2 cases of squamous carcinoma, 2 cases of alveolar cell carcinoma, and 2 cases of poorly differentiated carcinoma. This study was approved by the Committee on the Ethics of Treatment of Human Subjects of the First Affiliated Hospital of Nanjing Medical University (Permit Number: 20A6-2869), and a written informed consent was also obtained from each participant.'\n","Skipping entity [14712, 14716, MedTech] in the following text because the character span 'DMEM' does not align with token boundaries:\n","\"T-tests were employed in this study for statistical analysis (using Prism software). * indicates 0.01<p0.05; ** indicates p0.01. The data set was exported from GeXP software after normalization to kanamycin. All multiplexes were combined into one Excel data file. The normalized intensity of each replicate was used to calculate an average intensity for each group (i.e. transduced cells and non-transduced cells). The sample groups were compared to each other using a two-tailed T-test, with statistical significance set at p<0.05 (n\\x8a=\\x8a4). Normalized values were displayed in a bar graph using Excel. Transduced primary hMDM were analyzed at day 6 post-transduction. The expression of 24 reference genes was analyzed using GenomeLab GeXP Genetic Analysis System (Beckman Coulter, Brea, CA). RNA was extracted from hMDM using Qiagen RNeasy Kit (cat# 74104). First, cells were scraped and pelleted through centrifugation at 4,000 RPM for 5 min. The supernatant was discarded and the cell pellets were resuspended in resuspension buffer. The cells were lysed with lysis buffer and neutralized with neutralization buffer. Cell debris was removed and RNA was captured using an RNeasy spin column, washed twice with wash buffer, and then eluted with RNase free molecular grade water. RNA concentration was determined via spectrophotometer at 260 nm. Human Reference RNA (Beckman Coulter, A54267) was used to develop a standard curve for a human reference multiplex. XP-PCR was performed on sample RNA with reverse transcriptase minus and the test included no template reactions as negative controls and reference RNA as a positive control. KanR RNA (Beckman Coulter, #A85017) was added to each sample as an internal control and two genes were designated as reference genes. XP-PCR was conducted in two stages: the first was the RT reaction in which sample RNA was incubated with DNase/RNase free water, 5 RT buffer, reverse transcriptase, and the reverse primer plex for 1 min at 48°C, 60 min at 42°C, and 5 min at 95°C. And next, PCR was performed where cDNA from the RT reaction was incubated with 5 PCR buffer, 25 mM MgCl2, DNA polymerase, and the forward primer plex for 10 min at 95°C, and 35 cycles of 30 s at 94°C, 30 s at 55°C, and 1 min at 70°C. Finally, amplicons were combined with DNA size standard-400 and sample loading solution (Beckman Coulter, #608098 and #608082), then subjected to capillary electrophoresis. The results were analyzed by Express Profiler and Quant Tool (Beckman Coulter)and normalized to the internal control KanR to eliminate any inter-capillary differences and then normalized a second time to the pre-selected reference genes. Finally, results were compared to the standard curve created for the specific multiplex and normalized values and significance differences between samples were calculated using T-tests. SH-SY5Y cells (ATCC, Manassas, VA) were pretreated with conditioned media (110 & 120 dilutions) from LV-hBDNF-transduced HTB-11 cells for 1 h, then exposed to HIV-1 Tat protein (200 ng/mL) for 48 hours, and cell viability was determined by MTT assay as described above. T-tests were used to assess any significant differences in cell viability between cells receiving conditioned media from transduced or non-transduced cells. Similarly, HTB-10 cells were exposed to TNF- at 300 ng/mL, 150 ng/mL, and 75 ng/mL, diluted with culture medium or conditioned culture medium (110 dilution) collected from LV-hBDNF-transduced cells. The cell viability in normal culture media or conditioned culture media without TNF- is normalized to be 100%. Additionally, U937 cells were exposed to 16 ng/mL, 8 ng/mL, 4 ng/mL, 2 ng/mL and 1 ng/mL TNF-, which was diluted with culture medium or conditioned culture media from LV-hBDNF-transduced cells. The cell viability from HTB-10 cells not exposed to TNF-, but normal medium or conditioned medium (110 dilution of collected medium from LV-hBDNF-transduced cells) alone was considered 100%. After 4-hour incubation, MTT assay was performed to determine cell viability with results read at 570 nm using a microplate reader (Beckman Coulter AD340). A neuroprotection assay was conducted to assess the degree of that hBDNF in neutralizedthe neurotoxic effects of TNF- (R & D, USA). LV-hBDNF-HTB-10 cells, mock cells, and non-transduced HTB-10 cells were all exposed to TNF- at three different concentrations: 500 ng/mL, 100 ng/mL, and 20 ng/mL, respectively. After 48 hours, MTT assay was performed to determine cell viability by absorbance reading at 570 nm using a microplate reader (Beckman Coulter AD340). T-tests were used to assess any significant differences in cell viability between transduced and non-transduced cells as well as cells receiving conditioned media from transduced or non-transduced cells. Expression of eGFP was evaluated by immunochemical staining using selected monoclonal antibodies conjugated with TRITC. Briefly, cells in 48-well plates were washed twice with DPBS, then blocked by DPBS containing 0.5% (w/v) bovine serum albumin (BSA), and then directly stained with the appropriately diluted primary antibody - goat anti-green fluorescent protein (GFP) antibody (Rockland, USA). After incubation for 2 hours in a dark room, the cells were washed with DPBS. The secondary antibody used was Donkey anti-Goat IgG with TRITC (Rockland, USA).Cells were visually examined and documented using an inverted fluorescence microscope (Nikon Eclipse TE2000-U) with a digital camera attachment. Cells at the exponential growth phase were harvested using trypsin-versene solution (Sigma) and seeded into 96-well plates at 2104 cells/well in 100\\x85µL. Following overnight incubation and the formation of a cell monolayer, test cells were treated with Tumor necrosis factor-1 alpha (TNF-) at selected concentrations. Seventy-two hours later, 20\\x85µL of MTT solution (5 mg/mL) was added to the 100\\x85µL of medium in each well, and the plate incubated at selected temperatures for 4 hour. The solution was removed followed by adding 100\\x85µL/well of DMSO to solubilize the purple formazan crystals produced. Absorbance in each well was measured at 570 nm using a 96-well plate reader (Beckman Coulter, Fullerton, CA). The supernatant or lysate of transduced- or non-transduced- cells including HTB-10, HTB-11, and CHME-5 cells, was mixed with 6 sodium dodecyl sulfate (SDS) sample buffer (100 mM Tris-HCl at pH 6.8, 200 mM dithiothreitol, 4% SDS, 0.2% bromophenol blue, and 20% glycerol) and loaded on 5% stacking/15% separating SDS-polyacrylamide gels. Following electrophoresis at 30 mA for 1 hour, separated proteins were transferred onto a nitrocellulose membrane (NCM) (Invitrogen, Carlsbad, CA). The membranes were saturated with 1% bovine serum albumin (BSA) (Sigma-Aldrich) in TBST buffer containing 10 mM Tris- HCl at pH 8.0, 150 mM NaCl, and 0.05% Tween-20 for 1 hour at room temperature, followed by overnight incubation with diluted rabbit polyclonal antibodies to human hBDNF (Antigenix America, USA) at 4°C. Following extensive washing with TBST, the NCM was incubated with diluted biotin-conjugated rabbit polyclonal antibodies to human hBDNF (Antigenix America, USA) at room temperature for 1 hour, then washed three times with TBST and exposed to a 3,3-diaminobenzidine tetrahydrochloride (DAB) substrate (PIERCE, Rockford, IL) for identification of protein bands. A 96-well plate was coated with rabbit polyclonal antibodies to human hBDNF (Antigenix America, USA) and incubated overnight at 4°C. The plate was washed three times with 0.05% Tween-20 in PBS and blocked with 1% BSA (Sigma-Aldrich) for 1 hour at room temperature on an orbital shaker. After washing three times with PBS, the plate was incubated with diluted hBDNF containing supernatant samples for 1 hour and then incubated with biotin-conjugated rabbit polyclonal antibodies to human hBDNF (Antigenix America, USA) for 1 hour. The plate was then washed and incubated with streptavidin-horseradish peroxidase (Rockland) for 1 hour at room temperature. The presence of human hBDNF protein was detected with one-Step Ultra TMB (tetramethylbenzidine) (Pierce). The enzymatic reaction was stopped via the addition of 1 M sulfuric acid. The quantitation of hBDNF protein was based on optical density values at 450 nm with 570 nm as the reference wavelength, and compared with a standard curve of purified human hBDNF protein (Antigenix America, USA), using an ELISA reader (Beckman Coulter AD340). hMDM were seeded to 12.5 cm2 tissue culture flasks at a density of 1.8107 cells per flask. Six days later, transductions were performed using MOI\\x8a=\\x8a50 by thawing the titrated virus stocks at room temperature, mixing the appropriate volume of virus concentrate with 8 mg/mL Polybrene (Sigma), and adding the mixture to the target cells together with RPMI to achieve a total volume of 1 mL per flask. mMDM were seeded to 12.5 cm2 tissue culture flasks at a density of 1.0107 cells per flask. Four days later, transductions were performed using MOI\\x8a=\\x8a50 by thawing the titrated virus stocks at room temperature, mixing the appropriate volume of virus concentrate with 8 mg/mL Polybrene (Sigma), and adding the mixture to the target cells together with RPMI to achieve a total volume of 1 mL per flask. After 2-hour incubation at 37°C, an additional 3 mL of complete RPMI was added per flask. Most of the culture medium was aspirated and replaced by fresh RPMI in the day following transduction. The transduced cells were examined daily using visual inspection by inverted light and fluorescence microscopy. Briefly, 5105 HTB-10, HTB-11 (neuronal) or CHME-5 (microglial) cells were subcultured in 25 cm2 tissue culture flasks for 24 hours before transduction. For transduction, cell culture media were removed and cells were washed twice with DPBS followed by adding 0.5 mL vector suspension (1107 IU/mL, MOI\\x8a=\\x8a10, for HTB-11, MOI\\x8a=\\x8a1) containing 8\\x85µg/mL polybrene, and affected cells were incubated at 37°C for 2 hours. Vector suspension was aspirated and transduced cells were added to 4 mL/flask of fresh growth medium and incubated at 37°C with 5% CO2. The medium was replaced 24 hours post-infection and transduction efficiencies were evaluated on day 3 post-infection. The percentage of GFP positive cells was determined by calculating the number of GFP positive cells and total cells from randomly selected microscopic fields under a fluorescence microscope (Nikon Eclipse TE2000-U). A total of 5 microscopic fields, each containing at least 100 cells, were counted for each transduction test. For murine myeloid monocyte-derived macrophages (mMDM), the method was conducted according to Dou et al. In brief, BALB/c mice (Frederick National Laboratory of the National Cancer Institute) were euthanized via CO2 chamber and femurs of the mice were removed. After removing the muscle, bone marrow cells were flushed with DPBS from the bone shafts into single-cell suspensions, and cultured for 5 days supplemented with 1000 U/mL M-CSF (Wyeth). Cultured mMDM proved to be >98% CD11b+ by immunofluorescent staining with PE-conjugated rat monoclonal antibodies directed against mouse CD11b (Caltag Laboratories, CA, USA). To verify the purity of the attached cells, these monolayer cultures were stained with human CD11b monoclonal antibody conjugated with PE (Caltag Laboratories, CA, USA). Briefly, the antibody was diluted 100 times with 1% BSA (Sigma) in DPBS and hMDM cultures, and at day 7 were stained with the diluted antibody. After 12 hours incubation, stained cells were rinsed twice with DPBS, and then examined under an inverted fluorescence microscope. Blood was drawn from consented healthy donors into BD Vacutainer\\x84¢ ACD (Beckton Dickinson), and peripheral blood mononuclear cells (PBMC) were isolated by density-gradient centrifugation through Ficoll-Paque\\x84¢ PLUS (GE Healthcare). The purified PBMC were then resuspended in RPMI-1640 medium supplemented with 10% heat-inactivated human serum (Sigma- Aldrich), 20% heat-inactivated FBS, 100 IU/mL penicillin, 100\\x85µg/mL streptomycin sulfate and 0.292 mg/mL L-glutamine. The purified PBMC were then seeded into 25 cm2 tissue culture flasks (Corning) at 1.8107 cells and incubated at 37°C with 5% CO2 to allow the attachment of freshly isolated monocytes. Following a 3 hours incubation, non-adherent cells were removed by aspiration and the remaining adherent cells were washed extensively with Dulbecco's phosphate-buffered saline (DPBS) (Sigma), prior to the addition of 4 mL RPMI-1640 growth medium per flask, and subsequent cultivation at 37°C. Cell growth and monolayer formation were observed daily using a phase-contrast inverted microscope (Nikon). Generation of a HIV-1-based lentiviral vector containing an expression cassette for hBDNF was constructed. Briefly, the synthetic gene was amplified by PCR, using primer pair (5-GCCTCGAGCTAATGACCATCCTTTTCCTTAC TA-3 and 5-GCGGATCCGACCTATCTTCCCCTTTTAATGGT-3) containing Xho I and BamH I restriction sites within the 5 and 3 termini, respectively, and inserted into the pDHR-HB7-IRES-GFP plasmid (generously provided by Dr. V. Planelles, University of Utah) that was digested with the same enzymes. The final bicistronic plasmid construct, pDHR-hBDNF-eGFP, was designed to co-express the hBDNF protein and the enhanced green fluorescent protein (eGFP). LVs encoding the hBDNF or control gene were generated by transient transfection of 293T cells. Vector production, concentration, and titration were performed as described previously, except that 293T cells were used for vector titration. Briefly, 293T cells were incubated with 10-fold dilutions of concentrated virus and 8 mg/mL Polybrene (Sigma-Aldrich) for 1 hour. Fluorescence-positive cell counting to determine eGFP-positive cells was carried out at 72 hours post-transduction, and vector titers were calculated as follows: titer\\x8a=\\x8aFC 0/VD (where D is the virus dilution factor, V is the number of 293T cells at the time of seeding per inoculum, F is the frequency of eGFP-positive 293T cells, and C 0 is the number of 293T cells at the time of seeding per mL). 1. Blood was anonymously collected from unidentified health donors with approval from the University of Hawaii Committee on Human Studies (UHCHS); Healthy blood donors signed a written consent form outlining the procedure for this study; both the consent form and procedures were approved by the UHCHS. 2. Laboratory mice were used for experimental tests with approval from the University of Hawaii Animal Care and Use Committee in accordance with the Animal Welfare Act and NIH guidelines. Human embryonic kidney (HEK) 293T cells (GenHunter Co., Nashville, TN) were maintained in Dulbecco's Modified Eagle's Medium (DMEM)(Sigma-Aldrich) containing 1.0 g/L glucose, 2 mM L-glutamine (Sigma-Aldrich), 100 IU/mL penicillin (Sigma-Aldrich), 0.1\\x85µg/mL streptomycin (Sigma-Aldrich) and 10% fetal bovine serum (FBS) (HyClone). Three human neuronal cell lines, neuroepithelioma cells (HTB-10, or SK-N-MC), neuroblastoma cells (HTB-11 or SK-N-SH, and SH-SY5Y), and glioblastoma cells (HTB-14; or U-87) (ATCC, Manassas, VA), were cultured in Minimum Essential Medium (Eagle) (MEM) (Sigma-Aldrich) supplemented with 2 mM L-glutamine, 1.0 mM sodium pyruvate, 100 IU/mL penicillin, 0.1\\x85µg/mL streptomycin and 10% FBS. The human embryonic microglial cell line CHME-5 (provided by Dr. Pierre Talbot, Universite du Quebec) was cultured in Dulbecco's Modified Eagle's Medium (DMEM) (Sigma-Aldrich) containing 4.5 g/L glucose, 2 mM L-glutamine, 100 IU/mL penicillin, 0.1\\x85µg/mL streptomycin and 10% FBS. Human monocytic cell line U937 (ATCC Manassas, VA), was cultured in RPMI1640 (Sigma- Aldrich) supplemented with 2 mM L-glutamine, 1.0 mM sodium pyruvate, 100 IU/mL penicillin, 0.1\\x85µg/mL streptomycin and 10% FBS.\"\n","Skipping entity [7609, 7613, MedTech] in the following text because the character span 'DMSO' does not align with token boundaries:\n","'For the clinical trial data, a strategy for analysis of data was developed a priori, including our decision to stratify the population based on SCI severity at admission. Baseline characteristics were compared among the 3 treatment groups to determine whether there was any evidence of differences among groups. Categorical variables were compared using chi-squared analysis and continuous or ordinal variables were compared using Kruskal-Wallis tests. The primary outcome for the trial was the TSCIS score on day 42. The TSCIS on day 3 was considered a secondary outcome. The association of TSCIS with treatment group and other individual variables was assessed using generalized linear modeling. Individual variables significantly associated with TSCIS were analyzed using multivariable generalized linear modeling using maximum likelihood estimating methods. Multiple comparisons among groups were adjusted using the method of Sidak. Model fit was assessed graphically using diagnostic plots of residuals. Comparisons of proportions among treatments groups (e.g., frequency of adverse events) were made using chi-squared or, when appropriate, Fishers exact tests. Significance was set at P<0.05 for all analyses. Analyses were performed using S-PLUS statistical software (Version 8.2, TIBCO, Inc., Seattle, WA). Activities of MMP-2/MMP-9 in CSF and serum were compared between healthy control dogs and the dogs with SCI using the Wilcoxon rank-sum test. The Wilcoxon rank-sum test was also used to compare CSF and serum MMP-2/MMP-9 activities between dogs with and without selected characteristics that were potential modifiers of MMP-2/MMP-9 (i.e., age, breed, sex, and markers of disease duration or severity). To compare serum MMP-2/MMP-9 values among treatment groups, the serum MMP-2/MMP-9 activities were converted to ranks, and the ranks were compared using a generalized linear model; multiple pair-wise comparisons between treatments were made using the method of Sidak. Model fit was assessed graphically using diagnostic plots of residuals. Noncompartmental pharmacokinetic analysis was performed (Phoenix WinNonLin 6.3, Pharsight, St. Louis, MO), and estimates of the parameters of Tmax, Cmax, T1/2, and area AUC0-obs and AUC0-\\x9e were calculated for the single dose. To assess ability of GM6001 to inhibit MMP-9, activity (described above) was determined using human recombinant MMP-9 (Sigma, St. Louis, MO) that was serially diluted in DMSO to final concentrations of 0.01 µM to 200 µM. Controls consisted of enzyme and substrate only and substrate and GM6001 only. Activity of MMP-2 and MMP-9 in serum and CSF samples was assessed in a blinded manner using a previously developed electrophoretic method that included a synthetic peptide (AAPPtec, Louisville, KY) (sequence: Ac-NGDPVGLTAGAGK-NH2), tagged with a fluorophore BODIPY-FL-SE (Invitrogen, Carlsbad, CA). The substrate was mixed with either serum or CSF, with phosphate buffered saline as the negative control. After reacting for 1 hour, aliquots were loaded onto 20% polyacrylamide gels and the samples were electrophoresed. Gels were imaged using a BioDoc-It M-26 transilluminator (UVP, Upland, CA, USA). The image was scanned in a Storm 840 workstation (Molecular Dynamics, Sunnyvale, CA, USA) with ImageQuant v5.2 software and fluorescent signal was quantified using ImageJ (1.440, National Institutes of Health, Bethesda, MD). CSF and serum samples (n\\x8a=\\x8a16/treatment group) were randomly selected at the end of the trial by computerized sorting on random numbers. Purpose-bred Beagle dogs (n\\x8a=\\x8a5) were sampled as controls. Serum samples with overt hemolysis were excluded from analysis. Vertebral column MRI was performed on enrolled dogs, except in cases where animals were evaluated outside of normal operating hours or the scanner was unavailable due to mechanical failure. Between September 2008 and July 2011, a 1.0 T system (Siemens Magnetom, Malvern, PA) was utilized to acquire images; for the remainder of the trial images were generated using a 3.0 T MRI (Simens Verio, Malvern, PA). Dogs that received MRI had sagittal T2-weighted (T2W) images reviewed by 1 investigator (JML) using commercially available software (eFILM, Merge Healthcare, Chicago, IL) prior to un-blinding. The acquisition parameters for sagittal T2W images generated at 1T and 3T included a repetition time of 3500 ms, echo time of 90 ms, and slice thickness of 2.0 mm. The presence of spinal cord T2W hyperintensity was determined by visually comparing injured parenchyma to surrounding spinal cord. This technique has been used extensively in human and canine MRI studies and produces repeatable results that correlate with behavioral measures of SCI severity and recovery. The Texas Spinal Cord Injury Score (TSCIS) was used to assess pelvic limb gait, posture and nociception. This is a more refined scale than the MFS with a larger array of sub-categories, including gait assessment that parallels the Basso, Beattie, Bresnahan Scale. The TSCIS gait score ranges from 0 to 6 in each pelvic limb and correlates to the degree of limb protraction and weight bearing. The gait classifications include: no voluntary movement seen when the dog is supported (score\\x8a=\\x8a0); intact limb protraction with no ground clearance (1); intact limb protraction with inconsistent ground clearance (2); intact protraction with ground clearance >75% of steps (3); ambulatory with consistent ground clearance and significant paresis-ataxia that results in occasional falling (4); ambulatory with consistent ground clearance and mild paresis-ataxia that does not result in falling (5); and normal gait (6). Pelvic limb postural responses using the TSCIS were scored in each limb as absent (0), delayed (1, correction occurred >1 second after positioning), and present (2). Nociception was scored in each limb as absent (0), deep nociception only present (1), or both deep and superficial nociception present (2). A modified Frankel scale (MFS) was developed to broadly parallel the American Spinal Cord Injury Association Impairment Scale (AIS). Dogs were scored as paraplegic with absent deep nociception (0; equivalent to AIS A), paraplegic with absent superficial nociception (1; equivalent to AIS B), paraplegic with intact nociception (2; equivalent to AIS B), or non-ambulatory with identifiable pelvic limb movement (3; equivalent to AIS C). The MFS was not a primary trial outcome, but instead was used to describe the baseline population (overall and by treatment group) and to stratify the study population for analysis. Clinicians responsible for neurologic scoring were blinded to treatment assignments. Two ordinal SCI scores were used to address injury severity at study entry, day 3 post-treatment, and day 42 post-treatment. In both scoring systems, dogs were considered ambulatory if they could spontaneously rise, bear weight, and take at least 10 steps without falling. Dogs that were non-ambulatory had pelvic limb movement evaluated using tail support. Postural responses were evaluated by placing the dorsum of the pes on a non-slick surface while manually supporting the animal and waiting for limb correction. Pelvic limb deep and superficial nociception were evaluated by applying hemostats to a nail-bed or interdigital webbing, respectively and evaluating for the presence of a behavioral or physiological response. Immediately after collection of CSF and blood, dogs were randomized to receive 100 mg/kg GM6001+ DMSO, DMSO, or saline placebo. The dose of both DMSO and saline was 0.4 mL/kg, a volume equivalent to that of 100 mg/kg GM6001+DMSO; this approach was taken to maintain blinding. A randomization sequence was developed prior to the initiation of this trial and randomization was accomplished by blocking the dogs by gender status in a 111 ratio to each of the treatment groups. Sealed envelopes contained treatment allocations and were delivered to a central location where treatments were formulated by individuals not involved in the assessment of animals. Treatments were covered and marked only with animal identifiers to ensure blinding. Dogs, enrolled in the clinical trial, had physical and neurological examinations, complete blood count and serum biochemistry profile. Anesthesia was induced with propofol (Rapinovet, Schering-Plough Animal Health Corp, Union, NJ) and maintained with inhalant sevoflurane (SevoFlo, Abbott Laboratories, North Chicago, IL). Diagnostic imaging consisting of myelography, computed tomography (CT), or MRI was performed to identify IVDH. Cerebrospinal fluid (CSF) was collected from the cisterna magna for routine analysis and a 200-µL aliquot was stored at 80¿C for determination of MMP-2/MMP-9 activity. Six mL of whole blood were obtained at the time of CSF collection and 3 days following treatment delivery; serum was isolated and frozen at 80¿C. For PK assessments, a single 100 mg/kg SC administration of GM6001 was delivered in 5 dogs with 5 additional dogs receiving a second 100 mg/kg SC of GM6001, 12 hours following the first dose. In dogs with single dosing, serial plasma samples were obtained at 5, 15, and 30 minutes and 1, 2, 3, 6, 12, 24, 36, 48, and 96 hours after GM6001 delivery. In dogs with multiple dosing, blood samples were collected shortly after the second dose, and then at 24, 48, 72, and 96 hours. All samples were stored in a 80¿C freezer until analyzed by high performance chromatography (Thermo Electron Co., Waltham, MA) and tandem mass spectroscopy (MDS-Sciex/Applied Biosystems API3000, Concord, ONT) (LC-MS/MS). Concentrations of GM6001 (m/z 389.0\\x86356.0) were determined, using MMP Inhibitor III (m/z 364.0\\x86356.0, Calbiochem, Billerica, MA) as the internal standard. A standard curve was created with blank dog plasma at concentrations 10.0 to 10,204.0 ng/mL, with linear regression and weighting of concentrations (1/x2). After thawing and addition of internal standard (300 µL of 100 ng/mL in 0.5% acetic acid in methanol), plasma samples or standards (100 µL) were centrifuged and reconstituted with 30/70 methanol/10 mM ammonium formate buffer, pH 3.0, for protein precipitation. Supernatant (100 µL) was collected, vortexed, and refrigerated until injection on LC-MS/MS. The mobile phase consisted of 0.1% formic acid in deionized water (A) and acetonitrile/methanol/formic acid (4060:0.1, v/v/v) (B) with a flow rate of 0.30 mL/minute using a linear gradient starting at 40% B from 0 to 0.01 minutes, to 80% B at 1.5 minutes, to 90% B at 3.5 minutes, to 40% B at 3.6 minutes, with a total run time of 5 minutes. Dogs, participating in the drug tolerance study, were acclimatized for 14 days and then randomized as follows: DMSO (at a volume equivalent to that present in a 100 mg/kg GM6001 treatment), 100 mg/kg GM6001, 150 mg/kg GM6001, or 300 mg/kg GM6001 subcutaneously (SC) every 12 hours for 3 days. The doses of GM6001 were selected to exceed those reported previously in a murine model of SCI. A SC route of administration was selected as 1) GM6001 does not remain solubilized in DMSO when exposed to hydrophilic solutions such as blood, prohibiting intravenous delivery and 2) intraperitoneal drug administration is not generally permitted in client-owned dogs at our institution, due to challenges in managing any local drug reactions. Adverse event monitoring was performed for 7 days following the completion of drug administration. All dogs had physical examinations, injection site evaluations, and assessment of food and water intake twice daily. A complete blood count, serum biochemistry profile, urinalysis, and coagulation profile were performed 3 and 7 days following the completion of drug administration. Following the completion of this study, the vehicle and 300 mg/kg GM6001 dogs were euthanized via intravenous administration of 120 mg/kg pentobarbital (Fatal Plus, Vortech Pharmaceuticals, Dearborn, MI). The brain, heart, liver, kidney, lung, intestine, and injection sites were evaluated. For all canine studies, GM6001 (SAI Advantium, Hyderabad, India) was dissolved in 90% DMSO (Domoso, Fort Dodge Corp, Fort Dodge, IA) at a concentration of 250 mg/mL. The solution was sterilized using a 25-mm syringe filter with 0.22-µm HT Tuffryn membrane (Pall Corporation, East Hills, NY). All animal procedures were approved by the Texas A&M University Institutional Animal Care and Use Committee (AUP 2007115; AUP 2011057; AUP-2011145) and in the case of client-owned dogs were performed with signed consent. All studies adhered to the National Institutes of Health Guide for the Care and Use of Laboratory Animals. The primary outcome of the clinical trial was a validated ordinal SCI score (the Texas Spinal Cord Injury Score [TSCIS]) conducted at 42 days post-injury. The secondary outcome was TSCIS at 3 days after SCI. Dogs were stratified into those with behaviorally severe SCI (absent pelvic limb movement and deep nociception) and those with mild-to-moderate SCI (intact pelvic limb deep nociception with or without movement) at study entry to examine primary and secondary outcomes. This a priori stratification was utilized because a substantially lower proportion of dogs with severe SCI recover independent ambulation at long-term follow-up time-points (approximately 5060%) in comparison to dogs with mild-to-moderate SCI (approximately 85 to 95%); thus, injury severity might influence the ability to detect treatment-related effects. Guidelines for the conduct of SCI trials developed by the International Campaign for Cures of Spinal Cord Injury Paralysis were utilized to assist with the design of a randomized, double-blinded (clinicians and clients were unaware of treatment group), placebo-controlled canine trial including inclusion/exclusion criteria, randomization protocol, data handling, and the a priori definition of outcome metrics and statistical approaches.. Consolidated Standards of Reporting Trials (CONSORT) Statement Guidelines were used to assist with trial performance and data reporting. Client-owned dogs with IVDH-associated SCI, admitted to the Texas A&M University Veterinary Medical Teaching Hospital between September 2008 and February 2012, were recruited. The study interval was selected to generate a sample size of >100 dogs, which was considered robust based on previous human phase II and III SCI studies, animal model studies of SCI using MMP blockers, and completed canine SCI studies. A formal power calculation was not performed due to the absence of a phase I canine study examining the effects of GM6001. A preliminary drug tolerance study was constructed based on Food and Drug Administration guidelines (http://www.fda.gov/AnimalVeterinary/default.htm) and performed in 4 healthy, purpose-bred Beagles. Ten healthy, purpose-bred Beagles were obtained to evaluate pharmacokinetics (PK); this sample size was based on similar animal studies and general recommendations for canine PK investigations. Following surgical decompression, all dogs were recovered in an intensive care unit for 24 hours and during that time were provided post-operative opioid analgesia and bladder evacuation. Physical rehabilitation protocols were standardized for dogs participating in this study. Dogs received thoracic limb and pelvic limb passive range of motion exercises beginning 24 hours post-operatively and until dogs could independently ambulate. Each limb was gently flexed and extended at the carpal, elbow, and hip joints in 3 sets of 10 repetitions, 2 times daily. Supported standing exercises were performed twice daily for 5 minutes by placing a sling immediately cranial to the pelvic limbs and continued until dogs could independently ambulate. Dogs that were non-ambulatory were walked using a sling placed immediately cranial to the pelvic limbs for 5 minutes twice daily. Independently ambulatory dogs were permitted to walk on a leash for 5 minutes 34 times per day during hospitalization and were allowed to continue this activity until 42-day re-check. Participating dogs were housed in cages that permitted limited additional activity until 42-day re-check evaluation. Dogs had to meet the following criteria to be included in the clinical trial population: 1) duration of SCI was required to be 48 hours; 2) IVDH-associated SCI had to result in non-ambulatory paraparesis or paraplegia at enrollment; 3) IVDH-associated SCI had to be identified between the T8-L6 vertebral articulations and treated via surgical decompression. The exclusion criteria were: 1) concurrent disseminated neoplasia or systemic inflammation; 2) a history of recent breeding/pregnancy; and, 3) glucocorticoid treatment within 7 days of SCI.'\n","Skipping entity [249, 253, MedTech] in the following text because the character span 'DMSO' does not align with token boundaries:\n","\"HIV RNA-induction experiments were conducted using 5 to 6 replicates for every experimental condition, including vehicle-treated controls. Fold induction in HIV RNA levels was calculated as a ratio of mean signals from compound-treated and vehicle (DMSO)-treated control samples from the same donor under identical conditions. Student's t-test (one-tailed distribution, two-sample equal variance) was used to assess statistical significance of the difference between the detected HIV RNA copies in vehicle control- and HDACi-treated samples or between RMD- and VOR-treated samples. Time course experiments testing the latency reversal agents had matching vehicle-treated controls for each time point. For experiments using blood-derived resting CD4 T cells, culture supernatants from 3 replicate wells were combined and analyzed using the COBAS system. Primary results in absolute HIV RNA copy numbers from all experiments presented in this study are provided in Supplementary material (Supplementary Table S4). Single-genome sequencing of a portion of HIV-1 gag-pro-pol amplified from cell culture supernatants or cellular DNA extracts was performed as previously described. Sequences were aligned using ClustalW and neighbor-joining phylogenetic analysis was performed using MEGA5. Trees were rooted on the subtype B consensus sequence (www.hiv.lanl.gov). Resting CD4 T cells obtained from HIV-infected patients on suppressive cART were incubated with serial dilutions of HDACi in 96-well plates at 10,000 cells/well for indicated time and then lysed by repeated freezing and thawing. Total cellular HDAC activity was measured using the HDAC-Glo I/II assay kit (Promega) according to the manufacturer's protocol. For cell activation analysis, either complete PBMC cultures or resting CD4+ T cells isolated from HIV-infected cART-suppressed patients were incubated with the tested agents under specified conditions and were stained with relevant antibodies. All antibodies were purchased from BD Biosciences and included Alexa Fluor 700-labeled antibody to CD4, APC-H7-labeled antibody to CD8, PE-Cy7 -labeled antibody to CD19, PE-labeled antibody to CD69, APC-labeled antibody to CD25, and V450-labeled antibody to HLA-DR. Live cells were gated by forward and side scatter and exclusion of the dead cells by Live/Dead Fixable Aqua Stain (Invitrogen). Marker staining was assessed by flow cytometry analysis on a LSR Fortessa with data processing using FlowJo software. To assess HIV activation by HDACi, up to 5 million CD4 T cells were plated in 24-well plates in 2.5 ml of media, supplemented with antiretrovirals (ARVs) (100 nM elvitegravir and 100300 nM efavirenz) for the entire duration of culture incubation. HDACi were added at specified concentrations on day 0 and cells or culture supernatants were harvested for the analysis of HIV RNA at the indicated time points (648 hours or 67 days, respectively). For HDACi pulse treatment, cultures were washed twice using ARV-containing media at 4 or 24 hours after addition of HDACi and incubated with ARVs until harvest. To measure HIV RNA levels, 1 ml of culture supernatant was analyzed by a robotic COBAS AmpliPrep/TaqMan system (Roche Diagnostics, Indianapolis, IN), which extracts total nucleic acid and quantifies HIV RNA in copies per milliliter using the HIV-1 Test, v2.0 kit (Roche Diagnostics). For the measurement of cell-associated HIV RNA levels, cells were washed with PBS, counted, and lysed (Qiagen RLT buffer, 400 µl for every 2 million cells; Qiagen, Venlo, Netherlands). Lysates were filtered through a Qiashredder (Qiagen). Total RNA was prepared from the lysates (400 µl) using a robotic system (QIAsymphony, Qiagen) that incorporates a DNase I digestion step to eliminate cellular DNA. The resulting total RNA was eluted with 200 µl of buffer, diluted to 1 ml with nuclease-free water, and analyzed by COBAS using the HIV-1 Test v2.0 kit. Preparations of both cell-associated RNA and supernatant total nucleic acids were tested for potential contamination with HIV DNA and/or host DNA by performing the PCR amplification in the presence and absence of reverse transcriptase and by a detection of GAPDH-encoding host DNA sequence. These methods confirmed that there was no contaminating HIV DNA in either the intracellular RNA or supernatant total nucleic acid preparations (Supplementary Figs. S3 and S4). HIV-infected patients participating in the study were selected based on sustained plasma viral load suppression (<50 copies/ml for >12 months), CD4 counts (>350 cells/µL), and absence of co-infection with hepatitis B or C virus (Supplementary Table S1). Clinical laboratory results were reconfirmed 2 weeks before leukapheresis or blood draw. Leukapheresis was conducted for 34 hours, and samples were processed within 2 hours after collection. The leukapheresis product was diluted 11 with PBS and layered over Ficoll for isolation of PBMCs. PBMCs were treated with red blood cell lysis buffer (eBioscience) and rested overnight (10 million cells/ml) in tissue culture medium (RPMI 1640 supplemented with 10% FBS and PenStrep) before the isolation of memory CD4 T cells according to the manufacturer's recommendation (EasySep Human Memory CD4 T cell Enrichment Kit). Resting CD4 T cells were isolated by first enriching total CD4 T cells (StemCell Technologies) and subsequently depleting HLA-DR-, CD25-, and CD69-positive cells via negative selection (Miltenyi Biotec, Auburn, CA). Flow cytometry was used to assess the purity of both T-cell subsets (>98%). Resting CD4 T cells for the analysis of continuous RMD exposure and longitudinal responses to RMD in the same donors were purified from fresh blood using the same protocol. Determination of HDACi potency against individual HDAC enzymes was performed at Reaction Biology Corporation (Malvern, PA). In brief, human HDAC-1 to -11 enzymes were incubated with serial dilutions of HDACi and fluorogenic peptide substrates (50 µM) in a 96-well format assay. Concentrations of HDACi reducing the control activity of tested enzymes by 50% (IC50) were determined by a five-parameter curve fit of collected fluorescent signals. Fluorogenic peptide derived from p53 protein (amino acid residues 379382; RHKKAc) was used as a substrate for HDAC enzymes 1, 2, 3, 6, 10, and 11. A different fluorogenic peptide from p53 (amino acid residues 379382, RHKACKAC) was used as a substrate for HDAC-8. Fluorogenic peptide Boc-Lys(trifluoroacetyl)-AMC was used as a substrate for HDAC enzymes 4, 5, 7, and 9. Seven days post-infection, cells infected with NL4-3-Luc virus were stimulated for 48 hours with test compounds. Cells were stained with fixable viability dye v450 (eBioscience, San Diego, CA) in phosphate buffered saline (PBS) +2% FBS for 30 minutes at 4°C. Cells were fixed with IC Fixation Buffer (eBioscience) for 20 minutes at room temperature in the dark followed by treatment with a permeabilization buffer (eBioscience) for 30 minutes at 4°C. Intracellular p24 was stained with anti-p24 antibody clone KC57-RD1 (Beckman Coulter, Fullerton, CA), then washed and resuspended in PBS +2% FBS. The cells were analyzed on a LSR Fortessa (BD Biosciences, San Jose, CA) and collected data were processed using a FlowJo Analysis Software (TreeStar, Ashland, OR). Total peripheral blood mononuclear cells (PBMCs) were obtained from healthy HIV-negative donors by leukapheresis (AllCells, Inc, Emeryville, CA). Naive CD4+ T cells were purified by negative selection using EasySep magnetic beads (StemCells, Inc, Vancouver, Canada) and cultured in RPMI with 10% fetal bovine serum (FBS), penicillin/streptomycin, 1% nonessential amino acids (Life Technologies, Carlsbad, CA), 1% sodium pyruvate (Life Technologies) and 495 nM beta-mercaptoethanol (Sigma Aldrich, St. Louis, MO) in a 37°C, 5% CO2 incubator. Purified naive CD4+ T cells were activated by incubation with anti-CD3/CD28 magnetic Dynabeads (1 bead: 2 cells ratio, Life Technologies), 1 µg/ml anti-IL-4 (R&D Systems, Minneapolis, MN), 2 µg/ml anti-IL-12p70 antibodies (R&D Systems), and 10 ng/ml TGF- (R&D Systems) for 3 days. Following the removal of anti-CD3/CD28 beads and antibodies, cells were maintained in 30 U/ml IL-2 (Life Technologies) for 2 days. Cells were then infected with NL4.3-Luc in the presence of 50 µg/ml DEAE for 3 hours. Cells were maintained in the continued presence of 30 U/ml IL-2 throughout the infection and subsequent rest period with culture medium with fresh IL-2 replaced every 23 days. Seven days post-infection, 20 µl of latently infected cells were dispensed into 384 well plates using a MicroFlo dispenser (Biotek Insturments, VT) at 10,000 cells/well containing 100 nl of compound solutions delivered by the Echo acoustic-based liquid dispenser (Labcyte, Sunnyvale, CA). After a 48-hour incubation, 16 µl/well BriteGlo (Promega, Madison, WI) was added and luminescence measured using the Envision plate reader (Perkin Elmer, Waltham, MA). Compound-associated cytotoxicity was determined in latently infected cells in parallel with the virus activation assay. Cells were incubated with compounds for 48 hours at 37°C, and cell viability was determined using Cell Titer Glo reagent (Promega). Plasmid pKS13 is a NL4-3-based vector in which the vpr and env genes were inactivated by inserting a \\x98T base at the AflII site and two bases (TT) at the NdeI site, respectively, causing frame shifts in both open reading frames. A codon-optimized firefly luciferase gene was introduced in place of the nef gene by replacing the BamHI and NcoI fragment, yielding the plasmid pKS13. NL4-3-Luc virus was generated by co-transfection of HEK-293T cells with pKS13 and a plasmid containing the HIV-1 env gene using Lipofectamine 2000 (Life Technologies, Grand Island, NY). Romidepsin was obtained from a US-based pharmacy as a marketed commercial formulation of the drug. Vorinostat, panobinostat, givinostat, mocetinostat and SB939 were obtained from Selleck Chemicals (Houston, TX). Phorbol 12-myristate 13-acetate (PMA) and ionomycin were obtained from Sigma Aldrich (St. Louis, MO). All compounds were dissolved in DMSO. Effects of HDAC inhibitors in all assays and models used in this study have been compared to vehicle (i.e. DMSO) treated controls. HIV-infected patients were enrolled into the study at the Quest Clinical Research (QCR; San Francisco, CA) and the University of Pittsburgh Medical Center (UPMC; Pittsburgh, PA). The QCR and the UPMC part of the study were approved by the Western Institutional Review Board and the University of Pittsburgh Institutional Review Board, respectively. In both cases, written, informed consent was obtained from the patients prior to any study procedures.\"\n","Skipping entity [4140, 4143, MedTech] in the following text because the character span 'PBS' does not align with token boundaries:\n","\"Quantile normalized expression data of CD271k.d. cells and shCtl. cells were analysed with the GSEA analyses tool of the Broad institute (http://www.broadinstitute.org/gsea/index.jsp) and the molecular signature databases c2.all.v4.0.symbols.gmt (curated gene sets) and c3.all.v4.0.symbols.gmt (motif gene sets). Whole genome expression profiling data of SOX10k.d. cells stored at NCBI GEO, accession number GSE37059 were processed via the R/Bioconductor environment using statistical analysis tools mentioned above. Affymetrix chip data were processed via package affy and normalized with Robust Multi-array Average (RMA) from the same package. Package limma was used for making significance tests for differential gene expression. The results of these tests were adjusted for multiple testing via package q-value. Data were normalized with Robust Multi-array Average (RMA). Genes up-regulated (ratio >1.33) and down-regulated (ratio <0.75) with an average detection p-value <0.05 were analysed. Intersection sets were found via gene symbol annotations. Significantly differentially expressed genes were identified by statistical tests using the bioconductor limma package. The significance in terms of the false discovery rate (FDR), the q-value, was determined by the q-value package applied to several statistical tests (Student's t-test, Welch test and Wilcoxon test). Genes were defined significantly differentially expressed when the average Affymetrix (MAS5) detection p-value of at least one case (treatment or control) was <0.05 and one of the p- values of statistical tests (Student's t-test, Welch test and Wilcoxon test) was <0.05. Genes with ratios >1.33 or <0.75 were defined as significantly up-regulated or down-regulates, respectively. Heat maps were generated using Spearman's rank correlation as similarity measure and complete linkage. Genes with a limma false discovery rate (FDR) below (0.005) were defined as highly significantly regulated. Using this criteria we found 123 genes comprising 68 down-regulated (ratio <0.05) and 55 up-regulated genes (ratio >20). P-values depicting significance of experimental values were calculated by Excel using a two-tailed test. The microarray analysis data have been deposited to the NCBI Gene Expression Omnibus (GEO) with accession number GSE52456. Whole genome expression profiling of CD271k.d. and shCtl. cells was performed with three biological replicates. Illumina raw data of BeadChip HumanHT-12V4 platform were summarized via the BeadStudio without normalization and background correction. Follow-up processing was done via the R/Bioconductor environment. The lumi package was used for data annotation. Following statistical tests, data were quantile normalized. For knock-down of CD271, melanoma cells were transfected with 2 µg of each shRNA plasmid (#2: 5-ACAACCTCATCCCTGTCTATT-3; #3: 5-CCCGAGCACATAGA CTCCTTT-3 and #4: 5-CCGAGCACATAGACTCCTTTA-3) or control shRNA (shCtl., 5-GGAATCTCATTCGATGCATAC-3; all from Qiagen) using Lipofectamine2000 (Invitrogen). Cells were selected with puromycin (10 µg/ml) over a period of two weeks. For over-expression melanoma cells were stably transfected with a plasmid expressing GFP-tagged human CD271 (RG207966, OriGene) and selected with G418 (300 µg/ml, PAA). FFPE material was sectioned to 2 µm after de-waxing, peroxidase blocking and antigen-retrieval using citrate buffer, pH 6.0. IHC for melanoma markers Melanosome (clone HMB45, mouse, 150, Dako); Tyrosinase (clone T311, mouse, 150, Novocastra); MITF (clones C5+D5, mouse, 125, Zytomed) and CD271 (clone D4B3, XP, rabbit, 1100, Cell signaling), as well as hematoxylin/eosin (H&E) was done with the Discovery XT biomarker platform (Ventana). For IF on tumor sections: sections were blocked in 5% goat serum at RT for 2 h and incubated with primary antibodies: human specific CD133 (clone W6B3C1, mouse, 110, Miltenyi); CD133 (polyclonal, rabbit 1100, Biorbyt); CD271 (clone D4B3-XP, rabbit, 1100, Cell signaling) and MART-1 (Melan-A clone A103/M2-7C10/M2-9E3, 150, Zymed) diluted in 2% BSA/PBS. Incubation was done in a wet chamber overnight at 4°C. Sections washed with PBS/0.1% Tween-20 were incubated with secondary antibodies AlexaFluor488, anti-rabbit and AlexaFluor594, anti-mouse (both Invitrogen) for 1 h, at RT. IF on cultured cells: after washing with PBS and a fixation step (4% PFA in PBS for 10 min at RT), cells were either permeabilized with 0.1% Triton-X100 for staining of nuclear proteins or directly blocked with 2% BSA in PBS. Fixed cells were incubated with primary antibodies (CD271-PE and CD271-APC, Miltenyi, clone ME20.4-1.H4; SOX10, clone 20B7, RnD; CD133 Miltenyi, clone W6B3C1; H2AX (phosphorylated (Ser139) H2AX, DNA-damage marker) and pSmad 2 clone 138D4, Cell signaling) overnight at 4°C. Next day, cells were washed and incubated with secondary antibodies AlexaFluor488 and AlexaFluor594, anti-rabbit/anti-mouse (both Invitrogen) and 4,6-Diamidino-2-phenylindole dihydrochloride (DAPI), (Sigma-Aldrich) for 1 h, at RT, washed and analyzed. All animal experiments were performed at the EPO GmbH, Berlin, Germany in accordance with the German Animal Protection Law and approved by the local responsible authorities (LAGeSo A0452/08). The in vivo procedures were consistent and in compliance with the UKCCCR guidelines. Briefly, a number of 11051106 cells was mixed with Matrigel (11) and subcutaneously injected in NSG mice (n\\x8a=\\x8a3/per group). Tumor volume given as [(tumor width)2  tumor length] 0.5 was measured with a caliper instrument and was monitored during the entire experiment with the measurements of two perpendicular tumor diameters. RNA isolation from frozen cell pellets was performed with the RNeasy Mini Kit (Qiagen, Germany) and from 5 µm FFPE sections using the RNeasy FFPE Kit (Qiagen), following the manufacturers protocol. Reverse transcription of 500 ng2.5 µg RNA was performed with SuperScript VILO cDNA synthesis kit (Invitrogen, Germany) and diluted to a final volume of 50 µl. qRT-PCR was carried out on a Step one plus (Applied Biosystems, Germany) for 3040 cycles. Primers were designed for 5560°C annealing temperatures (Table S1). In qPCR expression levels were calculated either with CT or CT, normalized to -actin. PCR products (15 µl) were separated on a 2% agarose (BioRad) gel. 25 µg40 µg of total protein lysates were separated on a 8% SDS-PAGE gel and transferred on a nitrocellulose membrane by using the iBlot Dry Blotting System (Invitrogen). Membranes were incubated with primary antibodies (CD271, clone D4B3, XP, Cell signaling; SOX10, clone 20B7, RnD; CD133, clone C24B9 and -Tubulin, clone 9F3 both from Cell Signaling Technology, Germany; all diluted 11000) overnight at 4°C and with a secondary antibody (goat anti-rabbit IgG LI-COR Biosciences, USA) for 1 h at RT. Detection was done with the Odyssey Infrared Imaging System (LI-COR Biosciences, USA). Labeling of cells using the fluorescent lipophilic dye PKH26 was done with accordance to the manufacturer's instructions. Labeled cells were re-plated and monitored. Analysis of co-labeling of PKH26 (PE), CD271 (APC) and CD133 (FITC) was done with Accuri C6. Analysis of labeled and sorted cells of each cell strain was performed in at least three independent experiments. Cells were harvested using Accutase, pelleted at 1000 rpm and re-suspended in MACS buffer. 2.5105 cells were incubated either with human specific CD271-PE or CD271-APC antibody (Miltenyi, clone ME20.4-1.H4, mouse IgG1, 120), human specific CD133 antibody (Miltenyi, clone W6B3C1, mouse IgG1, 110,) or mouse IgG1 Isotype control (Miltenyi) at 4°C for 10 min. Washed cells were re-suspended in 500 µl PBS and stored on ice or labeled with a secondary antibody. Cells were incubated in 100 µl buffer with AlexaFluor488, anti-mouse (Invitrogen).Stained cells were used for cell sorting or flow cytometry using FACSAria III or FACScalibur and Accuri C6 (Beckton&Dickinson), respectively. Data analysis was done with FlowJo software. MACS separation into CD133+ and CD133 fractions was done with accordance to the manufacturer's instructions using the indirect CD133 MicroBead kit and LS columns (Miltenyi, Germany). Briefly, 80% confluent cells were washed with PBS and detached with Accutase (PAA, Austria) and adjusted up to 4107 cells per column and sorted with a modified PBS buffer containing 5% FBS (MACS buffer). All primary low passage melanoma cell strains and A375 cells were cultured in Quantum 263 medium (PAA, Austria) supplemented with 1% Penicillin/Streptomycin (Invitrogen) while MeWo cells were cultured in Leibovitz's L-15 medium (Invitrogen) supplemented with 1% Penicillin-Streptomycin. Cells were cultured in 5% CO2, 37°C and routinely passaged at 80% confluence. Medium was changed every 3 days. Human neonatal epidermal Melanocytes were cultured in Melanocyte growth medium (both Lonza). Phenotype shift experiments were done in hESC-medium with and without 4 ng/ml of FGF-2 (Biosource) or in Quantum 263 medium supplemented for 3 days with 10 ng/ml TGF2 (Miltenyi). Written informed consent was obtained from each patient. Patient's material was collected between 20102012 at the Comprehensive Cancer Center of the Charit© University Hospital (CCCC) with coverage of the Tumor REsearch And Treatment  20 Patient Pilot (TREAT 20) project approved by the Federal Ministry of Education and Research (No. 031 5852A). Tumor specimens were obtained at the time of surgery and spitted into equal pieces for paraffin-embedding and establishment of tumor-derived cell cultures. We received material sufficient for both procedures from 16 out of 20 patients.\"\n","Skipping entity [2200, 2226, MedTech] in the following text because the character span 'rabbit polyclonal antibody' does not align with token boundaries:\n","'ShWNT5B- or shCtl- virus transduced MDA-MB-231 cells were distributed into 12-well plates the day before transfection. Cells at 80% confluence were co-transfected with TCF-driven Topflash reporter plasmid (Millipore) (1\\xa0g) and control Renilla luciferease (20\\xa0ng) using 2.5\\xa0l of Lipofectamine 2000 (Invitrogen). Cells were lysed in 1X passive lysis buffer in 48\\xa0h and the supernatant was collected for Dual-luciferase activity measurement (Promega, Madison, WI). For each sample, firefly luciferase activity was normalized with an internal control, Renilla luciferase activity. To measure soluble WNT5B in patients serum, we used WNT5B ELISA Kit (USCN Life Science Inc.). The manufactures protocol was totally followed for preparing samples and all the reactions. The plate was read by SpectramaxPlus (Molecular Devices). The XF24 flux analyzer (Seahorse Bioscience) was used to measure OCR in 24-well microplates. Six thousand cells transduced with shCtl and 12000 cells infected with shWNT5B lenti-virus were seeded onto 24-well plates and incubated 3\\xa0days. The measurement, recording procedure and data analysis were described previously. For cellular ATP measurement, we used ENLITEN ATP Assay System Bioluminescence Detection Kit (Promega Madison, WI). Briefly, the adherent cells in 6-well plate were collected by 2\\xa0mM EDTA in PBS on ice, TCA was add at final concentration of 1% and vortex vigorously for 10\\xa0sec. It was further diluted to 0.1% TCA by Tris-Acetate (pH=7.75). The standard as well as the samples were serially diluted by dilution buffer (0.1% TCA/0.08PBS/0.9Tris-Acetate) and subjected to luminescence measurement. The cells infected with shWNT5B or shCtl were collected in 3\\xa0days. The electron microscope was done in the core facility at COH following their standard protocol. It has been described in detail elsewhere. The stained sections were subjected to Electron microscopy, which was done on an FEI Tecnai 12 transmission electron microscope equipped with a Gatan Ultrascan 2\\xa0K CCD camera. All the formalin-fixed paraffin-embedded (FFPE) slides were prepared and stained by the Pathology Core Facility at COH using a standard protocol. Antibodies used in this study were: rabbit polyclonal antibodyWNT5B (SAB2900204) (1:20 dilution, Sigma-Aldrich), mouse monoclonal antibody Myc (SC-40) (1:75 dilution, Santa Cruz Biotechnology) and rabbit monoclonal antibody MCL1 (ab32087) (1: 100 dilution, Abcam). All antibodies were titrated with negative and positive controls to obtain optimal staining. Cells were trypsinized, resuspend in fresh medium followed by flowcytometry analysis. For cell cycle assay, cells were fixed with 70% ethanol and incubated on ice for 30\\xa0min. The cells were then suspended in PBS and treated with RNase A (final concentration 100ug/ml) at 37°C for 30\\xa0min. After removing RNase A, the cells were stained with propidium iodide (PI) at 5ug/ml for 10\\xa0min and the cell cycle was determined by flowcytometry analysis. For apoptosis assay, FITC Annexin V Apoptosis Detection Kit was used for staining the cells following products manual (BD bioscience). All flowcytometry data were analyzed using Summit v4.3 software. Cells were infected with shCtl or shWNT5B lentivirus and the morphology was observed and photographed after WNT5B expression was inhibited. Cell mobility was determined by a wound closure assay. Cells were placed onto 6-well plates at 80% confluence and cultured in serum-depleted media for 40\\xa0h. A wound was made by scraping the monolayer cells with a plastic pipette tip and fresh serum-free medium was replenished. Images of wound closure were photographed at 0, 16, 24 and 40\\xa0h post-scraping. The triple negative cell lines MDA-MB-231was purchased from ATCC and cultured in the recommended media. Specific lentivirus shRNA (Sigma-Aldrich) was used to disrupt the expression of WNT5B while shRNA targeting non-mammalian sequence (Sigma-Aldrich) served as control. WNT5B expression was determined by immunoblot analysis. MDA-MB-231 cells that expressed WNT5B or control shRNA (shWNT5B or shCtl cells) were cultured in growth medium to observe cell growth. Cells (3  104/ml) were seeded into 24-well plates, and cell number was counted every day for five days using a Cellometer Auto T4 (Nexcelom Bioscience, Lawrence, MA). Independent experiments were performed in triplicate. Cell protein was extracted from cells using RIPA buffer (Cell Signaling, Danvers, MA) with phosphatase inhibitor (Roche, Indianapolis, IN). Equal amount of protein was loaded and separated by SDS-PAGE. After the protein was transferred onto a membrane, the blot was blocked with 5% non-fat milk in TBS and probed overnight at 4°C using the following antibodies: WNT5B (Sigma-Aldrich, St. Louis, MO), Cyclin E (M20), TOM20 (F-10), Myc (9E10), AIF (E-1), MCL1 (S-19) (Santa Cruz Biotechnology, Santa Cruz, CA), Caspsae-3, Caspase-8,PGC-, Cyclin D1 (Cell Signaling, Danvers, MA) and -actin (Bio-Rad, Hercules, CA). Appropriate antibodies were used for secondary antibody reaction. Signal was detected by the ECL Plus Western Blot Detecting System (GE Healthcare, Piscataway, NJ). Total RNA extraction from MDA-MB-231 was carried out using the RNeasy Mini Kit (Qiagen). For cDNA synthesis, total RNA (1\\xa0g) was transcribed using random hexamers (Invitrogen, Carlsbad, CA), and SuperScript III reverse transcriptase (Invitrogen) following the manufacturers protocol. For quantification of OXPHOS-related genes, the cDNA amplication program included a denature at 95°C for 3\\xa0min, followed by 40\\xa0cycles of 95°C for 10\\xa0s; 58°C for 30\\xa0s. For MtDNA detection, total cellular DNA was isolated with DNAeasy Blood and Tissue Kit (Qiagen). Mitochondrial DNA content was determined by qPCR by using comparing the mitochondrially encoded Cox2 gene to an intron of the nuclear-encoded -globin (HBB) gene. All qPCR was performed using an iQ5 iCycler (Bio-Rad) according to the manufacturers instructions. Data were analyzed using Bio-Rad iQ5 Optical System Software v2.0. All products yielded a single band with the predicted size. All primers are listed in Additional file 1: Table S1 and all products yielded a single band with the predicted size. For differential expression analysis, differential expression P-values were determined via t-test in R. Significant results are expected to show P-value<0.05. Differential expression between TNBC and non-TNBC was determined using data from 3 cohorts (Chin et al., expO (GSE2109), and TCGA. Differential expression between patients that did or did not develop metastatic tumors was determined using 2 cohorts (expO (GSE2109), TCGA for WNT5B and 1 cohort (Desmedt et al.) for MCL1. For survival analysis, differences in survival between \\x9chigh\\x9d and \\x9clow\\x9d expression groups were visualized in Kaplan-Meier plots and compared using Cox regression analysis, with P-values calculated via log-rank test, using the \\x98survival package in R (A Language and Environment for Statistical Computing). The disease-free survival of WNT5B was quantified independently for 2 cohorts (Desmedt et al. and Wang et al.) respectively. And then meta-analysis was conducted by utilizing the same WNT5B probe (so that the signals would be comparable) for an 80\\xa0month observation period. The disease-free survival of MCL1 was analyzed by the same method using the cohort of Desmedt et al. All tumor tissues and serum were collected under the Institutional Review Board (IRB)-approved protocols at City of Hope National Medical Center (COH) or Zhejiang University respectively. The patients were given informed consent. One hundred and forty-two breast tumor tissues, including 21 TNBC and 121 Non-TNBC tissues were collected for immunohistochemistry staining. We also collected 30 sera from TNBC and Non-TNBC patients respectively with the assistance from the COH Translational Research Laboratory Core for ELISA assay. Immunohistochemical staining and FISH confirmed that ER/PR/HER2 were negatively expressed, as assessed by pathologists in the Department of Pathology of COH.'\n","Skipping entity [4943, 4945, MedTech] in the following text because the character span 'ER' does not align with token boundaries:\n","'Samples were evaluated for intrinsic subtype using the PAM50 single-sample classification algorithm developed by Parker and Perou. Of the 50 genes in the PAM50 classifier, 21 were identified in the combined, quality filtered dataset (7685 probes), and used for analysis. The 21 genes available for this analysis were BCL2, CDC6, NUF2, CENPF, CEP55, CXXC5, EGFR, ERBB2, ESR1, FGFR4, FOXC1, GRB7, NDC80, KRT14, MYBL2, PTTG1, RRM2, TMEM45B, TYMS, UBE2C, and UBE2T. The remaining 29 genes in the PAM50 were either not present on the cDNA array probe set or did not meet the quality control criteria described above. We used a 90 % confidence interval (CI) in the subtype call to establish that gene expression levels of the 21 genes were correlated to the centroid of the subtype (otherwise the sample was considered unclassified). To assess differential risk of recurrence (ROR ~ P) scores across sample types, we used ANOVA followed by Tukey analysis. To identify genes with differential expression in sorted CTC samples relative to primary tumors, normal epithelium, and PB samples, we used ANOVA followed by Tukey analysis with Benjamini-Hochberg (BH) multiple testing correction applied at each step to adjust for potential false discovery (FDR). Genes with a FDR adjusted p value < 0.05 were considered significant. Functional and pathway analysis of differentially expressed genes were carried out with the DAVID functional/pathway enrichment analysis tool, using multiple testing corrected p-values that control for false discovery with significance threshold p < 0.05. To characterize differential expression between different cell types, we performed principal component analysis (PCA) and hierarchical clustering using the stats package in R. To verify that unsupervised clustering can reliably classify specimens, the p-values for each cluster were calculated via bootstrap resampling using standard Bootstrap Probability (BP) and a multi-scale bootstrapping method that is a better approximation to Approximately Unbiased (AU) (R package pvclust). Only probes of the array which exhibited a hybridization signal 2 fold above background in at least 10 % of specimens and with none of the spots flagged as bad were used for PCA, clustering, and all other analyses (7685 probes). Two types of normalization were used. First, each set of arrays was normalized separately. Background correction was not performed as this increased noise in low signal features. Instead, a 2D spatial normalization approach using the robust local regression function \\x98loess was performed to correct for uneven hybridization across individual arrays using the marray software package in the R. To achieve consistency between arrays, quantile normalization across samples was performed using the limma package in R. The three sets of arrays were merged into a single dataset by including those features consistent to all array sets. This resulted in 21,006 total genes for the combined dataset of 85 samples. To test for residual batch effects after normalization, we calculated the pairwise correlation matrix for all samples to verify that clustering patterns were driven by cell type rather than cDNA array design and processing batch. cDNA microarray data included three different array formats, containing 21,600 (set 1), 21,632 (set 2), and 41,664 (set 3) features, respectively, representing in total 85 samples (5 sorted CTC, 7 plated BT474, 6 sorted BT474, 8 PB, 4 normal epithelium, and 55 primary breast tumors). Microarray data were deposited under GSE45965. The 20,862 human cDNAs used in these studies were purchased from Research Genetics (Huntsville, AL), now Invitrogen, and were provided by the Haqq laboratory as cDNA microarrays. On the basis of Unigene build 166, these clones represent 19,740 independent loci. Hybridization, washing, scanning, and primary data analysis were performed as previously described. Slides were then washed, scanned, and analyzed with Axon Imager 4000b (Molecular Devices, Sunnyvale, CA), using GENEPIXPRO3.0. BT474 and StratRef total RNA were converted to cDNA using M-MLV reverse transcriptase and random hexamers (Life Technologies). Samples were then incubated at 25 °C for 10 min then 48 °C for 30 min. Expression levels of 37 genes and a housekeeping gene GUS (beta-glucuronidase) were analyzed using a 5 nuclease assay and TaqMan Gene Expression Assays with an ABI PRISM 7700 instrument (Applied Biosystems (ABI), Foster City, CA). Relative expression levels were calculated relative to GUS. Calculated QPCR expression ratios were derived using the formula relative expression = 2( CT). Log-transformed (base 2) values for each gene expression ratio were plotted for both microarray and QPCR methods. We have previously published details regarding the selection strategy and gene list. Subsequent validation of our assay performance was performed by QPCR of spiked, sorted BT474 and PB for EpCAM, ER, HER2, CD45, HLA, and GAPDH using TaqMan assays. All sorted samples were stored at 80 °C as cell lysates until the time of RNA isolation with PicoPure RNA isolation kits (Life Technologies). For normal PB, RNA was extracted using the Qiagen Blood RNA kit (Qiagen). Total RNA and equivalent amounts of StratRef (Stratagene Universal Human Pooled Reference RNA, Stratagene, La Jolla, CA) were treated with 200 ng of poly dIdC (Sigma-Aldrich, St. Louis, MO). CTC, BT474, PB, and StratRef samples were linearly amplified with 2 rounds using Arcturus RiboAmpHS (Life Technologies). Primary tumor, normal epithelium, and StratRef were amplified using 2 round modified T7 amplification. Concentrations of amplified RNA products were measured using a UV spectrophotometer. The molecular weight and integrity of amplified RNA species were evaluated using the Agilent Bioanalyzer 2100 (Agilent Technologies, Palo Alto, CA). IE/FACS was performed as previously described but with an emphasis on the preservation of RNA to be isolated directly from CTCs. As we previously described in Magbanua et al., our IE/FACS assays involve immunomagnetic separation using EpCAM (MJ37) mAb-coated magnetic beads followed by FACS with EpCAM (EBA-1) mAb conjugated to phycoerythrin (PE), thioflavin nucleic acid dye, and CD45 (2D1) mAb conjugated to Cy5-PerCP (all from BD Biosciences, San Jose CA). A threshold of a single cell meeting these criteria was qualified as a positive test result. We prepared the FACS Aria II (BD Biosciences) with RNAse Zap decontamination solution (Ambion, Austin, TX) prior to all sorting. All samples were processed immediately following blood draws, and all lysates were immediately placed on ice. All CTC subjects specimens were analyzed with a FACS Aria processed using consistent gates. Ten to 20 mL of PB was drawn into EDTA tubes from five female MBC patients yielding 9-993 captured CTCs from which expression profiles were obtained; specimens were processed immediately by IE/FACS. Microarray data for primary tumors prior to neoadjuvant chemotherapy were obtained from the I-SPY1 trial (CALGB 150012, ACRIN 6657). Normal breast (n = 1) and skin samples (n = 3) were collected as core and punch biopsies. Negative control PB samples were obtained from healthy individuals. All patients gave informed consent under a protocol approved by the University of California San Francisco Institutional Review Board. All patients received standard of care therapy for MBC; all patients had been previously treated with several courses of chemotherapy, and samples were obtained at convenient time points in patient care during routine outpatient laboratory PB draws. We selected patients known to be CTC positive for this pilot study based on prior positive CTC test results. The BT474 cell line (from the Gray Lab) was used for proof of principle experiments. Cells were grown in Roswell Park Memorial Institute (RPMI) medium supplemented with 10 % FBS and 1 % antibiotics/antimycotics (Invitrogen, Carlsbad, CA). BT474 cells were trypsinized for 5 min and counted using a hemocytometer. Cell count was confirmed with a FACS Calibur using our previously described multi-marker FACS assay. 1000100,000 cells were spiked into 10 mL PB obtained from healthy females; all specimens were processed immediately, and over-exposure to trypsin was avoided by quenching with medium.'\n","Skipping entity [2231, 2242, MedTech] in the following text because the character span 'reagent mix' does not align with token boundaries:\n","\"Statistical analysis was performed using 2 commercially available software packages18 , 19. Normal distribution was assessed by inspection of histograms or D'Agostino and Pearson omnibus normality test. Subsequently, either parametric or non\\x90parametric tests were performed: A repeated measures model was used to assess the effect of the treatment interval (before versus after 6\\xa0weeks of treatment) and treatment type (EF versus placebo). Kruskal\\x90Wallis or t\\x90tests were performed to compare clinical and clinicopathological data, endoscopy and histology scores on visit 1 between treatment groups. Significance was set at P\\xa0<\\xa00.05. Clinical severity (as assessed by CCECAI) was selected as the primary outcome measure. Calculations were based on a type I error of 0.05 and a type II error of 0.1, which corresponds to a power of 90%. When assuming a difference (delta, ) of 1.5 times the sd (Ï\\x83) to be medically relevant, the necessary number of animals in each group (treatment versus placebo) was determined to be n\\xa0=\\xa011. To facilitate visualization of gene expression data and changes in gene expression among visits of the clinical trial, heatmaps were created using a statistical package17. Expression of the so\\x90called trefoil factors (TFF), genes associated with epithelial barrier function and repair,33 also was assessed. The nuclear receptor PPAR, usually responsible for blocking the transcription of inflammatory cytokines34 and supposedly especially up\\x90regulated in intestinal epithelial cells by enterococci,35 was investigated additionally. Several genes of the innate immune system interacting with bacteria (Toll\\x90like receptors26, 27 [TLRs] 2,4,5, and 9 as well as components of the inflammasome,28 NLRP3, casp\\x901, interleukin [IL]\\x901 and IL\\x9018) and the adaptive immune response in the intestine were investigated. For the latter, several signature cytokines and transcription factors of different T helper lymphocyte (Th) cell lines were assessed. These included Th17 cells (IL\\x9017A, IL\\x9022, IL\\x9023p40, RORC),29, 30 Th1 cells (IL\\x9012p35, IL\\x902, TNF), Th2 cells (IFN and IL\\x904,31 and Tregs (IL\\x9010 and TGF).32 qPCR reactions were carried out in triplicate at a total reaction volume of 20\\xa0L, using a commercially available reagent mix16 and a primer concentration of 200\\xa0nM. Cycling steps were: 95°C for 5\\xa0min, then 40 cycles of 94°C for 15\\xa0s, 55°C for 10\\xa0s, 72°C for 10\\xa0s and 80°C for 5\\xa0s. Reaction efficiency was determined using 10\\x90fold dilutions (107 molecules L1 to 101 molecules L1) of plasmids containing the cloned amplicon. Melting curves were generated to ensure a single amplicon had been produced. Averaged gene expression data for each sample were normalized against the geometric mean of the reference genes.25 Canine sequences for the genes of interest and reference genes were cloned as described before.23 Primers were designed based on publically available gene sequences (Genbank, ENSEMBL) using the Primer3 software (see Table\\xa02).24 A total of 16 genes were assessed for stability using suitable software10 with a commercially available canine reference gene array11. Based on these data, four reference genes were selected for canine duodenum and five for the colon, respectively (Table\\xa02). Primers were optimized using standard PCR12. Amplicons were visualized on a 1.5% agarose gel, purified13 and cloned into a holding vector14 for subsequent sequencing15 and for qPCR standards (see below). A single biopsy stored in RNAlater was thawed and homogenized in lysis buffer5 using 5\\xa0mm stainless steel beads5 and a tissue grinder6. Total RNA was extracted using a commercial kit7 including an on\\x90column DNAse treatment, and eluted in 30\\xa0L nuclease\\x90free water. RNA quantity and quality was assessed using an automated analyzer8. Reverse transcription of 1\\xa0g of RNA was performed with a cDNA synthesis kit9, which uses a mixture of oligo\\x90dT and random nonamer primers. Gastroduodenoscopy and colonoscopy were performed in all dogs according to standard procedures. Macroscopic endoscopic scoring was performed using the World Small Animal Veterinary Association (WSAVA) approved endoscopic examination report.20 Multiple biopsies (1520) from the duodenum and colon were taken with a single\\x90use endoscopic biopsy forceps, and 1015 from each site were transferred immediately into 2% neutral\\x90buffered formalin. The remaining biopsies (minimum of 2) were transferred to RNAlater4 , stored at 4°C for 24\\xa0h and then at 80°C until further processing. Clinical severity was assessed using the CCECAI at all three visits.1 Dogs were seen at three separate visits: visit 1 (recruitment), visit 2 (14\\xa0\\xa03\\xa0days after starting the trial medication) and visit 3 (42\\xa0\\xa03\\xa0days after starting the trial medication; see Fig\\xa01). The procedures performed at each visit are listed in Table\\xa01. Patients were randomized (using random permutated blocks of n\\xa0=\\xa012 that had been designed before the start of the study2) to receive either the synbiotic EF product or placebo in conjunction with a hydrolyzed protein diet3. All members of staff involved in the trial (apart from the study monitor, a head nurse) were blinded to the dogs' medications. Dogs with a history typical for chronic enteropathy (¥3\\xa0weeks of vomiting, diarrhea or both, with or without weight loss) were included. The diagnosis of chronic enteropathy was confirmed based on established criteria (no clinically relevant abnormalities on routine hematology, serum biochemistry; trypsin\\x90like immunoreactivity [TLI], canine pancreatic lipase [cPL] and adrenocorticotropic hormone [ACTH]\\x90stimulation test results within the reference range; no abnormalities on abdominal imaging [radiographs, abdominal ultrasound examination or both]). Histopathologic findings of intestinal mucosal biopsies had to show lymphoplasmacytic, eosinophilic or mixed inflammation with or without architectural changes.20, 21 Exclusion criteria were the presence of concurrent diseases or treatment with antimicrobials, anti\\x90inflammatory drugs or both 7\\xa0days before presentation. The clinical trial was approved by the RVC's Ethics and Welfare Committee (ASPA number 70/7393) and conducted between June 2010 and May 2013. The synbiotic used contained 1\\xa0\\xa0109\\xa0cfu EF NCIMB 10415 E1707, FOS and gum Arabic1. The placebo consisted of an identical capsule containing maltodextran. Initial histopathological assessment was performed at the pathology department of the RVC as part of the routine diagnostic procedure. After finalizing the study, all slides were reviewed by 1 of the authors (HB), and WSAVA standardized classification was applied to all slides.20, 21 As described previously,22 9 histologic variables, five representing morphological changes and four representing inflammatory changes were scored as absent\\xa0=\\xa00, mild\\xa0=\\xa01, moderate\\xa0=\\xa02, or severe\\xa0=\\xa03 according to the WSAVA guidelines.20, 21 The total histology score was recorded. At this stage of histologic analysis, HB was blinded to original diagnosis, visit number and treatment group.\"\n","Skipping entity [154, 162, MedTech] in the following text because the character span ' SAS9.3 ' does not align with token boundaries:\n","'Differences were considered significant if P0.05. Statistical analyses and graphical presentations were performed using Graphpad Prism5 (La Jolla, CA) and SAS9.3 (Cary, NC). Statistical analysis. Fishers exact test was used to compare the proportion of carriers that were virus positive and virus negative. Pearsons correlation coefficient was used to compare log-transformed FH levels and pneumococcal carriage density. Flow cytometry data were analyzed using Flow Jo v7.6.1 (Tree Star, Ashland, OR). To account for possible correlation among the repeated measurements in carriage status and density, GEE model was employed for the data analysis. For the GEE analysis of carriage presence (yes or no), binomial distribution and logit link function were used. For the GEE analysis of carriage density, log-transformed density was used as the dependent variable and normal distribution and identity link function were used. Three GEE models were estimated to assess the effects of FH levels and virus status on carriage status and density, separately and jointly. Model 1 has virus status (positive and negative) and day (days 2, 7, and 14) as predictors; model 2 has FH levels in log scale and day as predictors; model 3 has FH levels and virus status, day, and interaction between FH levels and virus as predictors. Anti-PspC antibody epitope mapping. Overlapping peptide arrays containing 15-mer peptides with a frameshift of four residues corresponding to the amino acid sequence of PspC3 (GenBank accession no. EF424119) were synthesized in a slide support (CelluSpots, Intavis, Cologne, Germany). Peptide arrays were incubated individually with 29 sera samples collected from 18 subjects inoculated with pneumococcus or from mice immunized with 3 doses of 5g of PspC3 containing 50g of Alum as previously described. Concentration of pneumococcal-specific antibodies was standardized so that each sample used for incubation contained 10gml1 of PspC IgG. Detection was performed using alkaline phosphatase-conjugated anti-mouse IgG and anti-human IgG (Sigma-Aldrich). MTT 0.12M (Methylthiazolyldiphenyl-tetrazolium bromide), BCIP 0.16M (5- Bromo-4-chloro-3-indolyl phosphate), and MgCl2 1M in citrate-buffered saline pH 7.0 was used for development. Pneumococcal adherence and internalization assays. D39 bacterial pellets were washed with PBS containing Ca2+ Mg2+ (Sigma-Aldrich) before being resuspended and diluted to 2  106 CFUml1 in EMEM with no antibiotics. Bacterial suspensions were added to wells containing D562 cells (500l per well), plates were gently shaken, and 20l was obtained from each well for dilution and initial CFU quantification on blood agar plates (Oxoid, Basingstoke, UK). The 24-well plates were then incubated at 37°C 5% CO2 for 3h. Wells were washed 5 times with PBS Ca2+ Mg2+ to remove unbound bacteria and cells were either lysed with 1% saponin for 10min (for adherence) or treated with 100gml1 of ampicillin in EMEM (1ml per well) for 3h at 37°C 5% CO2 before lysing (for internalization). Recovered bacteria were quantified by serial dilution on blood agar plates. Plates were incubated at 37°C 5% CO2 overnight and initial and recovered CFUs were counted. Averages of initial D39 CFU values were used to create a correction factor so that the recovered CFUs for both strains could be compared. All conditions were performed in duplicate and experiments were performed at least twice. For each experiment, the average recovered CFUs from the control condition (adherence of D39 untreated bacteria to noninflamed cells) was used to calculate fold change for all remaining conditions. Epithelial polymeric immunoglobulin receptor and platelet-activating factor receptor expression by flow cytometry. Stimulated (inflamed) and nonstimulated (noninflamed) cells were scraped from the wells after washing with cold PBS. Cell suspensions were incubated with platelet-activating factor receptor, human monoclonal antibody, and polymeric immunoglobulin receptor rabbit polyclonal antibody (both from Cambridge Bioscience, Cambridge, UK) at 4°C for 15min. Anti-human PE-conjugate and anti-rabbit FITC-conjugate antibodies were added for 15min at 4°C. Cells were washed and stored at 4°C before acquisition using a BD LSR II flow cytometer. A total of 10,000 gated events were recorded. PE goat anti-mouse IgG2a and FITC donkey anti-rabbit IgG (both from Biolegend, London, UK) were used as isotype controls. Pharyngeal epithelial cells and inflammation of epithelium. Detroit 562 pharyngeal epithelial cells (ATCC-CCL-138) were grown and maintained at 90100% confluence in T75 flasks (BD) in Eagles minimal essential media (EMEM; Sigma-Aldrich) supplemented with 10% fetal bovine serum (Invitrogen, Paisley, UK), 2mM L-glutamine (Sigma-Aldrich), 40Uml1 penicillin, 40gml1 streptomycin, and 80gml1 neomycin (Sigma-Aldrich) (complete media). Before adherence experiments, cells were released using EDTA/trypsin (Invitrogen), washed with media, and a suspension of cells were seeded at 1  105 per ml in complete media in a 24-well plate (1ml per well). Cells were ready for use in adherence experiments 3 or 4 days after seeding (90% confluence). For inflamed epithelium, cells were stimulated with interleukin-1, tumor necrosis factor, and interferon- (50ngml1 each) in 500l of complete EMEM for 1618h at 37°C 5% CO2. Cells were then washed 3 times with warmed PBS and 500l of EMEM without antibiotics was added to each well before addition of pneumococci. Samples were washed with PBS and incubated in 100l of goat anti-factor H (1/200) (Calbiochem) for 45min at 37°C before wash and incubation in 100l of FITC-conjugated anti-goat (1/500) (Sigma-Aldrich) at 4°C in the dark for 30min. Samples were washed twice and resuspended in 500l of PBS and 4% paraformaldehyde and stored at 4°C before acquisition using a BD LSR II flow cytometer. A total of 20,000 bacterial events were acquired and samples were gated relative to a negative control containing no anti-FH. Anti-PspC IgG binding was evaluated by incubating bacterial pellets with individually purified human anti-PspC IgG samples for 45min at 37°C before detection using anti-human IgG FITC antibody (1/10,000, Sigma-Aldrich). As FH binding to S. pneumoniae is often biphasic with strongly positive and weakly negative populations of bacteria, results are presented as fluorescence index (percentage of positive bacteria multiplied by the geometric mean fluorescence index in arbitrary units). FH binding and antibody binding assays. Bacterial stocks were washed and resuspended in PBS. For inhibition of FH binding, 107 CFUs were resuspended in 100l of individual nasal wash samples, purified IgA and IgG samples (undiluted), or purified anti-PspC IgG (1/2 diluted) and incubated for 45min at 37°C. To evaluate the saturating concentration of FH binding and for positive control of inhibition of FH binding experiments, bacterial pellets were resuspended in 100l of purified human FH (Calbiochem) (3 to 40gml1) and incubated for 45min at 37°C. Bacterial culture for  in vitro  assays. D39 pneumococci were grown in Todd Hewitt broth supplemented with 0.5% yeast extract (THY) at 37°C in 5% CO2 until optical density 0.40.5 was reached. Bacteria were either used immediately for flow binding assays or resuspended in THY 10% glycerol and stored at 80°C for adherence assays. Depletion and purification of antibodies from nasal wash and sera samples. Nasal wash samples from 10 subjects were pooled and IgG and IgA were depleted by anti-human IgG and anti-human IgA agarose (Sigma-Aldrich, Gillingham, UK), respectively. Samples were incubated with agarose and flow-through fractions were used in pneumococcal adherence assays. Dot blot assays were performed to confirm antibody depletion (data not shown). Serum samples from seven subjects were used for anti-PspC IgG purification. First, purified total IgG was obtained from individual samples by HiTrap protein G affinity column (GE Healthcare, Little Chalfont, UK). Purified samples were then loaded in CNBr-activated Sepharose coupled with recombinant purified PspC. Coupling (1mg of ligand PspC to 200mg of sepharose) and anti-PspC purification was performed as per the manufacturers instructions. To confirm anti-PspC IgG purification, ELISAs were performed using plates coated with PspC as previously described. Levels of FH, lactoferrin, SLPI, and -defensin 2 in nasal wash and anti-PspC in sera samples. To measure levels of FH, 96-well plates were coated with nasal wash samples serially diluted in carbonatebicarbonate buffer. Purified human FH (Calbiochem, Watford, UK) was used as a standard. Goat anti-FH antibodies (Calbiochem) followed by anti-goat IgG horseradish peroxidase-conjugated (R&D Systems, Abingdon, UK) were used for detection before development with TMB Substrate Reagent Set (BD, Oxford, UK). Levels of lactoferrin (AssayPro, St Charles, MO), SLPI (R&D), and -defensin 2 (Antigenix America, Huntington Station, NY) were measured by sandwich ELISAs following the manufacturers recommendation. Briefly, 96-well plates were coated with anti-human lactoferrin (AssayPro) or SLPI (R&D). Plates were blocked with phosphate-buffered saline (PBS)1% bovine serum albumin before nasal wash samples serially diluted in PBS0.1% bovine serum albumin were added. Human lactoferrin (AssayPro) and recombinant human SLPI (R&D) were used as standards. Biotinylated rabbit anti-lactoferrin or goat anti-SLPI followed by streptavidin-alkaline phosphatase (AbD Serotec, Kidlington, UK) were used for detection before development with 0.5mgml1 of P-nitrophenyl phosphate. Absorbance (450nm) was measured for all assays using the FLUOstar OMEGA plate reader (BMG Labtech, Aylesbury, UK). The -defensin 2 levels were measured using a kit as per the manufacturers instructions (Antigenix America). Levels of anti-PspC IgG were measured by ELISAs using plates coated with recombinant purified PspC and a standard sera sample with known anti-PspC concentration as previously described using the last dilution of the sample with optical density >0.1 to calculate the sample concentration. A four-parameter fit was used to generate the standard curve. All samples were run in triplicate, and samples with coefficient of variation of >15% were repeated. Detection and identification of URT virus. Oropharyngeal swabs were collected 5 days before intranasal inoculation and immediately put in viral transport medium (MWE, Corsham, UK). Viral multiplex PCR was performed as described elsewhere. Briefly, viral RNA was extracted from 200l of viral transport medium and eluted into 85l buffer using the standard Pathogen Complex 200 protocol on the QIAsymphony (QIAGEN, Crawley, UK). The eluates were then analyzed using four multiplex PCR assays on the LC480 real-time PCR machine (Roche Diagnostics, Burgess Hill, UK). The assay panel covered the qualitative detection of influenza A and B, respiratory syncytial virus, human metapneumovirus, human rhinovirus, parainfluenza viruses 14, and coronaviruses OC43, NL63, 229E, and HKU1. Nasal wash samples were obtained 5 days before intranasal inoculation with S. pneumoniae serotype 6B and then on days 2, 7, and 14 after inoculation. Inoculation doses were between 50  103 and 320  103 colony-forming units (CFUs) per 100l. Bacterial preparation, inoculation of subjects with 6B pneumococcus, nasal wash sampling, carriage detection, and quantification of carriage density were done as described previously. EHPC study and pneumococcal detection. Healthy adult subjects were enrolled with informed consent to an EHPC trial. All participants were nonsmoking adults aged 1860 years who had no close contact with at-risk individuals, including young children (under the age of 5 years) and the elderly (over 65 years with comorbidities). Ethical approval was obtained from the National Health Service Research Ethics Committee (11/nasal wash/0592 and 12/nasal wash/0873) and the study was sponsored by the Royal Liverpool and Broadgreen University Hospitals Trust.'\n","Skipping entity [9272, 9275, MedTech] in the following text because the character span 'PBS' does not align with token boundaries:\n","'Differences were considered significant if P0.05. Statistical analyses and graphical presentations were performed using Graphpad Prism5 (La Jolla, CA) and SAS9.3 (Cary, NC). Statistical analysis. Fishers exact test was used to compare the proportion of carriers that were virus positive and virus negative. Pearsons correlation coefficient was used to compare log-transformed FH levels and pneumococcal carriage density. Flow cytometry data were analyzed using Flow Jo v7.6.1 (Tree Star, Ashland, OR). To account for possible correlation among the repeated measurements in carriage status and density, GEE model was employed for the data analysis. For the GEE analysis of carriage presence (yes or no), binomial distribution and logit link function were used. For the GEE analysis of carriage density, log-transformed density was used as the dependent variable and normal distribution and identity link function were used. Three GEE models were estimated to assess the effects of FH levels and virus status on carriage status and density, separately and jointly. Model 1 has virus status (positive and negative) and day (days 2, 7, and 14) as predictors; model 2 has FH levels in log scale and day as predictors; model 3 has FH levels and virus status, day, and interaction between FH levels and virus as predictors. Anti-PspC antibody epitope mapping. Overlapping peptide arrays containing 15-mer peptides with a frameshift of four residues corresponding to the amino acid sequence of PspC3 (GenBank accession no. EF424119) were synthesized in a slide support (CelluSpots, Intavis, Cologne, Germany). Peptide arrays were incubated individually with 29 sera samples collected from 18 subjects inoculated with pneumococcus or from mice immunized with 3 doses of 5g of PspC3 containing 50g of Alum as previously described. Concentration of pneumococcal-specific antibodies was standardized so that each sample used for incubation contained 10gml1 of PspC IgG. Detection was performed using alkaline phosphatase-conjugated anti-mouse IgG and anti-human IgG (Sigma-Aldrich). MTT 0.12M (Methylthiazolyldiphenyl-tetrazolium bromide), BCIP 0.16M (5- Bromo-4-chloro-3-indolyl phosphate), and MgCl2 1M in citrate-buffered saline pH 7.0 was used for development. Pneumococcal adherence and internalization assays. D39 bacterial pellets were washed with PBS containing Ca2+ Mg2+ (Sigma-Aldrich) before being resuspended and diluted to 2  106 CFUml1 in EMEM with no antibiotics. Bacterial suspensions were added to wells containing D562 cells (500l per well), plates were gently shaken, and 20l was obtained from each well for dilution and initial CFU quantification on blood agar plates (Oxoid, Basingstoke, UK). The 24-well plates were then incubated at 37°C 5% CO2 for 3h. Wells were washed 5 times with PBS Ca2+ Mg2+ to remove unbound bacteria and cells were either lysed with 1% saponin for 10min (for adherence) or treated with 100gml1 of ampicillin in EMEM (1ml per well) for 3h at 37°C 5% CO2 before lysing (for internalization). Recovered bacteria were quantified by serial dilution on blood agar plates. Plates were incubated at 37°C 5% CO2 overnight and initial and recovered CFUs were counted. Averages of initial D39 CFU values were used to create a correction factor so that the recovered CFUs for both strains could be compared. All conditions were performed in duplicate and experiments were performed at least twice. For each experiment, the average recovered CFUs from the control condition (adherence of D39 untreated bacteria to noninflamed cells) was used to calculate fold change for all remaining conditions. Epithelial polymeric immunoglobulin receptor and platelet-activating factor receptor expression by flow cytometry. Stimulated (inflamed) and nonstimulated (noninflamed) cells were scraped from the wells after washing with cold PBS. Cell suspensions were incubated with platelet-activating factor receptor, human monoclonal antibody, and polymeric immunoglobulin receptor rabbit polyclonal antibody (both from Cambridge Bioscience, Cambridge, UK) at 4°C for 15min. Anti-human PE-conjugate and anti-rabbit FITC-conjugate antibodies were added for 15min at 4°C. Cells were washed and stored at 4°C before acquisition using a BD LSR II flow cytometer. A total of 10,000 gated events were recorded. PE goat anti-mouse IgG2a and FITC donkey anti-rabbit IgG (both from Biolegend, London, UK) were used as isotype controls. Pharyngeal epithelial cells and inflammation of epithelium. Detroit 562 pharyngeal epithelial cells (ATCC-CCL-138) were grown and maintained at 90100% confluence in T75 flasks (BD) in Eagles minimal essential media (EMEM; Sigma-Aldrich) supplemented with 10% fetal bovine serum (Invitrogen, Paisley, UK), 2mM L-glutamine (Sigma-Aldrich), 40Uml1 penicillin, 40gml1 streptomycin, and 80gml1 neomycin (Sigma-Aldrich) (complete media). Before adherence experiments, cells were released using EDTA/trypsin (Invitrogen), washed with media, and a suspension of cells were seeded at 1  105 per ml in complete media in a 24-well plate (1ml per well). Cells were ready for use in adherence experiments 3 or 4 days after seeding (90% confluence). For inflamed epithelium, cells were stimulated with interleukin-1, tumor necrosis factor, and interferon- (50ngml1 each) in 500l of complete EMEM for 1618h at 37°C 5% CO2. Cells were then washed 3 times with warmed PBS and 500l of EMEM without antibiotics was added to each well before addition of pneumococci. Samples were washed with PBS and incubated in 100l of goat anti-factor H (1/200) (Calbiochem) for 45min at 37°C before wash and incubation in 100l of FITC-conjugated anti-goat (1/500) (Sigma-Aldrich) at 4°C in the dark for 30min. Samples were washed twice and resuspended in 500l of PBS and 4% paraformaldehyde and stored at 4°C before acquisition using a BD LSR II flow cytometer. A total of 20,000 bacterial events were acquired and samples were gated relative to a negative control containing no anti-FH. Anti-PspC IgG binding was evaluated by incubating bacterial pellets with individually purified human anti-PspC IgG samples for 45min at 37°C before detection using anti-human IgG FITC antibody (1/10,000, Sigma-Aldrich). As FH binding to S. pneumoniae is often biphasic with strongly positive and weakly negative populations of bacteria, results are presented as fluorescence index (percentage of positive bacteria multiplied by the geometric mean fluorescence index in arbitrary units). FH binding and antibody binding assays. Bacterial stocks were washed and resuspended in PBS. For inhibition of FH binding, 107 CFUs were resuspended in 100l of individual nasal wash samples, purified IgA and IgG samples (undiluted), or purified anti-PspC IgG (1/2 diluted) and incubated for 45min at 37°C. To evaluate the saturating concentration of FH binding and for positive control of inhibition of FH binding experiments, bacterial pellets were resuspended in 100l of purified human FH (Calbiochem) (3 to 40gml1) and incubated for 45min at 37°C. Bacterial culture for  in vitro  assays. D39 pneumococci were grown in Todd Hewitt broth supplemented with 0.5% yeast extract (THY) at 37°C in 5% CO2 until optical density 0.40.5 was reached. Bacteria were either used immediately for flow binding assays or resuspended in THY 10% glycerol and stored at 80°C for adherence assays. Depletion and purification of antibodies from nasal wash and sera samples. Nasal wash samples from 10 subjects were pooled and IgG and IgA were depleted by anti-human IgG and anti-human IgA agarose (Sigma-Aldrich, Gillingham, UK), respectively. Samples were incubated with agarose and flow-through fractions were used in pneumococcal adherence assays. Dot blot assays were performed to confirm antibody depletion (data not shown). Serum samples from seven subjects were used for anti-PspC IgG purification. First, purified total IgG was obtained from individual samples by HiTrap protein G affinity column (GE Healthcare, Little Chalfont, UK). Purified samples were then loaded in CNBr-activated Sepharose coupled with recombinant purified PspC. Coupling (1mg of ligand PspC to 200mg of sepharose) and anti-PspC purification was performed as per the manufacturers instructions. To confirm anti-PspC IgG purification, ELISAs were performed using plates coated with PspC as previously described. Levels of FH, lactoferrin, SLPI, and -defensin 2 in nasal wash and anti-PspC in sera samples. To measure levels of FH, 96-well plates were coated with nasal wash samples serially diluted in carbonatebicarbonate buffer. Purified human FH (Calbiochem, Watford, UK) was used as a standard. Goat anti-FH antibodies (Calbiochem) followed by anti-goat IgG horseradish peroxidase-conjugated (R&D Systems, Abingdon, UK) were used for detection before development with TMB Substrate Reagent Set (BD, Oxford, UK). Levels of lactoferrin (AssayPro, St Charles, MO), SLPI (R&D), and -defensin 2 (Antigenix America, Huntington Station, NY) were measured by sandwich ELISAs following the manufacturers recommendation. Briefly, 96-well plates were coated with anti-human lactoferrin (AssayPro) or SLPI (R&D). Plates were blocked with phosphate-buffered saline (PBS)1% bovine serum albumin before nasal wash samples serially diluted in PBS0.1% bovine serum albumin were added. Human lactoferrin (AssayPro) and recombinant human SLPI (R&D) were used as standards. Biotinylated rabbit anti-lactoferrin or goat anti-SLPI followed by streptavidin-alkaline phosphatase (AbD Serotec, Kidlington, UK) were used for detection before development with 0.5mgml1 of P-nitrophenyl phosphate. Absorbance (450nm) was measured for all assays using the FLUOstar OMEGA plate reader (BMG Labtech, Aylesbury, UK). The -defensin 2 levels were measured using a kit as per the manufacturers instructions (Antigenix America). Levels of anti-PspC IgG were measured by ELISAs using plates coated with recombinant purified PspC and a standard sera sample with known anti-PspC concentration as previously described using the last dilution of the sample with optical density >0.1 to calculate the sample concentration. A four-parameter fit was used to generate the standard curve. All samples were run in triplicate, and samples with coefficient of variation of >15% were repeated. Detection and identification of URT virus. Oropharyngeal swabs were collected 5 days before intranasal inoculation and immediately put in viral transport medium (MWE, Corsham, UK). Viral multiplex PCR was performed as described elsewhere. Briefly, viral RNA was extracted from 200l of viral transport medium and eluted into 85l buffer using the standard Pathogen Complex 200 protocol on the QIAsymphony (QIAGEN, Crawley, UK). The eluates were then analyzed using four multiplex PCR assays on the LC480 real-time PCR machine (Roche Diagnostics, Burgess Hill, UK). The assay panel covered the qualitative detection of influenza A and B, respiratory syncytial virus, human metapneumovirus, human rhinovirus, parainfluenza viruses 14, and coronaviruses OC43, NL63, 229E, and HKU1. Nasal wash samples were obtained 5 days before intranasal inoculation with S. pneumoniae serotype 6B and then on days 2, 7, and 14 after inoculation. Inoculation doses were between 50  103 and 320  103 colony-forming units (CFUs) per 100l. Bacterial preparation, inoculation of subjects with 6B pneumococcus, nasal wash sampling, carriage detection, and quantification of carriage density were done as described previously. EHPC study and pneumococcal detection. Healthy adult subjects were enrolled with informed consent to an EHPC trial. All participants were nonsmoking adults aged 1860 years who had no close contact with at-risk individuals, including young children (under the age of 5 years) and the elderly (over 65 years with comorbidities). Ethical approval was obtained from the National Health Service Research Ethics Committee (11/nasal wash/0592 and 12/nasal wash/0873) and the study was sponsored by the Royal Liverpool and Broadgreen University Hospitals Trust.'\n","Skipping entity [1353, 1356, MedTech] in the following text because the character span 'CD3' does not align with token boundaries:\n","'The study was conducted by the Kanto CML study group, which consists of 13 institutions in Japan, according to the Declaration of Helsinki, and the protocol was reviewed by institutional review boards or ethics committees for each participating center. Before entry into the study, all candidates were informed about the aim of the present study, and also merits and demerits of switching to dasatinib. Merits include induction of deeper response and leading to future TKI discontinuation. Demerits include newly developed adverse events caused by dasatinib, although certain adverse events due to imatinib treatment may be solved. Written informed consent was obtained from all participating patients before registration. This trial was registered at www.clinicaltrials.gov as # NCT01342679. Differences in immunophenotypes in peripheral lymphocytes between paired samples before and at 6\\xa0months after switching to dasatinib were evaluated statistically by paired t test using JMP Pro 11.2 (SAS Institute Inc.). A statistically significant result was considered to have a P value of <0.05. Immunophenotypes of peripheral blood lymphocytes were monitored before, at 2 weeks and at 1, 3, 6, 9 and 12\\xa0months after switching to dasatinib by two- or three-color flow cytometry performed by BML Inc. using monoclonal antibodies against the following antigensCD3, CD4, CD8, CD16, CD56, CD57, CD25 and CD127. Adverse events were monitored and assessed according to the Common Terminology Criteria for Adverse Events version 4.0 in all participating patients throughout the study. Chest X-ray was carried out to check pleural effusion at baseline, at 2\\xa0weeks and at 1, 2, 3, 6, 9, 12, 15 and 18\\xa0months after switching to dasatinib or more frequently if necessary. An ECG was carried out to screen for arrhythmia and to monitor the QTc interval at baseline, at 2\\xa0weeks and at 1, 2, 3, 6, 9, 12, 15 and 18\\xa0months after switching to dasatinib or more frequently if necessary. The primary endpoint in this study was the achievement of DMR at 12\\xa0months after switching to dasatinib. Secondary endpoints included the dose intensity of dasatinib at 12\\xa0months, progression-free survival, safety profiles, and immunophenotypic alterations of peripheral lymphocytes after switching to dasatinib. Progressive disease was defined as loss of complete hematological response, loss of MMR, progression to accelerated/blastic phases, or death from any cause during the treatment period. After confirmation of MMR with detectable levels of BCR-ABL1 transcripts by RQ-PCR, dasatinib therapy was initiated for eligible patients at a dose of 100\\xa0mg once a day. The study treatment was continued until disease progression or development of unacceptable adverse events. During the study, any treatment for CML other than dasatinib was not allowed. Therapy for comorbidities and/or adverse events was permitted. Interruption, dose reduction, or dose re-escalation of dasatinib due to adverse events were allowed. Before switching to dasatinib, we screened for 25 clinically important BCR-ABL1 mutations at 18 nucleotide positions by RT-PCR using PCR-Invador assay (BML, Inc., Tokyo, Japan) in all enrolled patients. The BCR-ABL1 transcript level in peripheral blood was monitored before and at 1, 3, 6, 9, 12, 15 and 18\\xa0months after switching to dasatinib. To measure the BCR-ABL1 transcript level, the RQ-PCR method with at least MR4 sensitivity (i.e., 0.0069% on IS) was carried out by Bio Medical Laboratories (BML, Inc., Tokyo, Japan) as described previously. In this analysis, we defined DMR as a peripheral major BCR-ABL1/ABL1 transcript ratio below the detection limit of RQ-PCR analysis. Inclusion criteria for the study were patients with CP-CML who were aged ¥20\\xa0years, treated with imatinib for at least 24\\xa0months and achieved MMR with detectable levels of BCR-ABL1 transcripts by RQ-PCR, and a sensitivity of at least MR4 below the standardized line. Exclusion criteria were patients with a performance status (PS) of grade ¥3 by ECOG definition, with active second primary cancer, with clinically defined pleural effusion, with a history of critical cardiovascular events, including acute myocardial infarction within 6\\xa0months, angina pectoris within 3\\xa0months, and congestive heart failure within 3\\xa0months, with prolonged QTc of ¥450\\xa0ms or the possibility of congenital prolonged QT syndrome, with a history of treatment by dasatinib, or with ABL1 mutations (T315I, F317L and V299L) for which dasatinib is ineffective, at the time of study inclusion. Pregnant or breastfeeding women were also ineligible.'\n","Skipping entity [3942, 3948, MedTech] in the following text because the character span 'FlowJo' does not align with token boundaries:\n","'Analysis of variance in conjunction with Students t test was performed to identify significant differences. All analyses were performed using GraphPad Prism 6.0 software (GraphPad, La Jolla, CA, USA). CD106+ MSCs or CD106 MSCs at passage 4 were stimulated by incubation with TNF- (PepTech, Burlington, MA, USA) 10\\xa0ng/ml for 1\\xa0h, 4\\xa0h, or 24\\xa0h, or 500\\xa0ng/ml LPS (Sigma-Aldrich) for 0.5\\xa0h, 1\\xa0h, 2\\xa0h, or 4\\xa0h. Cells were lysed in Bicine/CHAPS Lysis Buffer (ProteinSimple) supplemented with DMSO Inhibitor Mix (ProteinSimple) and Aqueous Inhibitor Mix (ProteinSimple) at 4\\xa0°C for 30\\xa0min, and the lysate was mixed with Premix G2 (pH\\xa0310) (ProteinSimple) and pI Standard Ladder 3 (44:1, ProteinSimple). The rabbit anti-NF-B (p65) antibody (primary antibody, Cell Signaling Technology) was diluted 1:50 in antibody dilution buffer, and the anti-human IgG-HRP (secondary antibody, Protein Simple) was diluted 1:100 in antibody dilution buffer. Luminol/peroxide was mixed at a 1:1 ratio. The NanoPro 1000 (ProteinSimple) was loaded and run according to the manufacturers specifications. Emitted light was quantified for 30\\xa0s, 60\\xa0s, 120\\xa0s, 240\\xa0s, 480\\xa0s, and 960\\xa0s. Compass software 2.5.11 (ProteinSimple) was used to identify and quantify chemiluminescence peaks and optimize tracings. BM-MSCs were blocked by incubation with 500\\xa0ng/ml of the CD106 blocking antibody ab47159 for 1\\xa0h at 4\\xa0°C. BM-MSCs were blocked by incubation with 10 nM of the NF-B-specific inhibitor BAY-11-7082 for 0.5\\xa0h, 1\\xa0h, or 12\\xa0h. Aliquots of cell preparations containing 2105 CD106+ MSCs or CD106 MSCs and 1105 UCB CD34+ cells in 15\\xa0l of Roswell Park Memorial Institute (RPMI) 1640 medium were injected into the tail vein of 28- to 35-day-old NOD/SCID irradiated mice (n=6 for each pair of cells) using a Hamilton syringe. Some low-dose irradiated mice (320\\xa0cGy) were sacrificed 42\\xa0days after xenotransplantation. CD45+ cells were analyzed after staining with human anti-CD45-APC antibody along with the appropriate isotype controls (BD Biosciences) according to the manufacturers instructions. Some high-dose irradiated mice (360\\xa0cGy) (n=6 for each pair of cells) were observed until 56\\xa0days after xenotransplantation or death. Cocultures of CD34+ cells with MSCs were incubated for 14\\xa0days. Nonadherent viable cells (5102 in each well of a 24-well-plate) were plated on semisolid Iscoves modified Dulbeccos medium (IMDM; Gibco) supplemented with 1% methylcellulose, 10% fetal bovine serum (FBS), 1% BSA, 104 M mercaptoethanol, 2\\xa0mMl-glutamine, and 100\\xa0ng/ml thrombopoietin. Cultures were incubated at 37\\xa0°C in a humidified atmosphere with 5% CO2. CFU-MK colonies were identified after 14\\xa0days of culture under an Olympus IX71 inverted microscope (Olympus), and typical colonies were selected for immunostaining. Cell smears were prepared using Cytospin (Thermo-Fisher Scientific), stained with anti-CD41a antibody, and observed using an UltraVIEWVoX Confocal Imaging System (PerkinElmer). Cocultures of CD34+ cells with MSCs or blocked MSCs were incubated for 14\\xa0days or 10\\xa0days. Nonadherent viable cells (2.5102 or 5102 in each well of a 24-well-plate) were plated on 0.5\\xa0ml of methylcellulose medium (Stem Cell Technologies) to evaluate the in vitro effects of MSCs on CFU growth. Colonies of >50 cells were scored after 14\\xa0days of culture. Experiments were performed in triplicate. A total of 1105 CD34+ cells suspended in serum-free StemSpan\\x84¢ H3000 (Stem Cell Technologies) culture medium was applied to feeder layers composed of BM-MSCs, which were blocked by BAY-11-7082, as described previously. Cocultures were incubated for 10\\xa0days, and the culture medium was replenished every 3.5\\xa0days. Nonadherent viable cells were stained for FCM analysis using CD34-APC along with the appropriate isotype controls (BD Biosciences) according to the manufacturers instructions. Data acquisition was also performed using an LSR II flow cytometer (BD Biosciences) and analyzed with FlowJo7.6 software (FlowJo). A total of 5104 CD34+ cells suspended in 1\\xa0ml of serum-free StemSpan\\x84¢ H3000 (Stem Cell Technologies, Vancouver, Canada) culture medium was applied to feeder layers composed of 5104 BM-MSCs, as described previously. Cocultures were incubated for 14\\xa0days, and the culture medium was replenished every 3.5\\xa0days. Nonadherent viable cells were stained for FCM analysis using the antibodies anti-CD34-APC, anti-CD61-PE, anti-CD41a-PE, and anti-CD42b-PE, along with the appropriate isotype controls (BD Biosciences) according to the manufacturers instructions. Data acquisition was performed using an LSR II flow cytometer (BD Biosciences) and analyzed with FlowJo7.6 software (FlowJo). CD34+ cells were freshly purified from umbilical cord blood (UCB) using a CD34/MACS isolation kit (MiltenyiBiotec, Bergisch, Gladbach, Germany) according to the manufacturers instructions. Cell fractions with 955% CD34+ cell purity was used for subsequent experiments. Supernatants obtained from MSC-conditioned medium were used to detect vascular endothelial growth factor (VEGF) levels using an enzyme-linked immunosorbent assay (ELISA; R&D Systems, Minneapolis, MN, USA) according to the manufacturers instructions. To confirm our in vitro data, we examined vasculogenesis in vivo by performing a Matrigel plug assay. Aliquots of 5105 MSCs were resuspended in 500\\xa0l of Matrigel (BD Biosciences) according to the manufacturers instructions and implanted into the back of 42-day-old nude mice (n=6 in each group). Mice implanted with Matrigel only were used as negative controls. After 21\\xa0days, the Matrigel plugs were harvested, assayed for microvessels identified as luminal structures with red blood cells using H&E staining, and counted. Wells in 96-well plates were covered with 50\\xa0l of growth factor-reduced Matrigel (BD Biosciences). Aliquots of CD106+ MSCs and CD106 MSCs were seeded at a density of 10,000 cells/cm2 and cultured in a humidified atmosphere with 5% CO2 for 24\\xa0h. The formation of capillary-like structures was observed using an Olympus IX71 inverted microscope (Olympus, Tokyo, Japan), and pictures were taken at different time points using an Olympus DP71 camera (Olympus). Western blotting procedures were performed according to the protocol described by Song et al.. Briefly, BM-MSCs were collected, washed, and lysed with RIPA lysis buffer (Beyotime Institute of Biotechnology, Shanghai, China) supplemented with PMSF (Invitrogen, Carlsbad, CA, USA). Total protein was extracted and quantified by the BCA protein assay kit (Pierce, Woodland Hills, CA, USA). A total of 30\\xa0g protein was denatured, separated by SDS-PAGE electrophoresis, and transferred to a PVDF membrane. The transferred membranes were blocked using 5% bovine serum albumin (BSA) in TBST, incubated with anti-human CD106 mouse monoclonal antibody (Abcam) overnight, and then incubated with the corresponding horseradish peroxidase (HRP)-conjugated secondary antibody at a dilution of 1:2000 for 2\\xa0h. Bands were visualized using enhanced chemiluminescence (ECL; Thermo-Fisher, Scientific, Waltham, MA, USA) detection reagents, and scanned images were quantified using Image J (https://imagej.nih.gov/ij/). The ratio of target gene to -actin was used for the semiquantification and comparison between the two groups. Samples were photographed using an UltraVIEWVoX Confocal Imaging System (PerkinElmer, Waltham, MA, USA). Selected CD106+/CD106 MSCs were stained for NF-B and nuclei to examine the NF-B expression level. The expression of NF-B was calculated as the mean of the fluorescence intensity in 6 continuous views. CD106+/CD106 MSCs were first washed and fixed with 4% formaldehyde for 15\\xa0min and then blocked with blocking buffer (PBS/5% normal serum/0.3% Triton\\x84¢ X-100) for 60\\xa0min. Then, CD106+/CD106 MSCs were labeled with rabbit-anti-human NF-B overnight and then labeled with Alexa Fluor 488-conjugated secondary antibody or Alexa Fluor 546-conjugated secondary antibody for 60\\xa0min. The nucleus was marked with DAPI. The expression of cell surface molecules was assessed according to the manufacturers instructions. Normal MSCs (N MSCs) were stained for CD106 and NF-kB. N MSCs were first washed and fixed with 4% formaldehyde for 15\\xa0min and then blocked with blocking buffer (phosphate-buffered saline (PBS)/5% normal serum) for 60\\xa0min. N MSCs were labeled with mouse-anti-human CD106 overnight and then labeled with Alexa Fluor 546-goat anti-mouse IgG conjugated secondary antibody for 60\\xa0min. N MSCs with 0.3% Triton\\x84¢ X-100 for 60\\xa0min were labeled with rabbit-anti-human NF-B overnight and then labeled with Alexa Fluor 488-conjugated secondary antibody (donkey anti-rabbit) for 60\\xa0min. The nucleus was marked with DAPI. H&E staining (Sigma-Aldrich) and histochemical staining (Abcam, Cambridge, UK) were performed according to the manufacturers instructions. Samples were photographed using a Nikon ElipseTi-U microscope (Nikon, Tokyo, Japan). Cells were stained with antibodies along with the appropriate isotype controls (BD Biosciences) according to the manufacturers instructions. Data acquisition was performed using an LSR II flow cytometer (BD Biosciences) and analyzed with FlowJo 7.6 software (FlowJo, Ashland, OR, USA). MSCs were labeled with PE-conjugated anti-CD106 antibody (BD Biosciences). CD106+ MSCs and CD106 MSCs were separated using a CD106-positive selection magnetic-activated cell sorting (MACS) isolation kit (Miltenyi Biotech, Bergisch Gladbach, Germany) according to the manufacturers instructions. CD106+ MSCs or CD106 MSCs (¥90% purity) were used for subsequent experiments. RNA samples were first treated with DNase I to degrade any possible DNA contamination. Next, mRNA was enriched using oligo (dT) magnetic beads for eukaryotes and fragmented into short fragments of approximately 200\\xa0bp. The first strand of cDNA was synthesized using a random hexamer-primer, and then buffer, dNTPs, RNase H, and DNA polymerase I were added to synthesize the second strand. Double-stranded cDNA was purified with magnetic beads, followed by end reparation and 3-end single nucleotide A (adenine) addition. Finally, sequencing adaptors were ligated to the fragments, which were enriched using polymerase chain reaction (PCR) amplification. A sample library was qualified and quantified using an Agilent 2100 Bioanalyzer (Agilent, Santa Clara, CA, USA) and an ABI StepOnePlus Real-Time PCR System (Applied Biosystems, Carlsbad, CA, USA) during the quantitative-competitive (QC) step. Library products were ready for sequencing via Illumina HiSeqTM 2000 (Illumina, San Diego, CA, USA) or other sequencer when necessary. Next, quantitative real-time polymerase chain reaction (qRT-PCR) was performed to confirm the gene expression levels of RNA transcripts with sequence-specific oligonucleotide primers as described previously. BM-MSCs were isolated and cultured in Dulbeccos modified Eagles medium. BM-MSCs were identified by their surface markers with a panel of monoclonal antibodies against CD13 (WM15), CD29 (MAR4), CD44 (G44-26), CD49e (IIA1), CD73 (AD2), CD105 (266), CD166 (3A6), CD31 (WM59), CD34 (581), CD45 (HI30), CD90 (5E10), HLA-ABC (G46-2.6), HLA-DR (G46-6), CD14 (M5E2), CD40 (5C3), and CD11b (ICRF44), along with the appropriate isotype monoclonal antibodies using a FACScanflow cytometer (BD Biosciences, San Jose, CA, USA). Our experimental research on NOD/SCID and nude mice followed internationally recognized guidelines. Ethical approval for the animal experiments was provided by the Ethical Committee of the Institute of Hematology and Blood Diseases Hospital, Chinese Academy of Medical Science and Peking Union Medical College. The ethical approval documents reference number is KT2012003-m-6. BM samples from 28 (17 male and 11 female) de novo acquired AA patients with a median age of 31\\xa0years (range 1859 years) were analyzed after the signing of a written informed consent form in accordance with the Declaration of Helsinki. The study was approved by the Committee for Medical Care and Safety, Institute of Hematology and Blood Diseases Hospital, Chinese Academy of Medical Science and Peking Union Medical College (ethical approval documents reference number KT2014005-EC-1). This cohort consisted of four patients with severe AA and 24 with nonsevere AA. The diagnosis and severity classification of AA was established by morphological examination of the BM and peripheral blood samples after excluding any other acquired BM failure syndromes, such as paroxysmal nocturnal hemoglobinuria, myelodysplastic syndrome, and congenital BM failure syndromes, according to international criteria. Samples from 19 (14 male and 5 female) age-matched (range 2056 years) healthy controls were obtained after they had signed the written informed consent form described above.'\n","Skipping entity [7686, 7689, MedTech] in the following text because the character span 'PBS' does not align with token boundaries:\n","'Analysis of variance in conjunction with Students t test was performed to identify significant differences. All analyses were performed using GraphPad Prism 6.0 software (GraphPad, La Jolla, CA, USA). CD106+ MSCs or CD106 MSCs at passage 4 were stimulated by incubation with TNF- (PepTech, Burlington, MA, USA) 10\\xa0ng/ml for 1\\xa0h, 4\\xa0h, or 24\\xa0h, or 500\\xa0ng/ml LPS (Sigma-Aldrich) for 0.5\\xa0h, 1\\xa0h, 2\\xa0h, or 4\\xa0h. Cells were lysed in Bicine/CHAPS Lysis Buffer (ProteinSimple) supplemented with DMSO Inhibitor Mix (ProteinSimple) and Aqueous Inhibitor Mix (ProteinSimple) at 4\\xa0°C for 30\\xa0min, and the lysate was mixed with Premix G2 (pH\\xa0310) (ProteinSimple) and pI Standard Ladder 3 (44:1, ProteinSimple). The rabbit anti-NF-B (p65) antibody (primary antibody, Cell Signaling Technology) was diluted 1:50 in antibody dilution buffer, and the anti-human IgG-HRP (secondary antibody, Protein Simple) was diluted 1:100 in antibody dilution buffer. Luminol/peroxide was mixed at a 1:1 ratio. The NanoPro 1000 (ProteinSimple) was loaded and run according to the manufacturers specifications. Emitted light was quantified for 30\\xa0s, 60\\xa0s, 120\\xa0s, 240\\xa0s, 480\\xa0s, and 960\\xa0s. Compass software 2.5.11 (ProteinSimple) was used to identify and quantify chemiluminescence peaks and optimize tracings. BM-MSCs were blocked by incubation with 500\\xa0ng/ml of the CD106 blocking antibody ab47159 for 1\\xa0h at 4\\xa0°C. BM-MSCs were blocked by incubation with 10 nM of the NF-B-specific inhibitor BAY-11-7082 for 0.5\\xa0h, 1\\xa0h, or 12\\xa0h. Aliquots of cell preparations containing 2105 CD106+ MSCs or CD106 MSCs and 1105 UCB CD34+ cells in 15\\xa0l of Roswell Park Memorial Institute (RPMI) 1640 medium were injected into the tail vein of 28- to 35-day-old NOD/SCID irradiated mice (n=6 for each pair of cells) using a Hamilton syringe. Some low-dose irradiated mice (320\\xa0cGy) were sacrificed 42\\xa0days after xenotransplantation. CD45+ cells were analyzed after staining with human anti-CD45-APC antibody along with the appropriate isotype controls (BD Biosciences) according to the manufacturers instructions. Some high-dose irradiated mice (360\\xa0cGy) (n=6 for each pair of cells) were observed until 56\\xa0days after xenotransplantation or death. Cocultures of CD34+ cells with MSCs were incubated for 14\\xa0days. Nonadherent viable cells (5102 in each well of a 24-well-plate) were plated on semisolid Iscoves modified Dulbeccos medium (IMDM; Gibco) supplemented with 1% methylcellulose, 10% fetal bovine serum (FBS), 1% BSA, 104 M mercaptoethanol, 2\\xa0mMl-glutamine, and 100\\xa0ng/ml thrombopoietin. Cultures were incubated at 37\\xa0°C in a humidified atmosphere with 5% CO2. CFU-MK colonies were identified after 14\\xa0days of culture under an Olympus IX71 inverted microscope (Olympus), and typical colonies were selected for immunostaining. Cell smears were prepared using Cytospin (Thermo-Fisher Scientific), stained with anti-CD41a antibody, and observed using an UltraVIEWVoX Confocal Imaging System (PerkinElmer). Cocultures of CD34+ cells with MSCs or blocked MSCs were incubated for 14\\xa0days or 10\\xa0days. Nonadherent viable cells (2.5102 or 5102 in each well of a 24-well-plate) were plated on 0.5\\xa0ml of methylcellulose medium (Stem Cell Technologies) to evaluate the in vitro effects of MSCs on CFU growth. Colonies of >50 cells were scored after 14\\xa0days of culture. Experiments were performed in triplicate. A total of 1105 CD34+ cells suspended in serum-free StemSpan\\x84¢ H3000 (Stem Cell Technologies) culture medium was applied to feeder layers composed of BM-MSCs, which were blocked by BAY-11-7082, as described previously. Cocultures were incubated for 10\\xa0days, and the culture medium was replenished every 3.5\\xa0days. Nonadherent viable cells were stained for FCM analysis using CD34-APC along with the appropriate isotype controls (BD Biosciences) according to the manufacturers instructions. Data acquisition was also performed using an LSR II flow cytometer (BD Biosciences) and analyzed with FlowJo7.6 software (FlowJo). A total of 5104 CD34+ cells suspended in 1\\xa0ml of serum-free StemSpan\\x84¢ H3000 (Stem Cell Technologies, Vancouver, Canada) culture medium was applied to feeder layers composed of 5104 BM-MSCs, as described previously. Cocultures were incubated for 14\\xa0days, and the culture medium was replenished every 3.5\\xa0days. Nonadherent viable cells were stained for FCM analysis using the antibodies anti-CD34-APC, anti-CD61-PE, anti-CD41a-PE, and anti-CD42b-PE, along with the appropriate isotype controls (BD Biosciences) according to the manufacturers instructions. Data acquisition was performed using an LSR II flow cytometer (BD Biosciences) and analyzed with FlowJo7.6 software (FlowJo). CD34+ cells were freshly purified from umbilical cord blood (UCB) using a CD34/MACS isolation kit (MiltenyiBiotec, Bergisch, Gladbach, Germany) according to the manufacturers instructions. Cell fractions with 955% CD34+ cell purity was used for subsequent experiments. Supernatants obtained from MSC-conditioned medium were used to detect vascular endothelial growth factor (VEGF) levels using an enzyme-linked immunosorbent assay (ELISA; R&D Systems, Minneapolis, MN, USA) according to the manufacturers instructions. To confirm our in vitro data, we examined vasculogenesis in vivo by performing a Matrigel plug assay. Aliquots of 5105 MSCs were resuspended in 500\\xa0l of Matrigel (BD Biosciences) according to the manufacturers instructions and implanted into the back of 42-day-old nude mice (n=6 in each group). Mice implanted with Matrigel only were used as negative controls. After 21\\xa0days, the Matrigel plugs were harvested, assayed for microvessels identified as luminal structures with red blood cells using H&E staining, and counted. Wells in 96-well plates were covered with 50\\xa0l of growth factor-reduced Matrigel (BD Biosciences). Aliquots of CD106+ MSCs and CD106 MSCs were seeded at a density of 10,000 cells/cm2 and cultured in a humidified atmosphere with 5% CO2 for 24\\xa0h. The formation of capillary-like structures was observed using an Olympus IX71 inverted microscope (Olympus, Tokyo, Japan), and pictures were taken at different time points using an Olympus DP71 camera (Olympus). Western blotting procedures were performed according to the protocol described by Song et al.. Briefly, BM-MSCs were collected, washed, and lysed with RIPA lysis buffer (Beyotime Institute of Biotechnology, Shanghai, China) supplemented with PMSF (Invitrogen, Carlsbad, CA, USA). Total protein was extracted and quantified by the BCA protein assay kit (Pierce, Woodland Hills, CA, USA). A total of 30\\xa0g protein was denatured, separated by SDS-PAGE electrophoresis, and transferred to a PVDF membrane. The transferred membranes were blocked using 5% bovine serum albumin (BSA) in TBST, incubated with anti-human CD106 mouse monoclonal antibody (Abcam) overnight, and then incubated with the corresponding horseradish peroxidase (HRP)-conjugated secondary antibody at a dilution of 1:2000 for 2\\xa0h. Bands were visualized using enhanced chemiluminescence (ECL; Thermo-Fisher, Scientific, Waltham, MA, USA) detection reagents, and scanned images were quantified using Image J (https://imagej.nih.gov/ij/). The ratio of target gene to -actin was used for the semiquantification and comparison between the two groups. Samples were photographed using an UltraVIEWVoX Confocal Imaging System (PerkinElmer, Waltham, MA, USA). Selected CD106+/CD106 MSCs were stained for NF-B and nuclei to examine the NF-B expression level. The expression of NF-B was calculated as the mean of the fluorescence intensity in 6 continuous views. CD106+/CD106 MSCs were first washed and fixed with 4% formaldehyde for 15\\xa0min and then blocked with blocking buffer (PBS/5% normal serum/0.3% Triton\\x84¢ X-100) for 60\\xa0min. Then, CD106+/CD106 MSCs were labeled with rabbit-anti-human NF-B overnight and then labeled with Alexa Fluor 488-conjugated secondary antibody or Alexa Fluor 546-conjugated secondary antibody for 60\\xa0min. The nucleus was marked with DAPI. The expression of cell surface molecules was assessed according to the manufacturers instructions. Normal MSCs (N MSCs) were stained for CD106 and NF-kB. N MSCs were first washed and fixed with 4% formaldehyde for 15\\xa0min and then blocked with blocking buffer (phosphate-buffered saline (PBS)/5% normal serum) for 60\\xa0min. N MSCs were labeled with mouse-anti-human CD106 overnight and then labeled with Alexa Fluor 546-goat anti-mouse IgG conjugated secondary antibody for 60\\xa0min. N MSCs with 0.3% Triton\\x84¢ X-100 for 60\\xa0min were labeled with rabbit-anti-human NF-B overnight and then labeled with Alexa Fluor 488-conjugated secondary antibody (donkey anti-rabbit) for 60\\xa0min. The nucleus was marked with DAPI. H&E staining (Sigma-Aldrich) and histochemical staining (Abcam, Cambridge, UK) were performed according to the manufacturers instructions. Samples were photographed using a Nikon ElipseTi-U microscope (Nikon, Tokyo, Japan). Cells were stained with antibodies along with the appropriate isotype controls (BD Biosciences) according to the manufacturers instructions. Data acquisition was performed using an LSR II flow cytometer (BD Biosciences) and analyzed with FlowJo 7.6 software (FlowJo, Ashland, OR, USA). MSCs were labeled with PE-conjugated anti-CD106 antibody (BD Biosciences). CD106+ MSCs and CD106 MSCs were separated using a CD106-positive selection magnetic-activated cell sorting (MACS) isolation kit (Miltenyi Biotech, Bergisch Gladbach, Germany) according to the manufacturers instructions. CD106+ MSCs or CD106 MSCs (¥90% purity) were used for subsequent experiments. RNA samples were first treated with DNase I to degrade any possible DNA contamination. Next, mRNA was enriched using oligo (dT) magnetic beads for eukaryotes and fragmented into short fragments of approximately 200\\xa0bp. The first strand of cDNA was synthesized using a random hexamer-primer, and then buffer, dNTPs, RNase H, and DNA polymerase I were added to synthesize the second strand. Double-stranded cDNA was purified with magnetic beads, followed by end reparation and 3-end single nucleotide A (adenine) addition. Finally, sequencing adaptors were ligated to the fragments, which were enriched using polymerase chain reaction (PCR) amplification. A sample library was qualified and quantified using an Agilent 2100 Bioanalyzer (Agilent, Santa Clara, CA, USA) and an ABI StepOnePlus Real-Time PCR System (Applied Biosystems, Carlsbad, CA, USA) during the quantitative-competitive (QC) step. Library products were ready for sequencing via Illumina HiSeqTM 2000 (Illumina, San Diego, CA, USA) or other sequencer when necessary. Next, quantitative real-time polymerase chain reaction (qRT-PCR) was performed to confirm the gene expression levels of RNA transcripts with sequence-specific oligonucleotide primers as described previously. BM-MSCs were isolated and cultured in Dulbeccos modified Eagles medium. BM-MSCs were identified by their surface markers with a panel of monoclonal antibodies against CD13 (WM15), CD29 (MAR4), CD44 (G44-26), CD49e (IIA1), CD73 (AD2), CD105 (266), CD166 (3A6), CD31 (WM59), CD34 (581), CD45 (HI30), CD90 (5E10), HLA-ABC (G46-2.6), HLA-DR (G46-6), CD14 (M5E2), CD40 (5C3), and CD11b (ICRF44), along with the appropriate isotype monoclonal antibodies using a FACScanflow cytometer (BD Biosciences, San Jose, CA, USA). Our experimental research on NOD/SCID and nude mice followed internationally recognized guidelines. Ethical approval for the animal experiments was provided by the Ethical Committee of the Institute of Hematology and Blood Diseases Hospital, Chinese Academy of Medical Science and Peking Union Medical College. The ethical approval documents reference number is KT2012003-m-6. BM samples from 28 (17 male and 11 female) de novo acquired AA patients with a median age of 31\\xa0years (range 1859 years) were analyzed after the signing of a written informed consent form in accordance with the Declaration of Helsinki. The study was approved by the Committee for Medical Care and Safety, Institute of Hematology and Blood Diseases Hospital, Chinese Academy of Medical Science and Peking Union Medical College (ethical approval documents reference number KT2014005-EC-1). This cohort consisted of four patients with severe AA and 24 with nonsevere AA. The diagnosis and severity classification of AA was established by morphological examination of the BM and peripheral blood samples after excluding any other acquired BM failure syndromes, such as paroxysmal nocturnal hemoglobinuria, myelodysplastic syndrome, and congenital BM failure syndromes, according to international criteria. Samples from 19 (14 male and 5 female) age-matched (range 2056 years) healthy controls were obtained after they had signed the written informed consent form described above.'\n","Skipping entity [8264, 8267, MedTech] in the following text because the character span 'PBS' does not align with token boundaries:\n","'Analysis of variance in conjunction with Students t test was performed to identify significant differences. All analyses were performed using GraphPad Prism 6.0 software (GraphPad, La Jolla, CA, USA). CD106+ MSCs or CD106 MSCs at passage 4 were stimulated by incubation with TNF- (PepTech, Burlington, MA, USA) 10\\xa0ng/ml for 1\\xa0h, 4\\xa0h, or 24\\xa0h, or 500\\xa0ng/ml LPS (Sigma-Aldrich) for 0.5\\xa0h, 1\\xa0h, 2\\xa0h, or 4\\xa0h. Cells were lysed in Bicine/CHAPS Lysis Buffer (ProteinSimple) supplemented with DMSO Inhibitor Mix (ProteinSimple) and Aqueous Inhibitor Mix (ProteinSimple) at 4\\xa0°C for 30\\xa0min, and the lysate was mixed with Premix G2 (pH\\xa0310) (ProteinSimple) and pI Standard Ladder 3 (44:1, ProteinSimple). The rabbit anti-NF-B (p65) antibody (primary antibody, Cell Signaling Technology) was diluted 1:50 in antibody dilution buffer, and the anti-human IgG-HRP (secondary antibody, Protein Simple) was diluted 1:100 in antibody dilution buffer. Luminol/peroxide was mixed at a 1:1 ratio. The NanoPro 1000 (ProteinSimple) was loaded and run according to the manufacturers specifications. Emitted light was quantified for 30\\xa0s, 60\\xa0s, 120\\xa0s, 240\\xa0s, 480\\xa0s, and 960\\xa0s. Compass software 2.5.11 (ProteinSimple) was used to identify and quantify chemiluminescence peaks and optimize tracings. BM-MSCs were blocked by incubation with 500\\xa0ng/ml of the CD106 blocking antibody ab47159 for 1\\xa0h at 4\\xa0°C. BM-MSCs were blocked by incubation with 10 nM of the NF-B-specific inhibitor BAY-11-7082 for 0.5\\xa0h, 1\\xa0h, or 12\\xa0h. Aliquots of cell preparations containing 2105 CD106+ MSCs or CD106 MSCs and 1105 UCB CD34+ cells in 15\\xa0l of Roswell Park Memorial Institute (RPMI) 1640 medium were injected into the tail vein of 28- to 35-day-old NOD/SCID irradiated mice (n=6 for each pair of cells) using a Hamilton syringe. Some low-dose irradiated mice (320\\xa0cGy) were sacrificed 42\\xa0days after xenotransplantation. CD45+ cells were analyzed after staining with human anti-CD45-APC antibody along with the appropriate isotype controls (BD Biosciences) according to the manufacturers instructions. Some high-dose irradiated mice (360\\xa0cGy) (n=6 for each pair of cells) were observed until 56\\xa0days after xenotransplantation or death. Cocultures of CD34+ cells with MSCs were incubated for 14\\xa0days. Nonadherent viable cells (5102 in each well of a 24-well-plate) were plated on semisolid Iscoves modified Dulbeccos medium (IMDM; Gibco) supplemented with 1% methylcellulose, 10% fetal bovine serum (FBS), 1% BSA, 104 M mercaptoethanol, 2\\xa0mMl-glutamine, and 100\\xa0ng/ml thrombopoietin. Cultures were incubated at 37\\xa0°C in a humidified atmosphere with 5% CO2. CFU-MK colonies were identified after 14\\xa0days of culture under an Olympus IX71 inverted microscope (Olympus), and typical colonies were selected for immunostaining. Cell smears were prepared using Cytospin (Thermo-Fisher Scientific), stained with anti-CD41a antibody, and observed using an UltraVIEWVoX Confocal Imaging System (PerkinElmer). Cocultures of CD34+ cells with MSCs or blocked MSCs were incubated for 14\\xa0days or 10\\xa0days. Nonadherent viable cells (2.5102 or 5102 in each well of a 24-well-plate) were plated on 0.5\\xa0ml of methylcellulose medium (Stem Cell Technologies) to evaluate the in vitro effects of MSCs on CFU growth. Colonies of >50 cells were scored after 14\\xa0days of culture. Experiments were performed in triplicate. A total of 1105 CD34+ cells suspended in serum-free StemSpan\\x84¢ H3000 (Stem Cell Technologies) culture medium was applied to feeder layers composed of BM-MSCs, which were blocked by BAY-11-7082, as described previously. Cocultures were incubated for 10\\xa0days, and the culture medium was replenished every 3.5\\xa0days. Nonadherent viable cells were stained for FCM analysis using CD34-APC along with the appropriate isotype controls (BD Biosciences) according to the manufacturers instructions. Data acquisition was also performed using an LSR II flow cytometer (BD Biosciences) and analyzed with FlowJo7.6 software (FlowJo). A total of 5104 CD34+ cells suspended in 1\\xa0ml of serum-free StemSpan\\x84¢ H3000 (Stem Cell Technologies, Vancouver, Canada) culture medium was applied to feeder layers composed of 5104 BM-MSCs, as described previously. Cocultures were incubated for 14\\xa0days, and the culture medium was replenished every 3.5\\xa0days. Nonadherent viable cells were stained for FCM analysis using the antibodies anti-CD34-APC, anti-CD61-PE, anti-CD41a-PE, and anti-CD42b-PE, along with the appropriate isotype controls (BD Biosciences) according to the manufacturers instructions. Data acquisition was performed using an LSR II flow cytometer (BD Biosciences) and analyzed with FlowJo7.6 software (FlowJo). CD34+ cells were freshly purified from umbilical cord blood (UCB) using a CD34/MACS isolation kit (MiltenyiBiotec, Bergisch, Gladbach, Germany) according to the manufacturers instructions. Cell fractions with 955% CD34+ cell purity was used for subsequent experiments. Supernatants obtained from MSC-conditioned medium were used to detect vascular endothelial growth factor (VEGF) levels using an enzyme-linked immunosorbent assay (ELISA; R&D Systems, Minneapolis, MN, USA) according to the manufacturers instructions. To confirm our in vitro data, we examined vasculogenesis in vivo by performing a Matrigel plug assay. Aliquots of 5105 MSCs were resuspended in 500\\xa0l of Matrigel (BD Biosciences) according to the manufacturers instructions and implanted into the back of 42-day-old nude mice (n=6 in each group). Mice implanted with Matrigel only were used as negative controls. After 21\\xa0days, the Matrigel plugs were harvested, assayed for microvessels identified as luminal structures with red blood cells using H&E staining, and counted. Wells in 96-well plates were covered with 50\\xa0l of growth factor-reduced Matrigel (BD Biosciences). Aliquots of CD106+ MSCs and CD106 MSCs were seeded at a density of 10,000 cells/cm2 and cultured in a humidified atmosphere with 5% CO2 for 24\\xa0h. The formation of capillary-like structures was observed using an Olympus IX71 inverted microscope (Olympus, Tokyo, Japan), and pictures were taken at different time points using an Olympus DP71 camera (Olympus). Western blotting procedures were performed according to the protocol described by Song et al.. Briefly, BM-MSCs were collected, washed, and lysed with RIPA lysis buffer (Beyotime Institute of Biotechnology, Shanghai, China) supplemented with PMSF (Invitrogen, Carlsbad, CA, USA). Total protein was extracted and quantified by the BCA protein assay kit (Pierce, Woodland Hills, CA, USA). A total of 30\\xa0g protein was denatured, separated by SDS-PAGE electrophoresis, and transferred to a PVDF membrane. The transferred membranes were blocked using 5% bovine serum albumin (BSA) in TBST, incubated with anti-human CD106 mouse monoclonal antibody (Abcam) overnight, and then incubated with the corresponding horseradish peroxidase (HRP)-conjugated secondary antibody at a dilution of 1:2000 for 2\\xa0h. Bands were visualized using enhanced chemiluminescence (ECL; Thermo-Fisher, Scientific, Waltham, MA, USA) detection reagents, and scanned images were quantified using Image J (https://imagej.nih.gov/ij/). The ratio of target gene to -actin was used for the semiquantification and comparison between the two groups. Samples were photographed using an UltraVIEWVoX Confocal Imaging System (PerkinElmer, Waltham, MA, USA). Selected CD106+/CD106 MSCs were stained for NF-B and nuclei to examine the NF-B expression level. The expression of NF-B was calculated as the mean of the fluorescence intensity in 6 continuous views. CD106+/CD106 MSCs were first washed and fixed with 4% formaldehyde for 15\\xa0min and then blocked with blocking buffer (PBS/5% normal serum/0.3% Triton\\x84¢ X-100) for 60\\xa0min. Then, CD106+/CD106 MSCs were labeled with rabbit-anti-human NF-B overnight and then labeled with Alexa Fluor 488-conjugated secondary antibody or Alexa Fluor 546-conjugated secondary antibody for 60\\xa0min. The nucleus was marked with DAPI. The expression of cell surface molecules was assessed according to the manufacturers instructions. Normal MSCs (N MSCs) were stained for CD106 and NF-kB. N MSCs were first washed and fixed with 4% formaldehyde for 15\\xa0min and then blocked with blocking buffer (phosphate-buffered saline (PBS)/5% normal serum) for 60\\xa0min. N MSCs were labeled with mouse-anti-human CD106 overnight and then labeled with Alexa Fluor 546-goat anti-mouse IgG conjugated secondary antibody for 60\\xa0min. N MSCs with 0.3% Triton\\x84¢ X-100 for 60\\xa0min were labeled with rabbit-anti-human NF-B overnight and then labeled with Alexa Fluor 488-conjugated secondary antibody (donkey anti-rabbit) for 60\\xa0min. The nucleus was marked with DAPI. H&E staining (Sigma-Aldrich) and histochemical staining (Abcam, Cambridge, UK) were performed according to the manufacturers instructions. Samples were photographed using a Nikon ElipseTi-U microscope (Nikon, Tokyo, Japan). Cells were stained with antibodies along with the appropriate isotype controls (BD Biosciences) according to the manufacturers instructions. Data acquisition was performed using an LSR II flow cytometer (BD Biosciences) and analyzed with FlowJo 7.6 software (FlowJo, Ashland, OR, USA). MSCs were labeled with PE-conjugated anti-CD106 antibody (BD Biosciences). CD106+ MSCs and CD106 MSCs were separated using a CD106-positive selection magnetic-activated cell sorting (MACS) isolation kit (Miltenyi Biotech, Bergisch Gladbach, Germany) according to the manufacturers instructions. CD106+ MSCs or CD106 MSCs (¥90% purity) were used for subsequent experiments. RNA samples were first treated with DNase I to degrade any possible DNA contamination. Next, mRNA was enriched using oligo (dT) magnetic beads for eukaryotes and fragmented into short fragments of approximately 200\\xa0bp. The first strand of cDNA was synthesized using a random hexamer-primer, and then buffer, dNTPs, RNase H, and DNA polymerase I were added to synthesize the second strand. Double-stranded cDNA was purified with magnetic beads, followed by end reparation and 3-end single nucleotide A (adenine) addition. Finally, sequencing adaptors were ligated to the fragments, which were enriched using polymerase chain reaction (PCR) amplification. A sample library was qualified and quantified using an Agilent 2100 Bioanalyzer (Agilent, Santa Clara, CA, USA) and an ABI StepOnePlus Real-Time PCR System (Applied Biosystems, Carlsbad, CA, USA) during the quantitative-competitive (QC) step. Library products were ready for sequencing via Illumina HiSeqTM 2000 (Illumina, San Diego, CA, USA) or other sequencer when necessary. Next, quantitative real-time polymerase chain reaction (qRT-PCR) was performed to confirm the gene expression levels of RNA transcripts with sequence-specific oligonucleotide primers as described previously. BM-MSCs were isolated and cultured in Dulbeccos modified Eagles medium. BM-MSCs were identified by their surface markers with a panel of monoclonal antibodies against CD13 (WM15), CD29 (MAR4), CD44 (G44-26), CD49e (IIA1), CD73 (AD2), CD105 (266), CD166 (3A6), CD31 (WM59), CD34 (581), CD45 (HI30), CD90 (5E10), HLA-ABC (G46-2.6), HLA-DR (G46-6), CD14 (M5E2), CD40 (5C3), and CD11b (ICRF44), along with the appropriate isotype monoclonal antibodies using a FACScanflow cytometer (BD Biosciences, San Jose, CA, USA). Our experimental research on NOD/SCID and nude mice followed internationally recognized guidelines. Ethical approval for the animal experiments was provided by the Ethical Committee of the Institute of Hematology and Blood Diseases Hospital, Chinese Academy of Medical Science and Peking Union Medical College. The ethical approval documents reference number is KT2012003-m-6. BM samples from 28 (17 male and 11 female) de novo acquired AA patients with a median age of 31\\xa0years (range 1859 years) were analyzed after the signing of a written informed consent form in accordance with the Declaration of Helsinki. The study was approved by the Committee for Medical Care and Safety, Institute of Hematology and Blood Diseases Hospital, Chinese Academy of Medical Science and Peking Union Medical College (ethical approval documents reference number KT2014005-EC-1). This cohort consisted of four patients with severe AA and 24 with nonsevere AA. The diagnosis and severity classification of AA was established by morphological examination of the BM and peripheral blood samples after excluding any other acquired BM failure syndromes, such as paroxysmal nocturnal hemoglobinuria, myelodysplastic syndrome, and congenital BM failure syndromes, according to international criteria. Samples from 19 (14 male and 5 female) age-matched (range 2056 years) healthy controls were obtained after they had signed the written informed consent form described above.'\n","done\n"]}],"source":["import glob\n","import json\n","import os\n","import spacy\n","from spacy.tokens import DocBin\n","import itertools\n","\n","#ATTENTION: please adjust all paths if use the code\n","\n","#a path to the directory with further data directories. Each subdirectory contains 10 annotated texts\n","path_to_labelled_texts = \"/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts\"\n","\n","def get_file_paths():\n","  \"\"\"\n","  collect all JSON-file names in the directory\n","  AnnotatedTexts\n","  \"\"\"\n","  dev_files = []\n","  train_files = []\n","  start_dir = path_to_labelled_texts\n","  #get all subdirectories that contain annotated data for the dev-set\n","  dev_subdirs = [\"/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/100-120\",\n","                 \"/content/drive/MyDrive/TextcorpusCreation/ValidatedTexts/140-151\"]                \n","\n","  #get all subdirectories that contain annotated data for train-set\n","  train_subdirs = [x[0] for x in os.walk(start_dir) if x[0] != start_dir and not x[0].endswith(\"ipynb_checkpoints\") and x[0] not in dev_subdirs] \n","  print(train_subdirs)\n","  \n","  #collect texts for spacy dev-set\n","  for item in dev_subdirs:   \n","    text_files = [f for f in os.listdir(item) if f.endswith('.json')]\n","    abs_paths = [item + \"/\" + f for f in text_files]    \n","    dev_files.extend(abs_paths)\n","  print(dev_files)\n","\n","  #collect texts for spacy train-set\n","  for item in train_subdirs:   \n","    text_files = [f for f in os.listdir(item) if f.endswith('.json')]\n","    abs_paths = [item + \"/\" + f for f in text_files]    \n","    train_files.extend(abs_paths)\n","  print(train_files)\n","\n","  return dev_files, train_files\n","\n","def read_data(path_to_file: str):\n","  \"\"\"\n","  path_to_file: path to a LS Json file\n","  read JSON file and save it as a dictionary\n","  \"\"\"\n","  with open(path_to_file) as f:\n","    data = f.read().strip()\n","    text_info = json.loads(data)    \n","  return text_info\n","\n","def get_entity_positions(ls_json):\n","  \"\"\"\n","  ls_json: content of a Label Studio Json file\n","  determine start and end position of each IVD concept in the text and\n","  save this information of the form (start_index, end_index, label) \n","  in the list\n","  return: text + list of (start_index, end_index) for each concept in this text\n","  \"\"\"\n","  label = \"MedTech\"\n","  total_text = \"\"  \n","  entities = []    \n","  total_size = 0\n","  for par in ls_json: \n","    #text paragraph    \n","    data = par[\"data\"]\n","    text = data[\"text\"].strip() + \" \"\n","    total_text += text   \n","    annotations = par[\"annotations\"]\n","    for annot in annotations:\n","      result = annot[\"result\"]\n","      if(len(result) > 0): \n","        for res in result:          \n","          value = res[\"value\"]                   \n","          start = value[\"start\"]\n","          end = value[\"end\"]\n","          total_start = total_size + start          \n","          total_end = total_size + end          \n","          entry = (total_start, total_end, label)\n","          entities.append(entry)\n","      total_size = len(total_text)\n","  return (total_text.strip(), entities)\n","\n","def correct_entity_positions(doc, start, end, label):\n","  \"\"\"\n","  correct wrongly defined positions of concepts\n","  \"\"\"\n","  span = doc.char_span(start, end, label=label)\n","  if span is not None:\n","    return span\n","  span = doc.char_span(start, end + 1, label=label)\n","  if span is not None:\n","    return span\n","  span = doc.char_span(start, end - 1, label=label)\n","  if span is not None:\n","    return span\n","  span = doc.char_span(start + 1, end, label=label)\n","  if span is not None:\n","    return span\n","  span = doc.char_span(start - 1, end, label=label)\n","  if span is not None:\n","    return span\n","  print(f\"Skipping entity [{start}, {end}, {label}] in the following text because the character span '{doc.text[start:end]}' does not align with token boundaries:\\n{repr(text)}\")\n","  return None\n","\n","def remove_doublicated_entries(annotations):\n","  \"\"\"\n","  remove overlapping entities and dublicates\n","  \"\"\"\n","  to_remove = []\n","  for a, b in itertools.combinations(annotations, 2):\n","    if a[0]<=b[0] and a[1]>=b[1]:\n","      to_remove.append(b)\n","      continue\n","    if a[0]>=b[0] and a[1]<=b[1]:\n","      to_remove.append(a)\n","      continue\n","    if a[0]<b[0] and a[1]>b[0]:\n","      to_remove.append(a)\n","      continue\n","    if a[0]>b[0] and a[1]<b[0]:\n","      to_remove.append(a)\n","\n","  for item in to_remove:\n","    if item in entity_annot:\n","      entity_annot.remove(item)\n","  return annotations\n","\n","#start point\n","train_data_path = \"/content/drive/MyDrive/SpacyData/textkorpus150/val_train.spacy\"\n","dev_data_path = \"/content/drive/MyDrive/SpacyData/textkorpus150/val_dev.spacy\"\n","dev_data, train_data = get_file_paths()\n","spacy_dev_items = []\n","spacy_train_items = []\n","\n","nlp = spacy.blank(\"en\")\n","db_train = DocBin()\n","db_dev = DocBin()\n","\n","#create dev-set\n","for item in dev_data:  \n","  total_text = \"\"  \n","  entities = []    \n","  total_size = 0\n","  #read LS JSON\n","  info = read_data(item)  \n","  #create items for spacy data format\n","  text, entities = get_entity_positions(info)\n","  #create span-index item\n","  spacy_item = [text, {\"entities\" : entities}]  \n","  spacy_dev_items.append(spacy_item)\n","\n","for text, annotations in spacy_dev_items:\n","  #process annotations  \n","  doc = nlp.make_doc(text) \n","  ents = []\n","  entity_annot = annotations[\"entities\"]\n","  corrected_entities = remove_doublicated_entries(entity_annot)    \n","  for start, end, label in entity_annot:    \n","    span = correct_entity_positions(doc, start, end, label)     \n","    if span is not None:\n","      ents.append(span)   \n","  doc.ents = ents\n","  db_dev.add(doc)\n","#save data in SpaCy format\n","db_dev.to_disk(dev_data_path)\n","\n","\n","#create train-set\n","for item in train_data:  \n","  total_text = \"\"  \n","  entities = []    \n","  total_size = 0\n","  #read LS JSON\n","  info = read_data(item)  \n","  #create items for spacy data format\n","  text, entities = get_entity_positions(info)\n","  #create span-index item\n","  spacy_item = [text, {\"entities\" : entities}]  \n","  spacy_train_items.append(spacy_item)\n","\n","for text, annotations in spacy_train_items:\n","  #process annotations  \n","  doc = nlp.make_doc(text) \n","  ents = []\n","  entity_annot = annotations[\"entities\"]\n","  corrected_entities = remove_doublicated_entries(entity_annot)    \n","  for start, end, label in entity_annot:    \n","    span = correct_entity_positions(doc, start, end, label)     \n","    if span is not None:\n","      ents.append(span)   \n","  doc.ents = ents\n","  db_train.add(doc)\n","#save data in SpaCy format\n","db_train.to_disk(train_data_path)\n","print(\"done\")"]},{"cell_type":"markdown","metadata":{"id":"OUeMVMx7TXj6"},"source":["Download necessary libs for training. Different variants are tried because of errors"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"GtlMi9ci97FY","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1670400340804,"user_tz":-60,"elapsed":330550,"user":{"displayName":"Victoria Dimanova","userId":"07172063658264964881"}},"outputId":"1fb215e2-7ff1-4ba8-f427-3d1da475bc63"},"outputs":[{"output_type":"stream","name":"stdout","text":["Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/\n","Looking in links: https://download.pytorch.org/whl/torch_stable.html\n","Collecting torch==1.10.0+cu111\n","  Downloading https://download.pytorch.org/whl/cu111/torch-1.10.0%2Bcu111-cp38-cp38-linux_x86_64.whl (2137.6 MB)\n","\u001b[K     |████████████▌                   | 834.1 MB 1.4 MB/s eta 0:15:57tcmalloc: large alloc 1147494400 bytes == 0x3926e000 @  0x7f7a81f9f615 0x5d631c 0x51e4f1 0x51e67b 0x4f7585 0x49ca7c 0x4fdff5 0x49caa1 0x4fdff5 0x49ced5 0x4f60a9 0x55f926 0x4f60a9 0x55f926 0x4f60a9 0x55f926 0x5d7c18 0x5d9412 0x586636 0x5d813c 0x55f3fd 0x55e571 0x5d7cf1 0x49ced5 0x55e571 0x5d7cf1 0x49ec69 0x5d7c18 0x49ca7c 0x4fdff5 0x49ced5\n","\u001b[K     |███████████████▉                | 1055.7 MB 1.4 MB/s eta 0:12:37tcmalloc: large alloc 1434370048 bytes == 0x7d8c4000 @  0x7f7a81f9f615 0x5d631c 0x51e4f1 0x51e67b 0x4f7585 0x49ca7c 0x4fdff5 0x49caa1 0x4fdff5 0x49ced5 0x4f60a9 0x55f926 0x4f60a9 0x55f926 0x4f60a9 0x55f926 0x5d7c18 0x5d9412 0x586636 0x5d813c 0x55f3fd 0x55e571 0x5d7cf1 0x49ced5 0x55e571 0x5d7cf1 0x49ec69 0x5d7c18 0x49ca7c 0x4fdff5 0x49ced5\n","\u001b[K     |████████████████████            | 1336.2 MB 1.3 MB/s eta 0:10:07tcmalloc: large alloc 1792966656 bytes == 0x26f6000 @  0x7f7a81f9f615 0x5d631c 0x51e4f1 0x51e67b 0x4f7585 0x49ca7c 0x4fdff5 0x49caa1 0x4fdff5 0x49ced5 0x4f60a9 0x55f926 0x4f60a9 0x55f926 0x4f60a9 0x55f926 0x5d7c18 0x5d9412 0x586636 0x5d813c 0x55f3fd 0x55e571 0x5d7cf1 0x49ced5 0x55e571 0x5d7cf1 0x49ec69 0x5d7c18 0x49ca7c 0x4fdff5 0x49ced5\n","\u001b[K     |█████████████████████████▎      | 1691.1 MB 1.3 MB/s eta 0:05:40tcmalloc: large alloc 2241208320 bytes == 0x6d4de000 @  0x7f7a81f9f615 0x5d631c 0x51e4f1 0x51e67b 0x4f7585 0x49ca7c 0x4fdff5 0x49caa1 0x4fdff5 0x49ced5 0x4f60a9 0x55f926 0x4f60a9 0x55f926 0x4f60a9 0x55f926 0x5d7c18 0x5d9412 0x586636 0x5d813c 0x55f3fd 0x55e571 0x5d7cf1 0x49ced5 0x55e571 0x5d7cf1 0x49ec69 0x5d7c18 0x49ca7c 0x4fdff5 0x49ced5\n","\u001b[K     |████████████████████████████████| 2137.6 MB 48.2 MB/s eta 0:00:01tcmalloc: large alloc 2137636864 bytes == 0xf2e40000 @  0x7f7a81f9e1e7 0x4d3280 0x4d330c 0x5d631c 0x51e4f1 0x51e67b 0x4f7585 0x49ca7c 0x55e571 0x5d7cf1 0x49ca7c 0x55e571 0x5d7cf1 0x49ca7c 0x55e571 0x5d7cf1 0x49ca7c 0x55e571 0x5d7cf1 0x49ca7c 0x55e571 0x5d7cf1 0x49ca7c 0x5d7c18 0x49ca7c 0x55e571 0x5d7cf1 0x49ced5 0x4fdff5 0x49ced5 0x55e571\n","tcmalloc: large alloc 2672050176 bytes == 0x1e698a000 @  0x7f7a81f9f615 0x5d631c 0x51e4f1 0x51e67b 0x4f7585 0x49ca7c 0x55e571 0x5d7cf1 0x49ca7c 0x55e571 0x5d7cf1 0x49ca7c 0x55e571 0x5d7cf1 0x49ca7c 0x55e571 0x5d7cf1 0x49ca7c 0x55e571 0x5d7cf1 0x49ca7c 0x5d7c18 0x49ca7c 0x55e571 0x5d7cf1 0x49ced5 0x4fdff5 0x49ced5 0x55e571 0x5d7cf1 0x4fea58\n","\u001b[K     |████████████████████████████████| 2137.6 MB 6.7 kB/s \n","\u001b[?25hRequirement already satisfied: typing-extensions in /usr/local/lib/python3.8/dist-packages (from torch==1.10.0+cu111) (4.1.1)\n","Installing collected packages: torch\n","  Attempting uninstall: torch\n","    Found existing installation: torch 1.12.1+cu113\n","    Uninstalling torch-1.12.1+cu113:\n","      Successfully uninstalled torch-1.12.1+cu113\n","\u001b[31mERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\n","torchvision 0.13.1+cu113 requires torch==1.12.1, but you have torch 1.10.0+cu111 which is incompatible.\n","torchtext 0.13.1 requires torch==1.12.1, but you have torch 1.10.0+cu111 which is incompatible.\n","torchaudio 0.12.1+cu113 requires torch==1.12.1, but you have torch 1.10.0+cu111 which is incompatible.\u001b[0m\n","Successfully installed torch-1.10.0+cu111\n","Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/\n","Collecting spacy[cuda111,transformers]==3.2.0\n","  Downloading spacy-3.2.0-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (6.2 MB)\n","\u001b[K     |████████████████████████████████| 6.2 MB 4.4 MB/s \n","\u001b[?25hRequirement already satisfied: spacy-loggers<2.0.0,>=1.0.0 in /usr/local/lib/python3.8/dist-packages (from spacy[cuda111,transformers]==3.2.0) (1.0.3)\n","Requirement already satisfied: langcodes<4.0.0,>=3.2.0 in /usr/local/lib/python3.8/dist-packages (from spacy[cuda111,transformers]==3.2.0) (3.3.0)\n","Requirement already satisfied: numpy>=1.15.0 in /usr/local/lib/python3.8/dist-packages (from spacy[cuda111,transformers]==3.2.0) (1.21.6)\n","Requirement already satisfied: catalogue<2.1.0,>=2.0.6 in /usr/local/lib/python3.8/dist-packages (from spacy[cuda111,transformers]==3.2.0) (2.0.8)\n","Collecting pydantic!=1.8,!=1.8.1,<1.9.0,>=1.7.4\n","  Downloading pydantic-1.8.2-cp38-cp38-manylinux2014_x86_64.whl (13.7 MB)\n","\u001b[K     |████████████████████████████████| 13.7 MB 36.2 MB/s \n","\u001b[?25hRequirement already satisfied: requests<3.0.0,>=2.13.0 in /usr/local/lib/python3.8/dist-packages (from spacy[cuda111,transformers]==3.2.0) (2.23.0)\n","Requirement already satisfied: pathy>=0.3.5 in /usr/local/lib/python3.8/dist-packages (from spacy[cuda111,transformers]==3.2.0) (0.9.0)\n","Collecting typer<0.5.0,>=0.3.0\n","  Downloading typer-0.4.2-py3-none-any.whl (27 kB)\n","Requirement already satisfied: blis<0.8.0,>=0.4.0 in /usr/local/lib/python3.8/dist-packages (from spacy[cuda111,transformers]==3.2.0) (0.7.9)\n","Requirement already satisfied: wasabi<1.1.0,>=0.8.1 in /usr/local/lib/python3.8/dist-packages (from spacy[cuda111,transformers]==3.2.0) (0.10.1)\n","Requirement already satisfied: srsly<3.0.0,>=2.4.1 in /usr/local/lib/python3.8/dist-packages (from spacy[cuda111,transformers]==3.2.0) (2.4.5)\n","Requirement already satisfied: jinja2 in /usr/local/lib/python3.8/dist-packages (from spacy[cuda111,transformers]==3.2.0) (2.11.3)\n","Requirement already satisfied: preshed<3.1.0,>=3.0.2 in /usr/local/lib/python3.8/dist-packages (from spacy[cuda111,transformers]==3.2.0) (3.0.8)\n","Collecting thinc<8.1.0,>=8.0.12\n","  Downloading thinc-8.0.17-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (671 kB)\n","\u001b[K     |████████████████████████████████| 671 kB 62.1 MB/s \n","\u001b[?25hRequirement already satisfied: tqdm<5.0.0,>=4.38.0 in /usr/local/lib/python3.8/dist-packages (from spacy[cuda111,transformers]==3.2.0) (4.64.1)\n","Requirement already satisfied: murmurhash<1.1.0,>=0.28.0 in /usr/local/lib/python3.8/dist-packages (from spacy[cuda111,transformers]==3.2.0) (1.0.9)\n","Requirement already satisfied: packaging>=20.0 in /usr/local/lib/python3.8/dist-packages (from spacy[cuda111,transformers]==3.2.0) (21.3)\n","Requirement already satisfied: spacy-legacy<3.1.0,>=3.0.8 in /usr/local/lib/python3.8/dist-packages (from spacy[cuda111,transformers]==3.2.0) (3.0.10)\n","Requirement already satisfied: cymem<2.1.0,>=2.0.2 in /usr/local/lib/python3.8/dist-packages (from spacy[cuda111,transformers]==3.2.0) (2.0.7)\n","Requirement already satisfied: setuptools in /usr/local/lib/python3.8/dist-packages (from spacy[cuda111,transformers]==3.2.0) (57.4.0)\n","Collecting cupy-cuda111<10.0.0,>=5.0.0b4\n","  Downloading cupy_cuda111-9.6.0-cp38-cp38-manylinux1_x86_64.whl (97.9 MB)\n","\u001b[K     |████████████████████████████████| 97.9 MB 1.2 MB/s \n","\u001b[?25hCollecting spacy-transformers<1.2.0,>=1.1.2\n","  Downloading spacy_transformers-1.1.8-py2.py3-none-any.whl (53 kB)\n","\u001b[K     |████████████████████████████████| 53 kB 1.8 MB/s \n","\u001b[?25hRequirement already satisfied: fastrlock>=0.5 in /usr/local/lib/python3.8/dist-packages (from cupy-cuda111<10.0.0,>=5.0.0b4->spacy[cuda111,transformers]==3.2.0) (0.8.1)\n","Requirement already satisfied: pyparsing!=3.0.5,>=2.0.2 in /usr/local/lib/python3.8/dist-packages (from packaging>=20.0->spacy[cuda111,transformers]==3.2.0) (3.0.9)\n","Requirement already satisfied: smart-open<6.0.0,>=5.2.1 in /usr/local/lib/python3.8/dist-packages (from pathy>=0.3.5->spacy[cuda111,transformers]==3.2.0) (5.2.1)\n","Requirement already satisfied: typing-extensions>=3.7.4.3 in /usr/local/lib/python3.8/dist-packages (from pydantic!=1.8,!=1.8.1,<1.9.0,>=1.7.4->spacy[cuda111,transformers]==3.2.0) (4.1.1)\n","Requirement already satisfied: chardet<4,>=3.0.2 in /usr/local/lib/python3.8/dist-packages (from requests<3.0.0,>=2.13.0->spacy[cuda111,transformers]==3.2.0) (3.0.4)\n","Requirement already satisfied: idna<3,>=2.5 in /usr/local/lib/python3.8/dist-packages (from requests<3.0.0,>=2.13.0->spacy[cuda111,transformers]==3.2.0) (2.10)\n","Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.8/dist-packages (from requests<3.0.0,>=2.13.0->spacy[cuda111,transformers]==3.2.0) (2022.9.24)\n","Requirement already satisfied: urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 in /usr/local/lib/python3.8/dist-packages (from requests<3.0.0,>=2.13.0->spacy[cuda111,transformers]==3.2.0) (1.24.3)\n","  Downloading spacy_transformers-1.1.7-py2.py3-none-any.whl (53 kB)\n","\u001b[K     |████████████████████████████████| 53 kB 2.1 MB/s \n","\u001b[?25hCollecting transformers<4.21.0,>=3.4.0\n","  Downloading transformers-4.20.1-py3-none-any.whl (4.4 MB)\n","\u001b[K     |████████████████████████████████| 4.4 MB 46.4 MB/s \n","\u001b[?25hCollecting spacy-alignments<1.0.0,>=0.7.2\n","  Downloading spacy_alignments-0.8.6-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (1.1 MB)\n","\u001b[K     |████████████████████████████████| 1.1 MB 58.4 MB/s \n","\u001b[?25hRequirement already satisfied: torch>=1.6.0 in /usr/local/lib/python3.8/dist-packages (from spacy-transformers<1.2.0,>=1.1.2->spacy[cuda111,transformers]==3.2.0) (1.10.0+cu111)\n","Requirement already satisfied: filelock in /usr/local/lib/python3.8/dist-packages (from transformers<4.21.0,>=3.4.0->spacy-transformers<1.2.0,>=1.1.2->spacy[cuda111,transformers]==3.2.0) (3.8.0)\n","Collecting tokenizers!=0.11.3,<0.13,>=0.11.1\n","  Downloading tokenizers-0.12.1-cp38-cp38-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (6.6 MB)\n","\u001b[K     |████████████████████████████████| 6.6 MB 35.0 MB/s \n","\u001b[?25hRequirement already satisfied: pyyaml>=5.1 in /usr/local/lib/python3.8/dist-packages (from transformers<4.21.0,>=3.4.0->spacy-transformers<1.2.0,>=1.1.2->spacy[cuda111,transformers]==3.2.0) (6.0)\n","Collecting huggingface-hub<1.0,>=0.1.0\n","  Downloading huggingface_hub-0.11.1-py3-none-any.whl (182 kB)\n","\u001b[K     |████████████████████████████████| 182 kB 72.0 MB/s \n","\u001b[?25hRequirement already satisfied: regex!=2019.12.17 in /usr/local/lib/python3.8/dist-packages (from transformers<4.21.0,>=3.4.0->spacy-transformers<1.2.0,>=1.1.2->spacy[cuda111,transformers]==3.2.0) (2022.6.2)\n","Requirement already satisfied: click<9.0.0,>=7.1.1 in /usr/local/lib/python3.8/dist-packages (from typer<0.5.0,>=0.3.0->spacy[cuda111,transformers]==3.2.0) (7.1.2)\n","Requirement already satisfied: MarkupSafe>=0.23 in /usr/local/lib/python3.8/dist-packages (from jinja2->spacy[cuda111,transformers]==3.2.0) (2.0.1)\n","Installing collected packages: typer, pydantic, tokenizers, thinc, huggingface-hub, transformers, spacy-alignments, spacy, spacy-transformers, cupy-cuda111\n","  Attempting uninstall: typer\n","    Found existing installation: typer 0.7.0\n","    Uninstalling typer-0.7.0:\n","      Successfully uninstalled typer-0.7.0\n","  Attempting uninstall: pydantic\n","    Found existing installation: pydantic 1.10.2\n","    Uninstalling pydantic-1.10.2:\n","      Successfully uninstalled pydantic-1.10.2\n","  Attempting uninstall: thinc\n","    Found existing installation: thinc 8.1.5\n","    Uninstalling thinc-8.1.5:\n","      Successfully uninstalled thinc-8.1.5\n","  Attempting uninstall: spacy\n","    Found existing installation: spacy 3.4.3\n","    Uninstalling spacy-3.4.3:\n","      Successfully uninstalled spacy-3.4.3\n","\u001b[31mERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\n","en-core-web-sm 3.4.1 requires spacy<3.5.0,>=3.4.0, but you have spacy 3.2.0 which is incompatible.\u001b[0m\n","Successfully installed cupy-cuda111-9.6.0 huggingface-hub-0.11.1 pydantic-1.8.2 spacy-3.2.0 spacy-alignments-0.8.6 spacy-transformers-1.1.7 thinc-8.0.17 tokenizers-0.12.1 transformers-4.20.1 typer-0.4.2\n","Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/\n","Requirement already satisfied: transformers[sentencepiece] in /usr/local/lib/python3.8/dist-packages (4.20.1)\n","Requirement already satisfied: pyyaml>=5.1 in /usr/local/lib/python3.8/dist-packages (from transformers[sentencepiece]) (6.0)\n","Requirement already satisfied: tokenizers!=0.11.3,<0.13,>=0.11.1 in /usr/local/lib/python3.8/dist-packages (from transformers[sentencepiece]) (0.12.1)\n","Requirement already satisfied: requests in /usr/local/lib/python3.8/dist-packages (from transformers[sentencepiece]) (2.23.0)\n","Requirement already satisfied: regex!=2019.12.17 in /usr/local/lib/python3.8/dist-packages (from transformers[sentencepiece]) (2022.6.2)\n","Requirement already satisfied: filelock in /usr/local/lib/python3.8/dist-packages (from transformers[sentencepiece]) (3.8.0)\n","Requirement already satisfied: huggingface-hub<1.0,>=0.1.0 in /usr/local/lib/python3.8/dist-packages (from transformers[sentencepiece]) (0.11.1)\n","Requirement already satisfied: tqdm>=4.27 in /usr/local/lib/python3.8/dist-packages (from transformers[sentencepiece]) (4.64.1)\n","Requirement already satisfied: numpy>=1.17 in /usr/local/lib/python3.8/dist-packages (from transformers[sentencepiece]) (1.21.6)\n","Requirement already satisfied: packaging>=20.0 in /usr/local/lib/python3.8/dist-packages (from transformers[sentencepiece]) (21.3)\n","Requirement already satisfied: protobuf<=3.20.1 in /usr/local/lib/python3.8/dist-packages (from transformers[sentencepiece]) (3.19.6)\n","Collecting sentencepiece!=0.1.92,>=0.1.91\n","  Downloading sentencepiece-0.1.97-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (1.3 MB)\n","\u001b[K     |████████████████████████████████| 1.3 MB 4.7 MB/s \n","\u001b[?25hRequirement already satisfied: typing-extensions>=3.7.4.3 in /usr/local/lib/python3.8/dist-packages (from huggingface-hub<1.0,>=0.1.0->transformers[sentencepiece]) (4.1.1)\n","Requirement already satisfied: pyparsing!=3.0.5,>=2.0.2 in /usr/local/lib/python3.8/dist-packages (from packaging>=20.0->transformers[sentencepiece]) (3.0.9)\n","Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.8/dist-packages (from requests->transformers[sentencepiece]) (2022.9.24)\n","Requirement already satisfied: urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 in /usr/local/lib/python3.8/dist-packages (from requests->transformers[sentencepiece]) (1.24.3)\n","Requirement already satisfied: chardet<4,>=3.0.2 in /usr/local/lib/python3.8/dist-packages (from requests->transformers[sentencepiece]) (3.0.4)\n","Requirement already satisfied: idna<3,>=2.5 in /usr/local/lib/python3.8/dist-packages (from requests->transformers[sentencepiece]) (2.10)\n","Installing collected packages: sentencepiece\n","Successfully installed sentencepiece-0.1.97\n","Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/\n","Collecting en-core-web-trf==3.2.0\n","  Downloading https://github.com/explosion/spacy-models/releases/download/en_core_web_trf-3.2.0/en_core_web_trf-3.2.0-py3-none-any.whl (460.2 MB)\n","\u001b[K     |████████████████████████████████| 460.2 MB 30 kB/s \n","\u001b[?25hRequirement already satisfied: spacy<3.3.0,>=3.2.0 in /usr/local/lib/python3.8/dist-packages (from en-core-web-trf==3.2.0) (3.2.0)\n","Requirement already satisfied: spacy-transformers<1.2.0,>=1.1.2 in /usr/local/lib/python3.8/dist-packages (from en-core-web-trf==3.2.0) (1.1.7)\n","Requirement already satisfied: spacy-legacy<3.1.0,>=3.0.8 in /usr/local/lib/python3.8/dist-packages (from spacy<3.3.0,>=3.2.0->en-core-web-trf==3.2.0) (3.0.10)\n","Requirement already satisfied: spacy-loggers<2.0.0,>=1.0.0 in /usr/local/lib/python3.8/dist-packages (from spacy<3.3.0,>=3.2.0->en-core-web-trf==3.2.0) (1.0.3)\n","Requirement already satisfied: catalogue<2.1.0,>=2.0.6 in /usr/local/lib/python3.8/dist-packages (from spacy<3.3.0,>=3.2.0->en-core-web-trf==3.2.0) (2.0.8)\n","Requirement already satisfied: jinja2 in /usr/local/lib/python3.8/dist-packages (from spacy<3.3.0,>=3.2.0->en-core-web-trf==3.2.0) (2.11.3)\n","Requirement already satisfied: murmurhash<1.1.0,>=0.28.0 in /usr/local/lib/python3.8/dist-packages (from spacy<3.3.0,>=3.2.0->en-core-web-trf==3.2.0) (1.0.9)\n","Requirement already satisfied: typer<0.5.0,>=0.3.0 in /usr/local/lib/python3.8/dist-packages (from spacy<3.3.0,>=3.2.0->en-core-web-trf==3.2.0) (0.4.2)\n","Requirement already satisfied: pydantic!=1.8,!=1.8.1,<1.9.0,>=1.7.4 in /usr/local/lib/python3.8/dist-packages (from spacy<3.3.0,>=3.2.0->en-core-web-trf==3.2.0) (1.8.2)\n","Requirement already satisfied: setuptools in /usr/local/lib/python3.8/dist-packages (from spacy<3.3.0,>=3.2.0->en-core-web-trf==3.2.0) (57.4.0)\n","Requirement already satisfied: thinc<8.1.0,>=8.0.12 in /usr/local/lib/python3.8/dist-packages (from spacy<3.3.0,>=3.2.0->en-core-web-trf==3.2.0) (8.0.17)\n","Requirement already satisfied: pathy>=0.3.5 in /usr/local/lib/python3.8/dist-packages (from spacy<3.3.0,>=3.2.0->en-core-web-trf==3.2.0) (0.9.0)\n","Requirement already satisfied: wasabi<1.1.0,>=0.8.1 in /usr/local/lib/python3.8/dist-packages (from spacy<3.3.0,>=3.2.0->en-core-web-trf==3.2.0) (0.10.1)\n","Requirement already satisfied: preshed<3.1.0,>=3.0.2 in /usr/local/lib/python3.8/dist-packages (from spacy<3.3.0,>=3.2.0->en-core-web-trf==3.2.0) (3.0.8)\n","Requirement already satisfied: srsly<3.0.0,>=2.4.1 in /usr/local/lib/python3.8/dist-packages (from spacy<3.3.0,>=3.2.0->en-core-web-trf==3.2.0) (2.4.5)\n","Requirement already satisfied: requests<3.0.0,>=2.13.0 in /usr/local/lib/python3.8/dist-packages (from spacy<3.3.0,>=3.2.0->en-core-web-trf==3.2.0) (2.23.0)\n","Requirement already satisfied: packaging>=20.0 in /usr/local/lib/python3.8/dist-packages (from spacy<3.3.0,>=3.2.0->en-core-web-trf==3.2.0) (21.3)\n","Requirement already satisfied: numpy>=1.15.0 in /usr/local/lib/python3.8/dist-packages (from spacy<3.3.0,>=3.2.0->en-core-web-trf==3.2.0) (1.21.6)\n","Requirement already satisfied: blis<0.8.0,>=0.4.0 in /usr/local/lib/python3.8/dist-packages (from spacy<3.3.0,>=3.2.0->en-core-web-trf==3.2.0) (0.7.9)\n","Requirement already satisfied: cymem<2.1.0,>=2.0.2 in /usr/local/lib/python3.8/dist-packages (from spacy<3.3.0,>=3.2.0->en-core-web-trf==3.2.0) (2.0.7)\n","Requirement already satisfied: langcodes<4.0.0,>=3.2.0 in /usr/local/lib/python3.8/dist-packages (from spacy<3.3.0,>=3.2.0->en-core-web-trf==3.2.0) (3.3.0)\n","Requirement already satisfied: tqdm<5.0.0,>=4.38.0 in /usr/local/lib/python3.8/dist-packages (from spacy<3.3.0,>=3.2.0->en-core-web-trf==3.2.0) (4.64.1)\n","Requirement already satisfied: pyparsing!=3.0.5,>=2.0.2 in /usr/local/lib/python3.8/dist-packages (from packaging>=20.0->spacy<3.3.0,>=3.2.0->en-core-web-trf==3.2.0) (3.0.9)\n","Requirement already satisfied: smart-open<6.0.0,>=5.2.1 in /usr/local/lib/python3.8/dist-packages (from pathy>=0.3.5->spacy<3.3.0,>=3.2.0->en-core-web-trf==3.2.0) (5.2.1)\n","Requirement already satisfied: typing-extensions>=3.7.4.3 in /usr/local/lib/python3.8/dist-packages (from pydantic!=1.8,!=1.8.1,<1.9.0,>=1.7.4->spacy<3.3.0,>=3.2.0->en-core-web-trf==3.2.0) (4.1.1)\n","Requirement already satisfied: chardet<4,>=3.0.2 in /usr/local/lib/python3.8/dist-packages (from requests<3.0.0,>=2.13.0->spacy<3.3.0,>=3.2.0->en-core-web-trf==3.2.0) (3.0.4)\n","Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.8/dist-packages (from requests<3.0.0,>=2.13.0->spacy<3.3.0,>=3.2.0->en-core-web-trf==3.2.0) (2022.9.24)\n","Requirement already satisfied: idna<3,>=2.5 in /usr/local/lib/python3.8/dist-packages (from requests<3.0.0,>=2.13.0->spacy<3.3.0,>=3.2.0->en-core-web-trf==3.2.0) (2.10)\n","Requirement already satisfied: urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 in /usr/local/lib/python3.8/dist-packages (from requests<3.0.0,>=2.13.0->spacy<3.3.0,>=3.2.0->en-core-web-trf==3.2.0) (1.24.3)\n","Requirement already satisfied: spacy-alignments<1.0.0,>=0.7.2 in /usr/local/lib/python3.8/dist-packages (from spacy-transformers<1.2.0,>=1.1.2->en-core-web-trf==3.2.0) (0.8.6)\n","Requirement already satisfied: transformers<4.21.0,>=3.4.0 in /usr/local/lib/python3.8/dist-packages (from spacy-transformers<1.2.0,>=1.1.2->en-core-web-trf==3.2.0) (4.20.1)\n","Requirement already satisfied: torch>=1.6.0 in /usr/local/lib/python3.8/dist-packages (from spacy-transformers<1.2.0,>=1.1.2->en-core-web-trf==3.2.0) (1.10.0+cu111)\n","Requirement already satisfied: filelock in /usr/local/lib/python3.8/dist-packages (from transformers<4.21.0,>=3.4.0->spacy-transformers<1.2.0,>=1.1.2->en-core-web-trf==3.2.0) (3.8.0)\n","Requirement already satisfied: pyyaml>=5.1 in /usr/local/lib/python3.8/dist-packages (from transformers<4.21.0,>=3.4.0->spacy-transformers<1.2.0,>=1.1.2->en-core-web-trf==3.2.0) (6.0)\n","Requirement already satisfied: tokenizers!=0.11.3,<0.13,>=0.11.1 in /usr/local/lib/python3.8/dist-packages (from transformers<4.21.0,>=3.4.0->spacy-transformers<1.2.0,>=1.1.2->en-core-web-trf==3.2.0) (0.12.1)\n","Requirement already satisfied: huggingface-hub<1.0,>=0.1.0 in /usr/local/lib/python3.8/dist-packages (from transformers<4.21.0,>=3.4.0->spacy-transformers<1.2.0,>=1.1.2->en-core-web-trf==3.2.0) (0.11.1)\n","Requirement already satisfied: regex!=2019.12.17 in /usr/local/lib/python3.8/dist-packages (from transformers<4.21.0,>=3.4.0->spacy-transformers<1.2.0,>=1.1.2->en-core-web-trf==3.2.0) (2022.6.2)\n","Requirement already satisfied: click<9.0.0,>=7.1.1 in /usr/local/lib/python3.8/dist-packages (from typer<0.5.0,>=0.3.0->spacy<3.3.0,>=3.2.0->en-core-web-trf==3.2.0) (7.1.2)\n","Requirement already satisfied: MarkupSafe>=0.23 in /usr/local/lib/python3.8/dist-packages (from jinja2->spacy<3.3.0,>=3.2.0->en-core-web-trf==3.2.0) (2.0.1)\n","Installing collected packages: en-core-web-trf\n","Successfully installed en-core-web-trf-3.2.0\n","\u001b[38;5;2m✔ Download and installation successful\u001b[0m\n","You can now load the package via spacy.load('en_core_web_trf')\n","Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/\n","Requirement already satisfied: numpy in /usr/local/lib/python3.8/dist-packages (1.21.6)\n"]}],"source":["#change runtime to GPU and after that check CUDA version (current 11.2)\n","#!nvidia-smi\n","\n","# install PyTorch 1.10.0 for CUDA 11.1\n","!pip3 install torch==1.10.0+cu111 -f https://download.pytorch.org/whl/torch_stable.html\n","\n","# install spaCy transformers tuned for CUDA 11.1\n","!pip3 install -U spacy[cuda111,transformers]==3.2.0\n","!pip3 install transformers[sentencepiece]\n","\n","# install spacy transformer pipeline\n","!python -m spacy download en_core_web_trf\n","\n","# library, equivalent of NumPy library for GPU\n","#!pip3 install cupy\n","\n","!pip3 install numpy\n"]},{"cell_type":"code","execution_count":null,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":8261,"status":"ok","timestamp":1669388203575,"user":{"displayName":"Victoria Dimanova","userId":"07172063658264964881"},"user_tz":-60},"id":"mRY4C8EESU3C","outputId":"aa3daaef-140f-4125-ad3a-6d5b6712e9e0"},"outputs":[{"name":"stdout","output_type":"stream","text":["\r⠙ Loading compatibility table...\r\u001b[2K\u001b[38;5;2m✔ Loaded compatibility table\u001b[0m\n","\u001b[1m\n","================= Installed pipeline packages (spaCy v3.2.0) =================\u001b[0m\n","\u001b[38;5;4mℹ spaCy installation: /usr/local/lib/python3.7/dist-packages/spacy\u001b[0m\n","\n","NAME              SPACY            VERSION                            \n","en_core_web_trf   >=3.2.0,<3.3.0   \u001b[38;5;2m3.2.0\u001b[0m   \u001b[38;5;2m✔\u001b[0m\n","en_core_web_sm    >=3.4.0,<3.5.0   \u001b[38;5;3m3.4.1\u001b[0m   --> 3.2.0     \n","\n","\u001b[1m\n","============================== Install updates ==============================\u001b[0m\n","Use the following commands to update the packages:\n","python -m spacy download en_core_web_sm\n","\n"]}],"source":["#validate compatibility of installed modules\n","!python -m spacy validate"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"2mHmi0mDERZ-"},"outputs":[],"source":["#download a current transformer pipeline if needed\n","!python -m spacy download en_core_web_trf"]},{"cell_type":"markdown","metadata":{"id":"5psrh9nVh-fZ"},"source":["STEP 3: Create base configuration that can be adjusted for each model"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"UAmOkQrMH5iJ"},"outputs":[],"source":["#fill in base config with appropriate values. This configuration is needed for training with Spacy\n","!python -m spacy init fill-config /content/drive/MyDrive/SpacyData/base_config.cfg /content/drive/MyDrive/SpacyData/config.cfg "]},{"cell_type":"code","execution_count":null,"metadata":{"id":"syGkuixAgal4"},"outputs":[],"source":["#validate config file to avoid errors during the training\n","!python -m spacy debug data /content/drive/MyDrive/SpacyData/config.cfg"]},{"cell_type":"markdown","metadata":{"id":"Cbd1cj9VSRtT"},"source":["STEP 4: Train models: use various cfg-files to tune different transformers.\n","Default values: train and dev sets, number of epochs = 10.\n"]},{"cell_type":"code","source":["#Tune PubmedBERT\n","!python -m spacy train /content/drive/MyDrive/SpacyData/pubmedbert_config.cfg --output /content/drive/MyDrive/SpacyData/models/pubmed_bert150"],"metadata":{"id":"cmAL1pOPnNSI"},"execution_count":null,"outputs":[]},{"cell_type":"code","execution_count":null,"metadata":{"id":"f7X1WuNR6Fd9"},"outputs":[],"source":["#Tune RoBERTa model\n","!python -m spacy train /content/drive/MyDrive/SpacyData/roberta_config.cfg --output /content/drive/MyDrive/SpacyData/models/roberta_150"]},{"cell_type":"code","execution_count":null,"metadata":{"colab":{"background_save":true,"base_uri":"https://localhost:8080/"},"id":"qils6GD-VeTl","outputId":"2bd74553-d017-48d6-f2fb-951825c4830e"},"outputs":[{"name":"stdout","output_type":"stream","text":["\u001b[38;5;2m✔ Created output directory:\n","/content/drive/MyDrive/SpacyData/models/val_scibert_150\u001b[0m\n","\u001b[38;5;4mℹ Saving to output directory:\n","/content/drive/MyDrive/SpacyData/models/val_scibert_150\u001b[0m\n","\u001b[38;5;4mℹ Using CPU\u001b[0m\n","\u001b[1m\n","=========================== Initializing pipeline ===========================\u001b[0m\n","[2022-11-25 14:59:21,649] [INFO] Set up nlp object from config\n","INFO:spacy:Set up nlp object from config\n","[2022-11-25 14:59:21,663] [INFO] Pipeline: ['transformer', 'ner']\n","INFO:spacy:Pipeline: ['transformer', 'ner']\n","[2022-11-25 14:59:21,668] [INFO] Created vocabulary\n","INFO:spacy:Created vocabulary\n","[2022-11-25 14:59:21,671] [INFO] Finished initializing nlp object\n","INFO:spacy:Finished initializing nlp object\n","Downloading: 100% 385/385 [00:00<00:00, 292kB/s]\n","Downloading: 100% 223k/223k [00:00<00:00, 662kB/s]\n","Downloading: 100% 422M/422M [00:11<00:00, 39.7MB/s]\n","Some weights of the model checkpoint at allenai/scibert_scivocab_uncased were not used when initializing BertModel: ['cls.predictions.bias', 'cls.predictions.transform.dense.weight', 'cls.predictions.decoder.weight', 'cls.predictions.transform.dense.bias', 'cls.predictions.transform.LayerNorm.weight', 'cls.seq_relationship.bias', 'cls.predictions.transform.LayerNorm.bias', 'cls.seq_relationship.weight', 'cls.predictions.decoder.bias']\n","- This IS expected if you are initializing BertModel from the checkpoint of a model trained on another task or with another architecture (e.g. initializing a BertForSequenceClassification model from a BertForPreTraining model).\n","- This IS NOT expected if you are initializing BertModel from the checkpoint of a model that you expect to be exactly identical (initializing a BertForSequenceClassification model from a BertForSequenceClassification model).\n","/usr/local/lib/python3.7/dist-packages/torch/autocast_mode.py:141: UserWarning: User provided device_type of 'cuda', but CUDA is not available. Disabling\n","  warnings.warn('User provided device_type of \\'cuda\\', but CUDA is not available. Disabling')\n","[2022-11-25 14:59:47,342] [INFO] Initialized pipeline components: ['transformer', 'ner']\n","INFO:spacy:Initialized pipeline components: ['transformer', 'ner']\n","\u001b[38;5;2m✔ Initialized pipeline\u001b[0m\n","\u001b[1m\n","============================= Training pipeline =============================\u001b[0m\n","\u001b[38;5;4mℹ Pipeline: ['transformer', 'ner']\u001b[0m\n","\u001b[38;5;4mℹ Initial learn rate: 0.0\u001b[0m\n","E    #       LOSS TRANS...  LOSS NER  ENTS_F  ENTS_P  ENTS_R  SCORE \n","---  ------  -------------  --------  ------  ------  ------  ------\n","  0       0        6204.20   1151.73    0.51    0.27    4.29    0.01\n","  2     200       74866.91  32674.64   55.04   46.57   67.29    0.55\n","  4     400        2421.96   6347.78   56.56   59.67   53.75    0.57\n"]}],"source":["#Tune SciBERT model (allenai/scibert_scivocab_uncased)\n","!python -m spacy train /content/drive/MyDrive/SpacyData/scibert_config.cfg --output /content/drive/MyDrive/SpacyData/models/scibert_150"]}],"metadata":{"colab":{"provenance":[],"mount_file_id":"16jcZu8bPawdX6l5kli8eaR6bznvAldXi","authorship_tag":"ABX9TyM3Ni5bRsz8oMUcz9OyP4f5"},"gpuClass":"standard","kernelspec":{"display_name":"Python 3","name":"python3"},"language_info":{"name":"python"}},"nbformat":4,"nbformat_minor":0}